// @sberbusiness/triplex-next/components/SuggestField/desktop/SuggestFieldDesktop.js
import { SuggestFieldDesktopDropdown as gt } from "./SuggestFieldDesktopDropdown.js";
import { jsxs as q, jsx as g } from "react/jsx-runtime";
import { useState as f, useRef as k, useCallback as s, useEffect as E } from "react";
import { a as ft, A as ht, c as Dt } from "../../../chunks/vendor-Ba_pyhza.js";
import { Tooltip as x } from "../../Tooltip/Tooltip.js";
import { ETooltipSize as bt } from "../../Tooltip/enums.js";
import { SuggestFieldTarget as ut } from "../SuggestFieldTarget.js";
import "../../FormField/FormFieldContext.js";
import { EFormFieldStatus as St } from "../../FormField/enums.js";
import "../../FormField/components/FormFieldDescription.js";
import "../../FormField/components/FormFieldCounter.js";
import "../../FormField/components/FormFieldLabel.js";
import "../../FormField/components/FormFieldClear.js";
import "../../FormField/components/FormFieldTarget.js";
import "../../../chunks/FormFieldInput-CO7yZP0Q.js";
import "../../FormField/components/FormFieldMaskedInput.js";
import "../../FormField/components/FormFieldPostfix.js";
import "../../FormField/components/FormFieldPrefix.js";
import "../../FormField/components/FormFieldTextarea.js";
import "../../Dropdown/Dropdown.js";
import { DropdownListContext as kt } from "../../Dropdown/DropdownListContext.js";
import "../../Dropdown/desktop/DropdownDesktop.js";
import "../../Dropdown/desktop/DropdownList.js";
import "../../../chunks/DropdownListItem-D2hzyBGc.js";
import "../../Dropdown/mobile/DropdownMobile.js";
import "../../Dropdown/mobile/DropdownMobileBody.js";
import "../../Dropdown/mobile/DropdownMobileClose.js";
import "../../Dropdown/mobile/DropdownMobileFooter.js";
import "../../Dropdown/mobile/DropdownMobileHeader.js";
import "../../Dropdown/mobile/DropdownMobileInner.js";
import "../../Dropdown/mobile/DropdownMobileInput.js";
import "../../Dropdown/mobile/DropdownMobileList.js";
import "../../Dropdown/mobile/DropdownMobileListItem.js";
import "../../Loader/LoaderSmall/LoaderSmall.js";
import "../../Loader/LoaderMiddle/LoaderMiddle.js";
import { DataTestId as wt } from "../../../consts/DataTestId.js";
const Lt = "suggestFieldDesktop__83a89ae8", Ct = {
  suggestFieldDesktop: Lt
}, Pt = ({
  className: z,
  value: e,
  options: p,
  size: y,
  status: F,
  label: v,
  placeholder: B,
  loading: G,
  dropdownListLoading: J,
  tooltipHint: K,
  tooltipOpen: Q,
  clearInputOnFocus: A,
  onSelect: d,
  onFilter: h,
  onScrollEnd: R,
  onTargetInputFocus: D,
  onTargetInputBlur: b,
  renderTarget: N,
  renderTargetInput: U,
  renderTargetLabel: V,
  renderTargetPrefix: W,
  renderTargetPostfix: X,
  renderDropdown: H,
  renderDropdownList: Y,
  renderDropdownListItem: Z,
  "data-test-id": c,
  ...T
}) => {
  const [m, l] = f((e == null ? void 0 : e.label) || ""), [u, w] = f(), [i, M] = f(!1), [o, n] = f(!1), [O, S] = f(!1), j = k(ft()), L = k(null), _ = k(null), C = k(R);
  C.current = R;
  const I = s(
    (t) => {
      l((t == null ? void 0 : t.label) || ""), w(void 0), n(!1), S(!0), d(t);
    },
    [d]
  ), tt = s(
    (t) => {
      M(!0), n(!0), A && m.length !== 0 && l(""), D == null || D(t);
    },
    [A, m, D]
  ), et = s(
    (t) => {
      w(void 0), M(!1), n(!1), S(!1), m.length !== 0 ? l((e == null ? void 0 : e.label) || "") : d(void 0), b == null || b(t);
    },
    [e == null ? void 0 : e.label, m, d, b]
  ), ot = s(() => {
    i && !o && (n(!0), S(!1));
  }, [i, o]), rt = s(
    (t) => {
      const r = t.target.value;
      l(r), S(!1), h(r);
    },
    [h]
  ), st = s(() => {
    i ? (l(""), h("")) : d(void 0);
  }, [i, h, d]), it = s(
    (t) => {
      u !== t && w(t);
    },
    [u]
  ), P = s(
    ht((t) => {
      if (C.current === void 0)
        return;
      const r = t.target, a = r.parentElement;
      if (a === null)
        return;
      const dt = r.scrollHeight, lt = r.scrollTop, pt = a.scrollHeight, $ = window.getComputedStyle(a), ct = $.getPropertyValue("padding-top"), mt = $.getPropertyValue("padding-bottom"), at = Number.parseInt(mt) + Number.parseInt(ct);
      Math.abs(lt + pt - at - dt) <= 1 && C.current();
    }, 100),
    []
  );
  E(() => {
    l((e == null ? void 0 : e.label) || "");
  }, [e]), E(() => {
    i && (o && p.length === 0 ? n(!1) : !o && !O && p.length !== 0 && n(!0));
  }, [i, o, O, p.length]), E(() => {
    if (o) {
      const t = _.current;
      return t && t.addEventListener("scroll", P), () => {
        t && t.removeEventListener("scroll", P);
      };
    }
  }, [o, P]);
  const nt = () => {
    const t = Dt(Ct.suggestFieldDesktop, "hoverable", z), r = N === void 0 ? ut : N, a = H === void 0 ? gt : H;
    return /* @__PURE__ */ g("div", { className: t, "data-test-id": c, ...T, ref: L, children: /* @__PURE__ */ q(
      kt.Provider,
      {
        value: { activeDescendant: u, setActiveDescendant: it },
        children: [
          /* @__PURE__ */ g(
            r,
            {
              size: y,
              status: F,
              inputValue: m,
              label: v,
              placeholder: B,
              "aria-controls": j.current,
              "aria-activedescendant": u,
              loading: G,
              dropdownOpen: o,
              onFocus: tt,
              onBlur: et,
              onClick: ot,
              onChange: rt,
              onClear: st,
              renderInput: U,
              renderLabel: V,
              renderPrefix: W,
              renderPostfix: X,
              dataTestId: c
            }
          ),
          /* @__PURE__ */ g(
            a,
            {
              value: e,
              options: p,
              size: y,
              listId: j.current,
              opened: o && p.length > 0,
              listLoading: J,
              listRef: _,
              onSelect: I,
              renderList: Y,
              renderListItem: Z,
              setOpened: n,
              targetRef: L,
              dataTestId: c
            }
          )
        ]
      }
    ) });
  };
  return /* @__PURE__ */ q(
    x,
    {
      size: bt.SM,
      isOpen: !!(Q && i) && F !== St.DISABLED,
      toggle: () => {
      },
      targetRef: L,
      disableAdaptiveMode: !0,
      children: [
        /* @__PURE__ */ g(x.Body, { "data-test-id": c && `${c}${wt.Suggest.tooltip}`, children: K }),
        /* @__PURE__ */ g(x.Target, { children: nt() })
      ]
    }
  );
};
Pt.displayName = "SuggestFieldDesktop";
export {
  Pt as SuggestFieldDesktop
};
