// @sberbusiness/triplex-next/components/SuggestField/desktop/SuggestFieldDesktopDropdown.js
import { jsx as m } from "react/jsx-runtime";
import { Dropdown as $ } from "../../Dropdown/Dropdown.js";
import "../../Dropdown/DropdownListContext.js";
import "../../Dropdown/desktop/DropdownDesktop.js";
import { DropdownList as h } from "../../Dropdown/desktop/DropdownList.js";
import { D as C } from "../../../chunks/DropdownListItem-D2hzyBGc.js";
import "../../Dropdown/mobile/DropdownMobile.js";
import "../../Dropdown/mobile/DropdownMobileBody.js";
import "../../Dropdown/mobile/DropdownMobileClose.js";
import "../../Dropdown/mobile/DropdownMobileFooter.js";
import "../../Dropdown/mobile/DropdownMobileHeader.js";
import "../../Dropdown/mobile/DropdownMobileInner.js";
import "../../Dropdown/mobile/DropdownMobileInput.js";
import "../../Dropdown/mobile/DropdownMobileList.js";
import "../../Dropdown/mobile/DropdownMobileListItem.js";
import "../../Loader/LoaderSmall/LoaderSmall.js";
import "../../Loader/LoaderMiddle/LoaderMiddle.js";
import { EVENT_KEY_CODES as L } from "../../../utils/keyboard.js";
import { DataTestId as d } from "../../../consts/DataTestId.js";
const _ = [L.ENTER], G = ({
  size: e,
  value: t,
  options: i,
  targetRef: s,
  listId: c,
  dataTestId: r,
  opened: p,
  listLoading: E,
  listRef: x,
  renderList: n,
  renderListItem: D,
  onSelect: f,
  setOpened: l,
  ...w
}) => {
  const S = n === void 0 ? h : n, g = D === void 0 ? C : D;
  return /* @__PURE__ */ m(
    $,
    {
      size: e,
      targetRef: s,
      "data-test-id": r && `${r}${d.Suggest.dropdown}`,
      opened: p,
      fixedWidth: !0,
      setOpened: l,
      ...w,
      children: /* @__PURE__ */ m(
        S,
        {
          id: c,
          size: e,
          dropdownOpened: p,
          loading: E,
          onMouseDown: (o) => o.preventDefault(),
          children: i == null ? void 0 : i.map((o) => /* @__PURE__ */ m(
            g,
            {
              id: o.id,
              keyCodesForSelection: _,
              "data-test-id": r && `${r}${d.Suggest.dropdown}${d.Dropdown.listItem}`,
              selected: o.id === (t == null ? void 0 : t.id),
              onMouseDown: (u) => u.preventDefault(),
              onSelect: () => f(o),
              children: o.content || o.label
            },
            o.id
          ))
        }
      )
    }
  );
};
export {
  G as SuggestFieldDesktopDropdown
};
