// @sberbusiness/triplex-next/components/Link/Link.js
import { jsx as R, jsxs as t, Fragment as i } from "react/jsx-runtime";
import L from "react";
import { c as a } from "../../chunks/vendor-Ba_pyhza.js";
const v = "link__81e5301c", F = "wordWithContent__0a1c37b4", S = "before__2ff2f479", T = "after__61cfb574", e = {
  link: v,
  wordWithContent: F,
  before: S,
  after: T
}, q = L.forwardRef(
  ({ children: d, className: f, onBlur: N, onMouseDown: p, contentAfter: s, contentBefore: o, ...C }, _) => {
    const c = () => o ? /* @__PURE__ */ t(i, { children: [
      "​",
      o()
    ] }) : null, m = () => s ? s() : null, u = (n) => {
      const r = n.split(" ");
      if (r.length < 2 || r.length < 3 && o && s) {
        const y = a(e.wordWithContent, {
          [e.after]: !!s,
          [e.before]: !!o
        });
        return /* @__PURE__ */ t("span", { className: y, children: [
          c(),
          n,
          m()
        ] });
      }
      const l = r[0], h = r[r.length - 1], w = r.slice(1, -1).join(" "), k = a(e.wordWithContent, {
        [e.before]: !!o
      }), g = o ? /* @__PURE__ */ t("span", { className: k, children: [
        c(),
        l
      ] }) : l, x = a(e.wordWithContent, {
        [e.after]: !!s
      }), j = s ? /* @__PURE__ */ t("span", { className: x, children: [
        h,
        m()
      ] }) : h;
      return /* @__PURE__ */ t(i, { children: [
        g,
        " ",
        w,
        " ",
        j
      ] });
    }, W = (n) => {
      const r = o ? o() : null, l = s ? s() : null;
      return /* @__PURE__ */ t(i, { children: [
        r,
        n,
        l
      ] });
    }, b = o || s ? ((n) => typeof n == "string" ? u(n) : W(n))(d) : d;
    return /* @__PURE__ */ R(
      "a",
      {
        role: "link",
        ...C,
        className: a(f, e.link, "hoverable"),
        onBlur: N,
        onMouseDown: p,
        "data-tx": "1.4.0",
        ref: _,
        children: b
      }
    );
  }
);
q.displayName = "Link";
export {
  q as Link
};
