// @sberbusiness/triplex-next/components/SelectExtendedField/components/SelectExtendedFieldTarget.js
import { jsxs as N, jsx as l } from "react/jsx-runtime";
import L, { useRef as O } from "react";
import { CaretdownStrokeSrvIcon24 as A } from "@sberbusiness/icons-next";
import { EVENT_KEY_CODES as n } from "../../../utils/keyboard.js";
import { LoaderSmall as C } from "../../Loader/LoaderSmall/LoaderSmall.js";
import { ELoaderSmallTheme as y } from "../../Loader/LoaderSmall/enum.js";
import "../../Loader/LoaderMiddle/LoaderMiddle.js";
import { c as E } from "../../../chunks/vendor-Ba_pyhza.js";
import { FormField as B } from "../../FormField/FormField.js";
import "../../FormField/components/FormFieldDescription.js";
import "../../FormField/components/FormFieldCounter.js";
import { FormFieldLabel as P } from "../../FormField/components/FormFieldLabel.js";
import "../../FormField/components/FormFieldClear.js";
import { FormFieldTarget as k } from "../../FormField/components/FormFieldTarget.js";
import "../../../chunks/FormFieldInput-CO7yZP0Q.js";
import "../../FormField/components/FormFieldMaskedInput.js";
import { FormFieldPostfix as w } from "../../FormField/components/FormFieldPostfix.js";
import "../../FormField/components/FormFieldPrefix.js";
import "../../FormField/components/FormFieldTextarea.js";
import { EFormFieldStatus as s } from "../../FormField/enums.js";
import { EComponentSize as z } from "../../../enums/EComponentSize.js";
const W = "selectExtendedFieldTarget__08e42c17", j = "caretIcon__cd1cc205", K = "selectOpened__47e6ebed", G = "loading__68ee4945", U = "disabled__b849031d", t = {
  selectExtendedFieldTarget: W,
  caretIcon: j,
  selectOpened: K,
  loading: G,
  disabled: U
}, V = L.forwardRef(
  (x, a) => {
    const {
      children: Y,
      className: g,
      label: i,
      placeholder: c,
      onKeyDown: m,
      onClick: p,
      opened: o,
      setOpened: f,
      loading: r,
      size: F,
      status: d,
      tabIndex: _,
      fieldLabel: u,
      ...b
    } = x, S = O(null), D = E(
      t.selectExtendedFieldTarget,
      {
        [t.selectOpened]: o,
        [t.loading]: r,
        [t.disabled]: d === s.DISABLED
      },
      g
    ), I = () => d === s.DISABLED || r ? -1 : _ || 0, h = (e) => {
      r && d === s.DISABLED || (f(!o), p == null || p(e));
    }, R = (e) => {
      r && d === s.DISABLED || (o || [
        n.SPACE,
        n.ENTER,
        n.ARROW_DOWN,
        n.ARROW_UP
      ].includes(e.keyCode) && (e.preventDefault(), e.stopPropagation(), f(!o)), m == null || m(e));
    }, T = (e) => {
      S.current = e, typeof a == "function" ? a(e) : a && (a.current = e);
    };
    return /* @__PURE__ */ N(
      B,
      {
        onClick: h,
        tabIndex: I(),
        onKeyDown: R,
        status: d,
        size: F,
        className: D,
        "aria-expanded": o,
        "aria-haspopup": "listbox",
        "data-tx": "1.4.0",
        ...b,
        children: [
          /* @__PURE__ */ l(P, { floating: !!i || !!c, children: u }),
          /* @__PURE__ */ l(
            k,
            {
              ref: T,
              className: E(t.target, {
                [t.placeholder]: !!c && !i,
                [t.label]: !!i
              }),
              placeholder: c,
              children: i
            }
          ),
          /* @__PURE__ */ l(w, { children: r ? /* @__PURE__ */ l(C, { size: z.LG, theme: y.BRAND }) : /* @__PURE__ */ l(A, { paletteIndex: 5, className: t.caretIcon }) })
        ]
      }
    );
  }
);
V.displayName = "SelectExtendedFieldTarget";
export {
  V as SelectExtendedFieldTarget
};
