// @sberbusiness/triplex-next/components/SelectExtendedField/components/SelectExtendedFieldDropdownDefault.js
import { jsx as e, jsxs as M, Fragment as S } from "react/jsx-runtime";
import { createElement as s } from "react";
import { SelectExtendedField as m } from "../SelectExtendedField.js";
import "../../Dropdown/mobile/DropdownMobile.js";
import { DropdownMobileBody as y } from "../../Dropdown/mobile/DropdownMobileBody.js";
import { DropdownMobileClose as E } from "../../Dropdown/mobile/DropdownMobileClose.js";
import "../../Dropdown/mobile/DropdownMobileFooter.js";
import { DropdownMobileHeader as k } from "../../Dropdown/mobile/DropdownMobileHeader.js";
import "../../Dropdown/mobile/DropdownMobileInner.js";
import "../../Dropdown/mobile/DropdownMobileInput.js";
import { DropdownMobileList as B } from "../../Dropdown/mobile/DropdownMobileList.js";
import { DropdownMobileListItem as F } from "../../Dropdown/mobile/DropdownMobileListItem.js";
import "../../Loader/LoaderSmall/LoaderSmall.js";
import "../../Loader/LoaderMiddle/LoaderMiddle.js";
import { Text as L } from "../../Typography/Text.js";
import { ETextSize as j } from "../../Typography/enums.js";
const Q = ({
  dropdownRef: w,
  dropdownListItemClassName: l,
  fixedWidth: c,
  loading: D,
  listId: b,
  mobileTitle: h,
  onChange: n,
  opened: p,
  size: x,
  options: f,
  setOpened: i,
  targetRef: a,
  value: o
}) => /* @__PURE__ */ e(
  m.Dropdown,
  {
    opened: p && !D,
    forwardedRef: w,
    fixedWidth: typeof c > "u" ? !0 : c,
    setOpened: i,
    targetRef: a,
    size: x,
    mobileViewProps: {
      children: /* @__PURE__ */ M(S, { children: [
        /* @__PURE__ */ e(
          k,
          {
            closeButton: () => /* @__PURE__ */ e(E, { onClick: () => i(!1) }),
            children: /* @__PURE__ */ e(L, { tag: "div", size: j.B1, children: h })
          }
        ),
        /* @__PURE__ */ e(y, { children: /* @__PURE__ */ e(B, { children: f.map((r) => {
          const { label: d, ...t } = r;
          return /* @__PURE__ */ s(
            F,
            {
              ...t,
              className: l,
              id: r.id,
              key: r.id,
              selected: r.id === (o == null ? void 0 : o.id),
              onSelect: () => {
                n(r), i(!1);
              }
            },
            d
          );
        }) }) })
      ] })
    },
    children: /* @__PURE__ */ e(m.Dropdown.List, { id: b, dropdownOpened: p, children: f.map((r) => {
      const { label: d, ...t } = r;
      return /* @__PURE__ */ s(
        m.Dropdown.List.Item,
        {
          ...t,
          className: l,
          id: r.id,
          key: r.id,
          selected: r.id === (o == null ? void 0 : o.id),
          onSelect: () => {
            n(r), i(!1);
          }
        },
        d
      );
    }) })
  }
);
export {
  Q as SelectExtendedFieldDropdownDefault
};
