// @sberbusiness/triplex-next/components/Loader/LoaderSmall/LoaderSmall.js
import { jsxs as r, jsx as t } from "react/jsx-runtime";
import { c as o } from "../../../chunks/vendor-Ba_pyhza.js";
import { ELoaderSmallTheme as s } from "./enum.js";
import { createSizeToClassNameMap as n } from "../../../utils/classNameMaps.js";
const c = "loaderSmall__64fa4e1f", _ = "sm__0acad274", p = "dot__92555a29", i = "md__ae815afc", N = "lg__0464ee2e", f = "neutral__ed963c7b", b = "brand__30995eb3", S = "dot1__ef208edd", u = "dot2__c8a9a5f8", T = "dot3__c33b0654", a = {
  loaderSmall: c,
  sm: _,
  dot: p,
  md: i,
  lg: N,
  neutral: f,
  brand: b,
  dot1: S,
  dot2: u,
  dot3: T
}, g = {
  [s.BRAND]: a.brand,
  [s.NEUTRAL]: a.neutral
}, x = n(a), L = ({ className: e, theme: l, size: d, ...m }) => /* @__PURE__ */ r(
  "span",
  {
    className: o(a.loaderSmall, g[l], x[d], e),
    role: "status",
    "aria-label": "loading",
    ...m,
    children: [
      /* @__PURE__ */ t("span", { className: o(a.dot, a.dot1) }),
      /* @__PURE__ */ t("span", { className: o(a.dot, a.dot2) }),
      /* @__PURE__ */ t("span", { className: o(a.dot, a.dot3) })
    ]
  }
);
L.displayName = "LoaderSmall";
export {
  L as LoaderSmall
};
