// @sberbusiness/triplex-next/components/Button/ButtonDropdown.js
import { Button as x } from "./Button.js";
import "./ButtonBase.js";
import "./ButtonIcon.js";
import { jsx as o, jsxs as N, Fragment as q } from "react/jsx-runtime";
import F, { useRef as _, useState as V, createElement as M } from "react";
import { c as f, a as J } from "../../chunks/vendor-Ba_pyhza.js";
import { ButtonDropdownExtended as y } from "./ButtonDropdownExtended.js";
import { EButtonDotsTheme as u, EButtonTheme as d } from "./enums.js";
import { DotshorizontalStrokeSrvIcon32 as Q, DotshorizontalStrokeSrvIcon20 as B, CaretdownStrokeSrvIcon24 as X, CaretdownStrokeSrvIcon20 as Z, CaretdownStrokeSrvIcon16 as $ } from "@sberbusiness/icons-next";
import { isKey as D } from "../../utils/keyboard.js";
import { DropdownListContext as ee } from "../Dropdown/DropdownListContext.js";
import { DropdownMobileHeader as te } from "../Dropdown/mobile/DropdownMobileHeader.js";
import { DropdownMobileBody as oe } from "../Dropdown/mobile/DropdownMobileBody.js";
import { DropdownMobileList as re } from "../Dropdown/mobile/DropdownMobileList.js";
import { DropdownMobileListItem as ne } from "../Dropdown/mobile/DropdownMobileListItem.js";
import { DropdownMobileClose as ae } from "../Dropdown/mobile/DropdownMobileClose.js";
import { Text as ce } from "../Typography/Text.js";
import { ETextSize as se } from "../Typography/enums.js";
import { DropdownList as L } from "../Dropdown/desktop/DropdownList.js";
import { EComponentSize as i } from "../../enums/EComponentSize.js";
const le = "buttonDropdown__025b9ea2", de = "block__dbaf919a", ie = "buttonDropdownTarget__7f03c16a", me = "caretIcon__00b977ec", ue = "active__0410ebaf", De = "buttonDropdownMenuItem__31df2b01", n = {
  buttonDropdown: le,
  block: de,
  buttonDropdownTarget: ie,
  caretIcon: me,
  active: ue,
  buttonDropdownMenuItem: De
}, pe = {
  [i.SM]: /* @__PURE__ */ o(B, { paletteIndex: 0 }),
  [i.MD]: /* @__PURE__ */ o(B, { paletteIndex: 0 }),
  [i.LG]: /* @__PURE__ */ o(Q, { paletteIndex: 0 })
}, Y = {
  [i.SM]: /* @__PURE__ */ o($, { paletteIndex: 7, className: n.caretIcon }),
  [i.MD]: /* @__PURE__ */ o(Z, { paletteIndex: 7, className: n.caretIcon }),
  [i.LG]: /* @__PURE__ */ o(X, { paletteIndex: 7, className: n.caretIcon })
}, be = F.forwardRef(
  (g, p) => {
    const { buttonAttributes: R, children: h, className: z, theme: b, size: c, options: C, selected: l, block: w, disabled: T, ...G } = g, k = _(null), v = _(null), W = f(n.buttonDropdown, { [n.block]: !!w }, z), [S, H] = V(), I = _(J()), P = ({ opened: e, setOpened: r }) => {
      const a = f(n.buttonDropdownTarget, "hoverable", {
        [n.active]: e,
        [n.block]: !!w
      });
      return /* @__PURE__ */ N(
        x,
        {
          className: a,
          theme: b,
          size: c,
          onKeyDown: A({ opened: e, setOpened: r }),
          onClick: E({ opened: e, setOpened: r }),
          disabled: T,
          "aria-haspopup": "menu",
          "aria-expanded": e,
          "aria-controls": I.current,
          "aria-activedescendant": S,
          ...R,
          ref: O,
          children: [
            h,
            j()
          ]
        }
      );
    }, K = ({ opened: e, setOpened: r }) => {
      const a = f(n.buttonDropdownTarget, "hoverable", {
        [n.active]: e,
        [n.block]: !!w
      });
      return /* @__PURE__ */ o(
        x,
        {
          className: a,
          theme: b === u.DOTS_SECONDARY ? d.SECONDARY : d.SECONDARY_LIGHT,
          size: c,
          onKeyDown: A({ opened: e, setOpened: r }),
          onClick: E({ opened: e, setOpened: r }),
          disabled: T,
          "aria-haspopup": "menu",
          "aria-expanded": e,
          "aria-controls": I.current,
          "aria-activedescendant": S,
          ...R,
          ref: O,
          icon: pe[c]
        }
      );
    }, E = ({ opened: e, setOpened: r }) => () => r(!e), A = ({ opened: e, setOpened: r }) => (a) => {
      const { key: s } = a;
      (D(s, "SPACE") || D(s, "ARROW_UP") || D(s, "ARROW_DOWN")) && a.preventDefault(), !e && (D(s, "ARROW_UP") || D(s, "ARROW_DOWN")) && r(!0);
    }, j = () => {
      switch (b) {
        case d.GENERAL:
        case d.DANGER:
        case u.DOTS_SECONDARY:
        case u.DOTS_SECONDARY_LIGHT:
          return Y[c];
        case d.SECONDARY:
        case d.SECONDARY_LIGHT:
          return Y[c];
        default:
          return null;
      }
    }, U = ({ opened: e, setOpened: r, className: a }) => {
      const s = f(n.buttonDropdownMenu, a);
      return /* @__PURE__ */ o(ee.Provider, { value: { activeDescendant: S, setActiveDescendant: H }, children: /* @__PURE__ */ o(
        y.Dropdown,
        {
          size: c,
          className: s,
          opened: e,
          setOpened: r,
          targetRef: k,
          ref: v,
          mobileViewProps: {
            children: /* @__PURE__ */ N(q, { children: [
              /* @__PURE__ */ N(te, { children: [
                /* @__PURE__ */ o(ce, { tag: "div", size: se.B3, children: h }),
                /* @__PURE__ */ o(ae, { onClick: () => r(!1) })
              ] }),
              /* @__PURE__ */ o(oe, { children: /* @__PURE__ */ o(re, { children: C.map((t) => /* @__PURE__ */ M(
                ne,
                {
                  ...t,
                  key: t.id,
                  selected: t.id === (l == null ? void 0 : l.id),
                  onSelect: () => {
                    var m;
                    (m = t.onSelect) == null || m.call(t), r(!1);
                  }
                },
                t.label
              )) }) })
            ] })
          },
          children: /* @__PURE__ */ o(L, { dropdownOpened: e, id: I.current, size: c, children: C.map((t) => /* @__PURE__ */ M(
            L.Item,
            {
              ...t,
              className: n.buttonDropdownMenuItem,
              key: t.id,
              selected: t.id === (l == null ? void 0 : l.id),
              onSelect: () => {
                var m;
                (m = t.onSelect) == null || m.call(t), r(!1);
              }
            },
            t.label
          )) })
        }
      ) });
    }, O = (e) => {
      k.current = e, typeof p == "function" ? p(e) : p && (p.current = e);
    };
    return /* @__PURE__ */ o(
      y,
      {
        className: W,
        renderButton: [u.DOTS_SECONDARY, u.DOTS_SECONDARY_LIGHT].includes(
          b
        ) ? K : P,
        renderDropdown: U,
        dropdownRef: v,
        closeOnTab: !0,
        ...G
      }
    );
  }
);
be.displayName = "ButtonDropdown";
export {
  be as ButtonDropdown
};
