// @sberbusiness/triplex-next/components/Button/Button.js
import { jsxs as m, jsx as y } from "react/jsx-runtime";
import k from "react";
import { c as p } from "../../chunks/vendor-Ba_pyhza.js";
import { LoaderSmall as R } from "../Loader/LoaderSmall/LoaderSmall.js";
import { ELoaderSmallTheme as g } from "../Loader/LoaderSmall/enum.js";
import "../Loader/LoaderMiddle/LoaderMiddle.js";
import { ButtonBase as C } from "./ButtonBase.js";
import { EButtonTheme as o } from "./enums.js";
import { EComponentSize as d } from "../../enums/EComponentSize.js";
const A = "general__352d5337", B = "expanded__50b96a56", u = {
  general: A,
  expanded: B
}, D = "secondary__961a3c28", T = "expanded__50b96a56", b = {
  secondary: D,
  expanded: T
}, G = "secondaryLight__20282cb9", I = "expanded__50b96a56", x = {
  secondaryLight: G,
  expanded: I
}, v = "danger__ae19eb3a", O = "expanded__50b96a56", h = {
  danger: v,
  expanded: O
}, Y = "link__0827d549", $ = "expanded__50b96a56", f = {
  link: Y,
  expanded: $
}, w = "button__d729a0e8", z = "sm__2720d538", M = "icon__e1bdf7a9", j = "md__95523687", H = "lg__1677cae0", K = "block__dbaf919a", U = "content__6f62233b", q = "loading__c2f82391", F = "loader__d887fb1a", J = "hidden__2efe2c8c", e = {
  button: w,
  sm: z,
  icon: M,
  md: j,
  lg: H,
  block: K,
  content: U,
  loading: q,
  loader: F,
  hidden: J
}, P = (t, n) => {
  switch (t) {
    case o.GENERAL:
      return { [u.general]: !0, [u.expanded]: n };
    case o.SECONDARY:
      return {
        [b.secondary]: !0,
        [b.expanded]: n
      };
    case o.SECONDARY_LIGHT:
      return {
        [x.secondaryLight]: !0,
        [x.expanded]: n
      };
    case o.DANGER:
      return {
        [h.danger]: !0,
        [h.expanded]: n
      };
    case o.LINK:
      return {
        [f.link]: !0,
        [f.expanded]: n
      };
  }
}, Q = (t) => {
  switch (t) {
    case d.LG:
      return e.lg;
    case d.MD:
      return e.md;
    case d.SM:
      return e.sm;
  }
}, V = (t, n) => {
  const a = [o.SECONDARY, o.SECONDARY_LIGHT].includes(t) ? g.BRAND : g.NEUTRAL;
  return /* @__PURE__ */ y(R, { theme: a, size: n });
}, W = k.forwardRef((t, n) => {
  const { children: a, className: N, disabled: c, theme: r, size: l = d.MD, block: L, loading: s, icon: i, ...S } = t, { "aria-expanded": _ } = t, E = p(
    e.button,
    P(r, !!_),
    Q(l),
    { [e.block]: !!L, [e.loading]: !!s },
    { [e.icon]: !!i && !a },
    // Классы для иконок, начало.
    "hoverable",
    {
      active: !!_,
      disabled: !!c
    },
    // Классы для иконок, конец.
    N
  );
  return /* @__PURE__ */ m(C, { className: E, tabIndex: s ? -1 : void 0, disabled: c, ...S, ref: n, children: [
    /* @__PURE__ */ m("span", { className: e.content, children: [
      i,
      a
    ] }),
    /* @__PURE__ */ y("div", { className: p(e.loader, !s && e.hidden), children: V(r, l) })
  ] });
});
W.displayName = "Button";
export {
  W as Button
};
