// @sberbusiness/triplex-next/components/Button/ButtonDropdownExtended.js
import { jsxs as h } from "react/jsx-runtime";
import { useRef as f, useState as C, useCallback as N, useEffect as R } from "react";
import { Dropdown as _ } from "../Dropdown/Dropdown.js";
import { isKey as w } from "../../utils/keyboard.js";
import { DropdownList as g } from "../Dropdown/desktop/DropdownList.js";
import { c as A } from "../../chunks/vendor-Ba_pyhza.js";
const K = "buttonDropdownExtended__471c326e", S = "buttonDropdownExtendedBlock__f13c5176", E = {
  buttonDropdownExtended: K,
  buttonDropdownExtendedBlock: S
}, D = (s) => {
  const { className: v, opened: x, setOpened: c, renderButton: k, renderDropdown: b, dropdownRef: l, closeOnTab: p, ...L } = s, u = f(null), m = f(s.opened !== void 0), [O, y] = C((s.opened === void 0, !1)), e = m.current ? !!x : O, o = N(
    (t) => {
      if (m.current) {
        c && c(t);
        return;
      }
      y(t);
    },
    [c]
  );
  R(() => {
    if (!e)
      return;
    const t = (r) => {
      const n = r.code || r.keyCode;
      e && (w(n, "ESCAPE") || p && w(n, "TAB")) && o(!1);
    }, d = (r) => {
      const n = u.current, i = l.current;
      if (e) {
        const a = r.target;
        a && !(n != null && n.contains(a)) && !(i != null && i.contains(a)) && o(!1);
      }
    };
    return document.addEventListener("keydown", t), document.addEventListener("mousedown", d), document.addEventListener("touchstart", d), () => {
      document.removeEventListener("keydown", t), document.removeEventListener("mousedown", d), document.removeEventListener("touchstart", d);
    };
  }, [e, p, l, o, u]);
  const B = A(E.buttonDropdownExtended, v);
  return /* @__PURE__ */ h("div", { className: B, ref: u, ...L, children: [
    k({ opened: e, setOpened: o }),
    b({
      className: E.buttonDropdownExtendedBlock,
      opened: e,
      setOpened: o
    })
  ] });
};
D.Dropdown = _;
D.DropdownList = g;
export {
  D as ButtonDropdownExtended
};
