// @sberbusiness/triplex-next/components/LightBox/LightBoxSideOverlay/LightBoxSideOverlay.js
import { jsx as i, jsxs as B, Fragment as W } from "react/jsx-runtime";
import { useState as O, useRef as x, useEffect as L } from "react";
import { c as _, F as X } from "../../../chunks/vendor-Ba_pyhza.js";
import { OverlayBase as Y, EOverlayDirection as R } from "../../Overlay/OverlayBase.js";
import { OverlayMask as j } from "../../Overlay/OverlayMask.js";
import { LightBoxSideOverlayLoader as I } from "./LightBoxSideOverlayLoader.js";
import { FocusTrapUtils as T } from "../../../utils/focus/FocusTrapUtils.js";
import { LightBoxSideOverlayCloseMobile as A } from "./LightBoxSideOverlayCloseMobile.js";
import { LightBoxSideOverlayCloseDesktop as U } from "./LightBoxSideOverlayCloseDesktop.js";
const q = "lightBoxSideOverlayMask__de7cade6", E = "lightBoxSideOverlayWrapper__13dd0b96", J = "overflowXHidden__52e21450", K = "overflowYHidden__43f25ae1", Q = "closing__a7cdf5fb", V = "opened__700be6cf", Z = "lightBoxSideOverlayContent__55d6cff1", $ = "opening__0fdcbef9", P = "sizeSM__f9ea30b2", ee = "sizeMD__08b1884a", te = "sizeLG__6e960630", oe = "lightBoxSideOverlay__38639a17", e = {
  lightBoxSideOverlayMask: q,
  lightBoxSideOverlayWrapper: E,
  overflowXHidden: J,
  overflowYHidden: K,
  closing: Q,
  opened: V,
  lightBoxSideOverlayContent: Z,
  opening: $,
  sizeSM: P,
  sizeMD: ee,
  sizeLG: te,
  lightBoxSideOverlay: oe
};
var ie = /* @__PURE__ */ ((t) => (t[t.SM = 0] = "SM", t[t.MD = 1] = "MD", t[t.LG = 2] = "LG", t))(ie || {});
const h = ({
  children: t,
  className: M,
  focusTrapProps: l,
  isLoading: a,
  loadingTitle: S,
  isTopLevelSideOverlayOpened: p,
  isTopOverlayOpened: u,
  onClose: c,
  onOpen: d,
  opened: o,
  size: f = 1,
  ...D
}) => {
  const [r, v] = O(!1), [s, g] = O(!1), m = x(o), y = x(null);
  L(() => {
    m.current && !o ? (g(!1), v(!0)) : !m.current && o && (v(!1), g(!0)), m.current = o;
  }, [o]);
  const k = (n) => {
    const { target: z, currentTarget: C } = n;
    z === C && (r ? (v(!1), c == null || c()) : s && (g(!1), d == null || d()));
  }, H = ({ opened: n }) => /* @__PURE__ */ i(j, { opened: n, className: e.lightBoxSideOverlayMask }), b = () => /* @__PURE__ */ B(
    "div",
    {
      className: _(e.lightBoxSideOverlayContent, {
        [e.closing]: r,
        [e.opening]: s,
        [e.opened]: o
      }),
      onTransitionEnd: k,
      ref: y,
      children: [
        t,
        a && /* @__PURE__ */ i(I, { loadingTitle: S })
      ]
    }
  ), w = () => {
  }, G = _(M, e.lightBoxSideOverlayWrapper, {
    [e.closing]: r,
    [e.opened]: o,
    [e.overflowXHidden]: !!p || !!a,
    [e.overflowYHidden]: !!p || !!a || !!u,
    [e.sizeSM]: f === 0,
    [e.sizeMD]: f === 1,
    [e.sizeLG]: f === 2
    /* LG */
  }), F = (n) => /* @__PURE__ */ B(W, { children: [
    H(n),
    b()
  ] }), N = /* @__PURE__ */ i(
    "div",
    {
      className: _(e.lightBoxSideOverlay, {
        [e.closing]: r,
        [e.opening]: s
      }),
      children: /* @__PURE__ */ i(Y, { direction: R.RIGHT, opened: o, setOpened: w, children: F })
    }
  );
  return /* @__PURE__ */ i(
    X,
    {
      active: o && !s && !r,
      ...l,
      focusTrapOptions: {
        clickOutsideDeactivates: !0,
        initialFocus: () => T.getFirstInteractionElementByDataAttr(y.current),
        preventScroll: !0,
        ...l == null ? void 0 : l.focusTrapOptions
      },
      children: /* @__PURE__ */ i("div", { className: G, role: "dialog", "aria-modal": "true", ...D, children: N })
    }
  );
};
h.displayName = "LightBoxSideOverlay";
h.CloseDesktop = U;
h.CloseMobile = A;
export {
  ie as ELightBoxSideOverlaySize,
  h as LightBoxSideOverlay
};
