// @sberbusiness/triplex-next/components/LightBox/LightBoxSideOverlay/LightBoxSideOverlayCloseDesktop.js
import { jsx as e } from "react/jsx-runtime";
import { useRef as s } from "react";
import { TriggerClickOnKeyDownEvent as n } from "../../Triggers/TriggerClickOnKeyDownEvent.js";
import { EVENT_KEY_CODES as l } from "../../../utils/keyboard.js";
import { c as p } from "../../../chunks/vendor-Ba_pyhza.js";
import { CrossStrokeSrvIcon32 as f } from "@sberbusiness/icons-next";
import { EButtonTheme as a } from "../../Button/enums.js";
import { EComponentSize as c } from "../../../enums/EComponentSize.js";
import { Button as d } from "../../Button/Button.js";
import { s as u } from "../../../chunks/LightBoxSideOverlayClose.module-B_oIB3rc.js";
const C = ({
  className: r,
  clickByEsc: m,
  ...i
}) => {
  const o = s(null), t = () => /* @__PURE__ */ e(
    d,
    {
      "data-exclude-modal-focus": !0,
      className: p(r, u.lightBoxSideOverlayCloseDesktop),
      title: "Закрыть",
      ...i,
      ref: o,
      icon: /* @__PURE__ */ e(f, { paletteIndex: 0 }),
      size: c.LG,
      theme: a.SECONDARY_LIGHT
    }
  );
  return m ? /* @__PURE__ */ e(n, { eventKeyCode: l.ESCAPE, targetRef: o, children: t() }) : t();
};
C.displayName = "LightBoxSideOverlayCloseDesktop";
export {
  C as LightBoxSideOverlayCloseDesktop
};
