// @sberbusiness/triplex-next/components/Checkbox/Checkbox.js
import { jsxs as I, jsx as e } from "react/jsx-runtime";
import N from "react";
import { CheckboxbulkStrokeSrvIcon24 as z, CheckboxtickStrokeSrvIcon24 as C } from "@sberbusiness/icons-next";
import "../Typography/Title.js";
import { Text as S } from "../Typography/Text.js";
import "../Typography/Caption.js";
import { ETextSize as l } from "../Typography/enums.js";
import { EComponentSize as t } from "../../enums/EComponentSize.js";
import { createSizeToClassNameMap as u } from "../../utils/classNameMaps.js";
import { c as r } from "../../chunks/vendor-Ba_pyhza.js";
const y = "label__a7f16d39", T = "md__8c013058", M = "nonempty__c5c13c4e", g = "lg__31e4a4c4", j = "disabled__cb080a13", v = "checkbox__bf68ea73", B = "checkboxIcon__e1d726a3", D = "checkmarkIcon__5bb5c14a", c = {
  label: y,
  md: T,
  nonempty: M,
  lg: g,
  disabled: j,
  checkbox: v,
  checkboxIcon: B,
  checkmarkIcon: D
}, E = {
  [t.LG]: l.B2,
  [t.MD]: l.B3
}, b = u(c), L = N.forwardRef((i, k) => {
  const {
    children: s,
    className: p,
    disabled: n,
    bulk: d,
    labelAttributes: o,
    size: a = t.MD,
    ...x
  } = i, h = r(c.checkbox, p, b[a]), _ = r(
    c.label,
    b[a],
    { [c.disabled]: !!n, [c.nonempty]: !!s },
    o == null ? void 0 : o.className
  ), f = () => {
    const m = c.checkmarkIcon;
    return d ? /* @__PURE__ */ e(z, { className: m, paletteIndex: 7 }) : /* @__PURE__ */ e(C, { className: m, paletteIndex: 7 });
  };
  return /* @__PURE__ */ I("label", { ...o, className: _, "data-tx": "1.4.0", children: [
    /* @__PURE__ */ e("input", { type: "checkbox", className: h, disabled: n, ...x, ref: k }),
    /* @__PURE__ */ e("span", { className: c.checkboxIcon }),
    f(),
    s && /* @__PURE__ */ e(S, { size: E[a], children: s })
  ] });
});
L.displayName = "Checkbox";
export {
  L as Checkbox
};
