// @sberbusiness/triplex-next/components/LightBox/LightBoxViewManager/LightBoxViewManagerConsts.js
import { getScrollbarWidth as r } from "../../../utils/scroll/scrollbar.js";
const o = 64, i = 16, t = 16, e = o + i + t, a = 1024, l = {
  /**
   * ClassNames для обозначения ширины контейнера.
   * Используются для замены media query в Css, т.к. media query учитывает только ширину экрана, а не ширину элемента.
   */
  breakPointsClassNames: {
    // Ширина меньше или равна lightBoxMediaPoint0.
    "less-or-equal-media-point-0": "global-LB-less-or-equal-media-point-0",
    // Ширина больше lightBoxMediaPoint0.
    // У этого класса нет стилей, поэтому тут произвольный класс.
    "more-media-point-0": "global-LB-more-media-point-0"
  },
  // Расстояние от стрелки до лайтбокса.
  lightBoxArrowMarginToLightBox: i,
  // Расстояние от стрелки до края экрана.
  lightBoxArrowMarginToScreen: t,
  // Ширина стрелки лайтбокса.
  lightBoxArrowWidth: o,
  //  Ширина стрелки лайтбокса с полями по бокам.
  lightBoxArrowWithMarginWidth: e,
  // Breakpoint для экранов менее 1024px.
  lightBoxMediaPoint0: a + r()
};
export {
  l as LightBoxViewManagerConsts
};
