// @sberbusiness/triplex-next/components/LightBox/LightBoxViewManager/LightBoxViewManager.js
import { jsx as r, jsxs as N } from "react/jsx-runtime";
import { useState as m, useRef as L, useCallback as n, useLayoutEffect as c } from "react";
import { Portal as B } from "../../Portal/Portal.js";
import { LightBoxViewManagerConsts as l } from "./LightBoxViewManagerConsts.js";
import { u as k, i as P, k as p } from "../../../chunks/vendor-Ba_pyhza.js";
const g = "LightBoxMountNodeViewManager", $ = ({
  lightBoxMountNode: a,
  lightBoxViewManagerNode: x
}) => {
  const [t, h] = m(), [s, R] = m(""), o = L(null), d = n(
    (e) => {
      if (!e)
        return;
      let i = [];
      e.width <= l.lightBoxMediaPoint0 ? i.push(l.breakPointsClassNames["less-or-equal-media-point-0"]) : i.push(l.breakPointsClassNames["more-media-point-0"]), i = i.sort().join(" "), s !== i && R(i);
    },
    [s]
  ), w = () => {
    if (o.current) {
      const e = o.current.getBoundingClientRect();
      P(
        p(t, ["top", "left", "width", "height"]),
        p(e, ["top", "left", "width", "height"])
      ) || h(e);
    }
  }, f = n(() => {
    if (o.current) {
      const e = o.current.getBoundingClientRect();
      h(e), d(e);
    }
  }, [d]), u = n(() => {
    s && (Array.from(a.classList).forEach((e) => {
      e.includes("LB-") && a.classList.toggle(e);
    }), s.split(" ").forEach((e) => a.classList.add(e)));
  }, [s, a.classList]);
  c(() => {
    f(), a.classList.add(g);
  }, []), c(() => {
    u();
  }, [s, u]), c(() => {
    w();
  });
  const { ref: C } = k({
    handleWidth: !0,
    onResize: f,
    refreshMode: "throttle",
    refreshRate: 150
  });
  return /* @__PURE__ */ r(B, { container: x, children: /* @__PURE__ */ N("div", { ref: o, style: { height: "100%" }, children: [
    /* @__PURE__ */ r("div", { ref: C, style: { width: "100%", height: "100%", position: "absolute" } }),
    t && /* @__PURE__ */ r("style", { children: `
                            .${g} {
                                --lightBox-screen-left: ${t.x >= 0 ? t.x : 0}px;
                                --lightBox-screen-width: ${t.width}px;
                                --lightBox-screen-top: ${t.height}px;
                            }
                        ` })
  ] }) });
};
export {
  $ as LightBoxViewManager
};
