// @sberbusiness/triplex-next/components/Dropdown/mobile/DropdownMobileInner.js
import { jsxs as j, jsx as c } from "react/jsx-runtime";
import y, { useState as l, useEffect as d } from "react";
import { c as n } from "../../../chunks/vendor-Ba_pyhza.js";
import { useToken as B } from "../../ThemeProvider/useToken.js";
import { s as e } from "../../../chunks/DropdownMobile.module-Fa_t6vz1.js";
const C = y.forwardRef(
  ({
    children: f,
    className: m,
    opened: o,
    opening: s,
    closing: t,
    setOpened: p,
    setClosing: u,
    setOpening: w,
    ...N
  }, T) => {
    const [a, M] = l(!1), [r, S] = l(!1), { scopeClassName: b } = B();
    d(() => {
      setTimeout(() => M(s));
    }, [s]), d(() => {
      setTimeout(() => S(o));
    }, [o]);
    const D = n(e.dropdownMobileWrapper, b, m), x = n(e.dropdownMobileBackdrop, {
      [e.closing]: t,
      [e.opened]: r,
      [e.opening]: a
    }), g = n(e.dropdownMobile, {
      [e.closing]: t,
      [e.opened]: r,
      [e.opening]: a
    }), h = (k) => {
      const { target: v, currentTarget: E } = k;
      v === E && (t ? u(!1) : s && w(!1));
    }, i = () => {
      o && p(!1);
    };
    return !s && !o && !t ? null : /* @__PURE__ */ j("div", { className: D, children: [
      /* @__PURE__ */ c(
        "div",
        {
          className: x,
          onTransitionEnd: h,
          onTouchStart: i,
          onMouseDown: i
        }
      ),
      /* @__PURE__ */ c("div", { className: g, ref: T, ...N, children: f })
    ] });
  }
);
C.displayName = "DropdownMobileInner";
export {
  C as DropdownMobileInner
};
