// @sberbusiness/triplex-next/components/Col/Col.js
import { jsx as v } from "react/jsx-runtime";
import { c as j } from "../../chunks/vendor-Ba_pyhza.js";
const A = {
  "col-1": "col-1__840b11d6",
  "col-2": "col-2__42e76c4d",
  "col-3": "col-3__eac316e0",
  "col-4": "col-4__26e16a56",
  "col-5": "col-5__c48916d4",
  "col-6": "col-6__1da8fe88",
  "col-7": "col-7__edcd84c1",
  "col-8": "col-8__4ae9eb7f",
  "col-9": "col-9__272db6c3",
  "col-10": "col-10__83f092b2",
  "col-11": "col-11__9a35eefc",
  "col-12": "col-12__ae5f3af2",
  "offset-0": "offset-0__f18eee6f",
  "offset-1": "offset-1__e227c950",
  "offset-2": "offset-2__c2d8d0fb",
  "offset-3": "offset-3__3b7d4db6",
  "offset-4": "offset-4__9cfb72da",
  "offset-5": "offset-5__0ab9708d",
  "offset-6": "offset-6__e8c26d23",
  "offset-7": "offset-7__f03487fe",
  "offset-8": "offset-8__13cf2bcf",
  "offset-9": "offset-9__58589603",
  "offset-10": "offset-10__dc26af03",
  "offset-11": "offset-11__e4f68e39",
  "d-none": "d-none__8c7ab4d8",
  "d-block": "d-block__236eb3b8",
  "col-sm-1": "col-sm-1__234f970d",
  "col-sm-2": "col-sm-2__3d505435",
  "col-sm-3": "col-sm-3__24d19057",
  "col-sm-4": "col-sm-4__e2bf4747",
  "col-sm-5": "col-sm-5__8acbf0e2",
  "col-sm-6": "col-sm-6__de9b5a0c",
  "col-sm-7": "col-sm-7__a013a8c9",
  "col-sm-8": "col-sm-8__e2af247f",
  "col-sm-9": "col-sm-9__34dd8382",
  "col-sm-10": "col-sm-10__d196bb59",
  "col-sm-11": "col-sm-11__5e6bdb87",
  "col-sm-12": "col-sm-12__db92bd9a",
  "offset-sm-0": "offset-sm-0__54745404",
  "offset-sm-1": "offset-sm-1__19064341",
  "offset-sm-2": "offset-sm-2__c2984acd",
  "offset-sm-3": "offset-sm-3__cf775219",
  "offset-sm-4": "offset-sm-4__c5a71e09",
  "offset-sm-5": "offset-sm-5__d195b750",
  "offset-sm-6": "offset-sm-6__2f5fa19c",
  "offset-sm-7": "offset-sm-7__f560a254",
  "offset-sm-8": "offset-sm-8__6f0de97d",
  "offset-sm-9": "offset-sm-9__d3150c81",
  "offset-sm-10": "offset-sm-10__88644433",
  "offset-sm-11": "offset-sm-11__e58467db",
  "d-none-sm": "d-none-sm__bb6fe3f6",
  "d-block-sm": "d-block-sm__82e24c18",
  "col-md-1": "col-md-1__862cac00",
  "col-md-2": "col-md-2__a47f557f",
  "col-md-3": "col-md-3__7c9bf7c4",
  "col-md-4": "col-md-4__25308f95",
  "col-md-5": "col-md-5__aff747a5",
  "col-md-6": "col-md-6__411c2684",
  "col-md-7": "col-md-7__56a84ba0",
  "col-md-8": "col-md-8__30ea878d",
  "col-md-9": "col-md-9__387718fb",
  "col-md-10": "col-md-10__75b4f9db",
  "col-md-11": "col-md-11__fcd8c9b8",
  "col-md-12": "col-md-12__0bbc913c",
  "offset-md-0": "offset-md-0__6af5106b",
  "offset-md-1": "offset-md-1__519600da",
  "offset-md-2": "offset-md-2__566c30e8",
  "offset-md-3": "offset-md-3__0112f1e8",
  "offset-md-4": "offset-md-4__dfd7a2cb",
  "offset-md-5": "offset-md-5__6d47de32",
  "offset-md-6": "offset-md-6__8326b109",
  "offset-md-7": "offset-md-7__95e57c01",
  "offset-md-8": "offset-md-8__a45ae394",
  "offset-md-9": "offset-md-9__127bd226",
  "offset-md-10": "offset-md-10__06ad9a57",
  "offset-md-11": "offset-md-11__297dffee",
  "d-none-md": "d-none-md__050e7ece",
  "d-block-md": "d-block-md__15a87a3e",
  "col-lg-1": "col-lg-1__d02d024f",
  "col-lg-2": "col-lg-2__f10fd9ab",
  "col-lg-3": "col-lg-3__761509fe",
  "col-lg-4": "col-lg-4__3232976f",
  "col-lg-5": "col-lg-5__f207f7dc",
  "col-lg-6": "col-lg-6__7ae0e83a",
  "col-lg-7": "col-lg-7__e0971854",
  "col-lg-8": "col-lg-8__48f75a7c",
  "col-lg-9": "col-lg-9__3a32f8a8",
  "col-lg-10": "col-lg-10__9e5e23ed",
  "col-lg-11": "col-lg-11__b4d36669",
  "col-lg-12": "col-lg-12__03a406bf",
  "offset-lg-0": "offset-lg-0__9f776d06",
  "offset-lg-1": "offset-lg-1__ccc71661",
  "offset-lg-2": "offset-lg-2__3a07902c",
  "offset-lg-3": "offset-lg-3__4d96e97a",
  "offset-lg-4": "offset-lg-4__b493d9ca",
  "offset-lg-5": "offset-lg-5__9b09c70f",
  "offset-lg-6": "offset-lg-6__8361aac7",
  "offset-lg-7": "offset-lg-7__57143664",
  "offset-lg-8": "offset-lg-8__6c459daf",
  "offset-lg-9": "offset-lg-9__16563d06",
  "offset-lg-10": "offset-lg-10__e9f8b9c8",
  "offset-lg-11": "offset-lg-11__9ccb1de6",
  "d-none-lg": "d-none-lg__6b6a1201",
  "d-block-lg": "d-block-lg__160a28d9",
  "col-xl-1": "col-xl-1__0961dd75",
  "col-xl-2": "col-xl-2__30c5db9d",
  "col-xl-3": "col-xl-3__cb4bd642",
  "col-xl-4": "col-xl-4__4404f9ab",
  "col-xl-5": "col-xl-5__e97df39d",
  "col-xl-6": "col-xl-6__93cff378",
  "col-xl-7": "col-xl-7__82a7792b",
  "col-xl-8": "col-xl-8__8cad0e01",
  "col-xl-9": "col-xl-9__8c2897a1",
  "col-xl-10": "col-xl-10__b726cc88",
  "col-xl-11": "col-xl-11__4cedb772",
  "col-xl-12": "col-xl-12__d9ab6ba6",
  "offset-xl-0": "offset-xl-0__6a11e48b",
  "offset-xl-1": "offset-xl-1__bba46f05",
  "offset-xl-2": "offset-xl-2__a4a0e9ae",
  "offset-xl-3": "offset-xl-3__af01612a",
  "offset-xl-4": "offset-xl-4__438fbad3",
  "offset-xl-5": "offset-xl-5__a6d47c4e",
  "offset-xl-6": "offset-xl-6__be8279ce",
  "offset-xl-7": "offset-xl-7__d795b8c2",
  "offset-xl-8": "offset-xl-8__a4a4dcd3",
  "offset-xl-9": "offset-xl-9__42885d54",
  "offset-xl-10": "offset-xl-10__063cd47a",
  "offset-xl-11": "offset-xl-11__96b6be76",
  "d-none-xl": "d-none-xl__78ac179c",
  "d-block-xl": "d-block-xl__73d2004d"
};
function _({
  block: d,
  hidden: t,
  offset: l,
  prefix: o,
  size: e
}) {
  const f = [], c = o ? `${o}-` : "", s = o ? `-${o}` : "";
  return d === !0 && f.push(`d-block${s}`), t === !0 && f.push(`d-none${s}`), l !== void 0 && f.push(`offset-${c}${l}`), e && f.push(`col-${c}${e}`), f;
}
const B = ({
  children: d,
  className: t,
  hidden: l,
  hiddenSm: o,
  hiddenMd: e,
  hiddenLg: f,
  hiddenXl: c,
  block: s,
  blockSm: m,
  blockMd: a,
  blockLg: b,
  blockXl: x,
  size: g = 12,
  sizeSm: n,
  sizeMd: i,
  sizeLg: r,
  sizeXl: k,
  offset: p,
  offsetSm: u,
  offsetMd: h,
  offsetLg: $,
  offsetXl: N,
  ...C
}) => {
  const y = [
    ..._({ block: s, hidden: l, offset: p, size: g }),
    ..._({ block: m, hidden: o, offset: u, prefix: "sm", size: n }),
    ..._({ block: a, hidden: e, offset: h, prefix: "md", size: i }),
    ..._({ block: b, hidden: f, offset: $, prefix: "lg", size: r }),
    ..._({ block: x, hidden: c, offset: N, prefix: "xl", size: k })
  ].map((z) => A[z]).filter(Boolean);
  return /* @__PURE__ */ v("div", { ...C, className: j(t, ...y), children: d });
};
B.displayName = "Col";
export {
  B as Col
};
