// @sberbusiness/triplex-next/components/Pagination/index.js
import { PaginationExtended as i } from "./components/PaginationExtended.js";
import { PaginationNavigation as n } from "./components/PaginationNavigation.js";
import { PaginationNavigationButton as r } from "./components/PaginationNavigationButton.js";
import { PaginationNavigationExtended as x } from "./components/PaginationNavigationExtended.js";
import { PaginationNavigationExtendedItem as m } from "./components/PaginationNavigationExtendedItem.js";
import { PaginationPageButton as f } from "./components/PaginationPageButton.js";
import { PaginationPageEllipsis as v } from "./components/PaginationPageEllipsis.js";
import { PaginationSelect as N } from "./components/PaginationSelect.js";
import { EPaginationNavigationIconDirection as l } from "./enums.js";
import { Pagination as u } from "./Pagination.js";
export {
  l as EPaginationNavigationIconDirection,
  u as Pagination,
  i as PaginationExtended,
  n as PaginationNavigation,
  r as PaginationNavigationButton,
  x as PaginationNavigationExtended,
  m as PaginationNavigationExtendedItem,
  f as PaginationPageButton,
  v as PaginationPageEllipsis,
  N as PaginationSelect
};
