// @sberbusiness/triplex-next/components/Chip/ChipSuggest/ChipSuggestDropdown.js
import { jsxs as E, jsx as r } from "react/jsx-runtime";
import S, { useCallback as h } from "react";
import { EVENT_KEY_CODES as C } from "../../../utils/keyboard.js";
import { useSuggestContext as L } from "../../Suggest/SuggestContext.js";
import { Dropdown as O } from "../../Dropdown/Dropdown.js";
import "../../Dropdown/DropdownListContext.js";
import "../../Dropdown/desktop/DropdownDesktop.js";
import { DropdownList as _ } from "../../Dropdown/desktop/DropdownList.js";
import { D as x } from "../../../chunks/DropdownListItem-D2hzyBGc.js";
import "../../Dropdown/mobile/DropdownMobile.js";
import "../../Dropdown/mobile/DropdownMobileBody.js";
import "../../Dropdown/mobile/DropdownMobileClose.js";
import "../../Dropdown/mobile/DropdownMobileFooter.js";
import "../../Dropdown/mobile/DropdownMobileHeader.js";
import "../../Dropdown/mobile/DropdownMobileInner.js";
import "../../Dropdown/mobile/DropdownMobileInput.js";
import "../../Dropdown/mobile/DropdownMobileList.js";
import "../../Dropdown/mobile/DropdownMobileListItem.js";
import "../../Loader/LoaderSmall/LoaderSmall.js";
import "../../Loader/LoaderMiddle/LoaderMiddle.js";
import { ChipSuggestDesktopDropdownField as R } from "./desktop/ChipSuggestDesktopDropdownField.js";
import { SuggestMobileDropdownContent as b } from "../../Suggest/SuggestMobileDropdownContent.js";
const y = [C.ENTER], k = ({ children: s, size: p, targetRef: l, ...c }, e) => {
  const {
    value: t,
    options: i,
    dropdownListId: f,
    dropdownOpen: d,
    dropdownListLoading: w,
    dropdownRef: D,
    closeDropdown: n,
    onSelect: g,
    setDropdownOpen: m
  } = L(), u = (o) => {
    D.current = o, typeof e == "function" ? e(o) : e && (e.current = o);
  }, a = h(
    (o) => {
      o ? m(!0) : n();
    },
    [m, n]
  );
  return /* @__PURE__ */ E(
    O,
    {
      size: p,
      targetRef: l,
      opened: d,
      fixedWidth: !1,
      setOpened: a,
      mobileViewProps: {
        children: /* @__PURE__ */ r(b, {})
      },
      ...c,
      ref: u,
      children: [
        /* @__PURE__ */ r(R, { children: s }),
        /* @__PURE__ */ r(
          _,
          {
            id: f,
            size: p,
            dropdownOpened: d,
            loading: w,
            style: i.length === 0 ? { display: "none" } : void 0,
            children: i.map((o) => /* @__PURE__ */ r(
              x,
              {
                id: o.id,
                keyCodesForSelection: y,
                selected: o.id === (t == null ? void 0 : t.id),
                onSelect: () => g(o),
                children: o.content || o.label
              },
              o.id
            ))
          }
        )
      ]
    }
  );
}, $ = S.forwardRef(k);
export {
  $ as ChipSuggestDropdown
};
