// @sberbusiness/triplex-next/components/Chip/ChipSuggest/ChipSuggestTarget.js
import { jsx as f } from "react/jsx-runtime";
import x, { useCallback as a } from "react";
import { useSuggestContext as T } from "../../Suggest/SuggestContext.js";
import "../../Suggest/Suggest.js";
import { C as B } from "../../../chunks/Chip-G6cGXuGj.js";
import { ChipClearButton as E } from "../ChipClearButton.js";
import { ChipDropdownArrow as P } from "../ChipDropdownArrow.js";
import { isKey as e } from "../../../utils/keyboard.js";
const R = ({ onKeyDown: t, onClick: p, clearSelected: s, ...n }, u) => {
  const { value: m, dropdownOpen: o, setDropdownOpen: i } = T(), g = a(
    (r) => {
      i(!o), p == null || p(r);
    },
    [o, i, p]
  ), C = a(
    (r) => {
      (e(r.code, "ENTER") || e(r.code, "SPACE")) && (i(!o), t == null || t(r));
    },
    [o, i, t]
  ), d = a(
    (r) => {
      r.stopPropagation(), s == null || s();
    },
    [s]
  ), h = a(() => m === void 0 ? /* @__PURE__ */ f(P, { rotated: o }) : /* @__PURE__ */ f(E, { onClick: d }), [m, o, d]);
  return /* @__PURE__ */ f(
    B,
    {
      selected: m !== void 0,
      "aria-expanded": o,
      postfix: h(),
      onKeyDown: C,
      onClick: g,
      ...n,
      ref: u
    }
  );
}, N = x.forwardRef(R);
export {
  N as ChipSuggestTarget
};
