// @sberbusiness/triplex-next/components/Chip/ChipSuggest/desktop/ChipSuggestDesktopDropdownField.js
import { jsxs as i, jsx as t } from "react/jsx-runtime";
import { useState as z, useRef as T, useCallback as o } from "react";
import { useSuggestContext as b } from "../../../Suggest/SuggestContext.js";
import { FormField as k } from "../../../FormField/FormField.js";
import "../../../FormField/components/FormFieldDescription.js";
import "../../../FormField/components/FormFieldCounter.js";
import { FormFieldLabel as B } from "../../../FormField/components/FormFieldLabel.js";
import { FormFieldClear as D } from "../../../FormField/components/FormFieldClear.js";
import "../../../FormField/components/FormFieldTarget.js";
import { F as L } from "../../../../chunks/FormFieldInput-CO7yZP0Q.js";
import "../../../FormField/components/FormFieldMaskedInput.js";
import { FormFieldPostfix as M } from "../../../FormField/components/FormFieldPostfix.js";
import "../../../FormField/components/FormFieldPrefix.js";
import "../../../FormField/components/FormFieldTextarea.js";
import { EComponentSize as a } from "../../../../enums/EComponentSize.js";
import { Tooltip as n } from "../../../Tooltip/Tooltip.js";
import { ETooltipSize as O } from "../../../Tooltip/enums.js";
import { LoaderSmall as R } from "../../../Loader/LoaderSmall/LoaderSmall.js";
import { ELoaderSmallTheme as W } from "../../../Loader/LoaderSmall/enum.js";
import "../../../Loader/LoaderMiddle/LoaderMiddle.js";
const E = "chipSuggestInputWrapper__7c7828be", j = {
  chipSuggestInputWrapper: E
}, oe = ({ children: m }) => {
  const {
    inputValue: r,
    placeholder: u,
    noOptionsText: c,
    dropdownListId: d,
    activeDescendant: g,
    loading: f,
    tooltipOpen: h,
    clearInputOnFocus: p,
    onFilter: e
  } = b(), [F, l] = z(!1), s = T(null), S = o(() => e(""), [e]), I = o(
    (v) => {
      e(v.target.value);
    },
    [e]
  ), C = o(() => {
    l(!0), r.length !== 0 && p === !0 && e("");
  }, [r.length, p, e]), x = o(() => l(!1), []);
  return /* @__PURE__ */ i(k, { className: j.chipSuggestInputWrapper, size: a.SM, children: [
    /* @__PURE__ */ t(B, { children: m }),
    /* @__PURE__ */ i(
      n,
      {
        size: O.SM,
        isOpen: !!h && F,
        toggle: () => {
        },
        targetRef: s,
        disableAdaptiveMode: !0,
        children: [
          /* @__PURE__ */ t(n.Body, { children: c }),
          /* @__PURE__ */ t(n.Target, { children: /* @__PURE__ */ t(
            L,
            {
              value: r,
              placeholder: u,
              "aria-controls": d,
              "aria-activedescendant": g,
              autoFocus: !0,
              onFocus: C,
              onBlur: x,
              onChange: I,
              ref: s
            }
          ) })
        ]
      }
    ),
    /* @__PURE__ */ i(M, { children: [
      /* @__PURE__ */ t(D, { onClick: S }),
      f && /* @__PURE__ */ t(R, { theme: W.BRAND, size: a.SM })
    ] })
  ] });
};
export {
  oe as ChipSuggestDesktopDropdownField
};
