// @sberbusiness/triplex-next/components/Page/components/FooterPage.js
import { jsx as e } from "react/jsx-runtime";
import F, { useRef as R } from "react";
import { c as T } from "../../../chunks/vendor-Ba_pyhza.js";
import { Footer as s } from "../../Footer/Footer.js";
import { EFooterPageType as i } from "./enums.js";
import { useStickyCornerRadius as d } from "./useStickyCornerRadius.js";
import { Island as y } from "../../Island/Island.js";
import { EIslandType as g } from "../../Island/enums.js";
import "../../Island/components/IslandBody.js";
import "../../Island/components/IslandHeader.js";
import "../../Island/components/IslandFooter.js";
import { s as c } from "../../../chunks/Page.module-DjGbJ-Uw.js";
const w = Object.assign(
  F.forwardRef(
    ({ className: m, type: r, size: n, sticky: f, ...p }, o) => {
      const a = R(null);
      d(a, "bottom", r === i.FIRST && f);
      const l = (t) => {
        a.current = t, typeof o == "function" ? o(t) : o && (o.current = t);
      }, u = T(m, c.footerPageTypeFirst, {
        [c.sticky]: r === i.FIRST && f
      });
      return r === i.FIRST ? /* @__PURE__ */ e(
        y,
        {
          className: u,
          type: g.TYPE_1,
          ref: l,
          size: n,
          children: /* @__PURE__ */ e(s, { ...p })
        }
      ) : /* @__PURE__ */ e(s, { ref: o, className: m, ...p });
    }
  ),
  {
    Description: s.Description
  }
);
export {
  w as FooterPage
};
