// @sberbusiness/triplex-next/components/Stepper/Stepper.js
import { jsx as s } from "react/jsx-runtime";
import { useState as D, useRef as N, useEffect as v } from "react";
import { StepperExtended as C } from "./StepperExtended.js";
import { StepperWrapper as G } from "./StepperWrapper.js";
import { StepperStep as H } from "./StepperStep.js";
import { CarouselExtended as X } from "../CarouselExtended/CarouselExtended.js";
import { ButtonIcon as _ } from "../Button/ButtonIcon.js";
import { CaretrightStrokeSrvIcon24 as $, CaretleftStrokeSrvIcon24 as q } from "@sberbusiness/icons-next";
import { EScreenWidth as F } from "../../helpers/breakpoints.js";
import { scrollSmoothHorizontally as g } from "../../utils/scroll.js";
import { c as i } from "../../chunks/vendor-Ba_pyhza.js";
import { EStepperSize as l } from "./enums.js";
const J = "stepper__c10ca561", K = "stepperCarousel__0ecc2c7b", O = "buttonPrev__12d12c03", Q = "buttonNext__baace22d", T = "sm__38a06386", U = "md__bfbd0a1b", V = "lg__56d8fc57", n = {
  stepper: J,
  stepperCarousel: K,
  buttonPrev: O,
  buttonNext: Q,
  sm: T,
  md: U,
  lg: V
}, R = (p) => {
  switch (p) {
    case l.LG:
      return n.lg;
    case l.MD:
      return n.md;
    case l.SM:
      return n.sm;
  }
}, P = ({
  className: p,
  steps: S,
  size: a = l.SM,
  selectedStepId: c,
  ...E
}) => {
  const [h, w] = D({ stepNext: 0, stepPrev: 0 }), m = N(null), b = N({}), B = i(n.stepper, p), M = S.findIndex((e) => e.id === c), I = ({ hidden: e, ...t }) => e ? null : /* @__PURE__ */ s(
    _,
    {
      className: i(n.buttonPrev, R(a)),
      ...t,
      "aria-label": "Прокрутить назад",
      children: /* @__PURE__ */ s(q, { className: i(n.buttonPrevIcon), paletteIndex: 5 })
    }
  ), W = ({ hidden: e, ...t }) => e ? null : /* @__PURE__ */ s(
    _,
    {
      className: i(n.buttonNext, R(a)),
      ...t,
      "aria-label": "Прокрутить вперёд",
      children: /* @__PURE__ */ s($, { className: i(n.buttonNextIcon), paletteIndex: 5 })
    }
  );
  v(() => {
    const { current: e } = m;
    if (e) {
      const t = e.getBoundingClientRect().width * 0.3;
      w({ stepNext: t, stepPrev: t });
    }
  }, []);
  const L = (e, t, r, o) => {
    r > 0 && (t.nextElementSibling && (r -= o - t.nextElementSibling.getBoundingClientRect().right), g(e, Math.ceil(r)));
  }, k = (e, t, r, o) => {
    r < 0 && (t.previousElementSibling && (r -= o - t.previousElementSibling.getBoundingClientRect().left), g(e, Math.floor(r)));
  }, y = (e, t) => {
    t && g(e, t);
  }, A = (e, t) => {
    const { left: r, right: o, width: j } = e.getBoundingClientRect(), { left: u, right: x, width: z } = t.getBoundingClientRect(), f = r + j / 2, d = u + z / 2;
    window.matchMedia(`(max-width: ${F.SM_MAX})`).matches ? y(e, d - f) : f > d ? k(e, t, u - r - 32, u) : f < d && L(e, t, x - o + 32, x);
  };
  return v(() => {
    if (c) {
      const { current: e } = m, t = b.current[c];
      e && t && A(e, t);
    }
  }, [c]), /* @__PURE__ */ s(
    X,
    {
      className: i(n.stepperCarousel),
      buttonPrev: I,
      buttonNext: W,
      stepPrev: h.stepPrev,
      stepNext: h.stepNext,
      ref: m,
      children: /* @__PURE__ */ s(C, { className: B, selectedStepId: c, size: a, ...E, children: S.map(({ label: e, ...t }, r) => /* @__PURE__ */ s(
        C.Step,
        {
          forwardedRef: (o) => b.current[t.id] = o,
          ...t,
          isInActiveStep: r > M,
          children: e
        },
        t.id
      )) })
    }
  );
};
P.Wrapper = G;
P.Step = H;
export {
  P as Stepper
};
