// @sberbusiness/triplex-next/components/Stepper/StepperStep.js
import { jsxs as I, jsx as y } from "react/jsx-runtime";
import { useContext as V, useState as W, useRef as k } from "react";
import { StepperStepIcon as q } from "./StepperStepIcon.js";
import { isKey as A } from "../../utils/keyboard.js";
import { StepperExtendedContext as D } from "./StepperExtendedContext.js";
import { EFocusSource as s } from "../../enums/EFocusSource.js";
import { c as C } from "../../chunks/vendor-Ba_pyhza.js";
import { s as t, R as F } from "../../chunks/RightBorderArrow-BHW7rkmN.js";
import { EStepperSize as h, EStepperStepType as j } from "./enums.js";
const H = ({
  children: x,
  className: z,
  id: c,
  isInActiveStep: R,
  onFocus: m,
  onBlur: a,
  onKeyDown: i,
  onMouseDown: f,
  onClick: l,
  disabled: r,
  icon: g,
  forwardedRef: o,
  type: O,
  ...G
}) => {
  const { selectedId: T, size: n, onSelectStep: S } = V(D), [u, E] = W(s.NONE), p = k(null), N = c === T, Y = C(
    t.stepperStep,
    {
      [t.active]: N,
      [t.inactive]: R,
      [t.completed]: !N && !R,
      [t.disabled]: !!r,
      [t.nonempty]: !!x,
      [t.error]: !r && O === j.ERROR,
      [t.warning]: !r && O === j.WARNING,
      [t.focusVisible]: u === s.KEYBOARD,
      [t.sm]: n === h.SM,
      [t.md]: n === h.MD,
      [t.lg]: n === h.LG
    },
    z
  ), v = (e) => {
    !u && p.current === e.target && E(s.KEYBOARD), m == null || m(e);
  }, B = (e) => {
    p.current !== document.activeElement && p.current === e.target && E(s.NONE), a == null || a(e);
  }, K = (e) => {
    const d = e.code || e.keyCode;
    A(d, "ENTER") ? r || S(c) : A(d, "SPACE") && (e.preventDefault(), r || S(c)), i == null || i(e);
  }, L = (e) => {
    u || E(s.MOUSE), f == null || f(e);
  }, M = (e) => {
    r || S(c), l == null || l(e);
  }, P = (e) => {
    p.current = e, typeof o == "function" ? o(e) : o && (o.current = e);
  }, U = () => {
    const e = C(t.icon, "hoverable", { disabled: !!r });
    return /* @__PURE__ */ y("span", { className: e, children: g });
  };
  return /* @__PURE__ */ I(
    "li",
    {
      className: Y,
      onFocus: v,
      onBlur: B,
      onKeyDown: K,
      onMouseDown: L,
      onClick: M,
      tabIndex: r ? -1 : 0,
      "aria-disabled": r,
      "aria-current": N || void 0,
      role: "button",
      ...G,
      ref: P,
      children: [
        /* @__PURE__ */ I("div", { className: t.content, "data-tx": "1.4.0", children: [
          g && U(),
          x
        ] }),
        /* @__PURE__ */ y(F, { size: n })
      ]
    }
  );
};
H.Icon = q;
export {
  H as StepperStep
};
