// @sberbusiness/triplex-next/components/ExpandAnimation/ExpandAnimation.js
import { jsx as g } from "react/jsx-runtime";
import N, { useRef as y } from "react";
import { y as U, c as j } from "../../chunks/vendor-Ba_pyhza.js";
const F = "expandAnimation__e4ff3b4b", I = {
  expandAnimation: F
}, L = 300, M = N.forwardRef(
  ({
    children: A,
    className: E,
    expanded: v,
    animationTime: x = L,
    style: R,
    onStart: d,
    onEnd: h,
    transitionProps: e,
    ..._
  }, u) => {
    const m = y({
      entering: { height: void 0, overflow: "hidden" },
      entered: {},
      exiting: { height: 0, overflow: "hidden" },
      exited: { height: 0, overflow: "hidden", visibility: "hidden" },
      unmounted: {}
    }), l = y(null), w = (c) => {
      var f;
      l.current && (m.current.entering.height = l.current.scrollHeight), d == null || d(), (f = e == null ? void 0 : e.onEnter) == null || f.call(e, c);
    }, b = (c) => {
      var f;
      h == null || h(), (f = e == null ? void 0 : e.onEntered) == null || f.call(e, c);
    }, H = () => {
      var c;
      l.current && (l.current.style.height = l.current.scrollHeight + "px", l.current.scrollHeight), d == null || d(), (c = e == null ? void 0 : e.onExit) == null || c.call(e);
    }, T = () => {
      var c;
      h == null || h(), (c = e == null ? void 0 : e.onExited) == null || c.call(e);
    }, D = (c) => {
      l.current = c, typeof u == "function" ? u(c) : u && (u.current = c);
    };
    return /* @__PURE__ */ g(
      U,
      {
        in: v,
        timeout: x,
        nodeRef: l,
        ...e,
        onEnter: w,
        onEntered: b,
        onExit: H,
        onExited: T,
        children: (c) => /* @__PURE__ */ g(
          "div",
          {
            className: j(I.expandAnimation, E),
            style: {
              transitionDuration: x + "ms",
              ...m.current[c],
              ...R
            },
            ..._,
            ref: D,
            children: A
          }
        )
      }
    );
  }
);
M.displayName = "ExpandAnimation";
export {
  M as ExpandAnimation
};
