// @sberbusiness/triplex-next/components/HelpBox/HelpBox.js
import { jsxs as h, Fragment as E, jsx as t } from "react/jsx-runtime";
import { useRef as M, useState as x, useId as R } from "react";
import { c as S, z as _ } from "../../chunks/vendor-Ba_pyhza.js";
import { HintFilledSrvIcon16 as L } from "@sberbusiness/icons-next";
import { ButtonIcon as j } from "../Button/ButtonIcon.js";
import { EButtonIconShape as k } from "../Button/enums.js";
import { Tooltip as l } from "../Tooltip/Tooltip.js";
import { TooltipMobileHeader as z } from "../Tooltip/components/mobile/components/TooltipMobileHeader.js";
import { MobileView as C } from "../MobileView/MobileView.js";
import { getDataHTMLAttributes as D } from "../../utils/html/DataAttributes.js";
import { getAriaHTMLAttributes as V } from "../../utils/html/AriaAttributes.js";
const X = "helpBoxButton__3d925752", $ = "helpBoxTooltipBody__27844b32", f = {
  helpBoxButton: X,
  helpBoxTooltipBody: $
}, q = ({
  children: T,
  className: y,
  focusTrapProps: e,
  mobileHeaderContent: s,
  isOpen: i,
  onShow: n,
  tooltipSize: I,
  preferPlace: b,
  toggle: r,
  tooltipAriaAttributes: c,
  tooltipDataAttributes: p,
  ...v
}) => {
  const d = M(null), [F, O] = x(!!i), [m, B] = x(null), u = R(), a = i ?? F, H = (o) => {
    i === void 0 && O(o), o || B(null), r == null || r(o);
  }, N = (o) => {
    B(o), n == null || n(o);
  }, g = (o) => /* @__PURE__ */ t(
    C,
    {
      fallback: /* @__PURE__ */ t(
        _.FocusTrap,
        {
          active: a,
          ...e,
          focusTrapOptions: {
            clickOutsideDeactivates: !0,
            initialFocus: `[id='${u}']`,
            preventScroll: !0,
            ...e == null ? void 0 : e.focusTrapOptions
          },
          containerElements: [o]
        }
      ),
      children: null
    }
  );
  return /* @__PURE__ */ h(E, { children: [
    /* @__PURE__ */ h(
      l,
      {
        id: u,
        tabIndex: -1,
        role: "dialog",
        toggleType: "hover",
        size: I,
        preferPlace: b,
        isOpen: a,
        toggle: H,
        onShow: N,
        targetRef: d,
        ...!!c && V(c),
        ...!!p && D(p),
        children: [
          /* @__PURE__ */ t(l.Target, { children: /* @__PURE__ */ t(
            j,
            {
              className: S(f.helpBoxButton, y),
              "aria-label": "Подсказка",
              shape: k.CIRCLE,
              ref: d,
              ...v,
              children: /* @__PURE__ */ t(L, { paletteIndex: 5 })
            }
          ) }),
          s && /* @__PURE__ */ t(z, { children: s }),
          /* @__PURE__ */ t(l.Body, { className: f.helpBoxTooltipBody, children: T }),
          /* @__PURE__ */ t(l.XButton, { "aria-label": "Закрыть" })
        ]
      }
    ),
    a && m && g(m)
  ] });
};
q.displayName = "HelpBox";
export {
  q as HelpBox
};
