// @sberbusiness/triplex-next/components/StatusTracker/StatusTracker.js
import { jsxs as n, jsx as s } from "react/jsx-runtime";
import p from "react";
import { c as a } from "../../chunks/vendor-Ba_pyhza.js";
import { s as r } from "../../chunks/StatusTracker.module-zTyojoH4.js";
import { EStatusTrackerType as t, EVerticalAlign as o } from "./enums.js";
import { StatusTrackerMedia as u } from "./components/StatusTrackerMedia.js";
import { StatusTrackerHeader as f } from "./components/StatusTrackerHeader.js";
import { StatusTrackerBody as k } from "./components/StatusTrackerBody.js";
import { StatusTrackerFooter as N } from "./components/StatusTrackerFooter.js";
const c = {
  [t.REJECTED]: r.rejected,
  [t.WAITING]: r.waiting,
  [t.WARNING]: r.warning,
  [t.APPROVED]: r.approved
}, g = {
  [o.MIDDLE]: r.verticalAlignMiddle,
  [o.BOTTOM]: r.verticalAlignBottom
}, v = Object.assign(
  p.forwardRef(function({ children: i, className: m, type: e, verticalAlign: l = o.TOP, ...d }, T) {
    return /* @__PURE__ */ n(
      "div",
      {
        className: a(r.statusTrackerWrapper, m),
        ...d,
        ref: T,
        "data-tx": "1.4.0",
        children: [
          /* @__PURE__ */ s("div", { className: a(r.statusTrackerBackground, c[e]), children: /* @__PURE__ */ s("div", { className: a(r.statusTrackerColor, c[e]) }) }, e),
          /* @__PURE__ */ s("div", { className: a(r.statusTracker, g[l]), children: i })
        ]
      }
    );
  }),
  {
    Media: u,
    Header: f,
    Body: k,
    Footer: N
  }
);
v.displayName = "StatusTracker";
export {
  v as StatusTracker
};
