// @sberbusiness/triplex-next/components/ThemeProvider/ThemeProvider.js
import { jsx as l } from "react/jsx-runtime";
import { useState as c, useEffect as i } from "react";
import { q as x, s as d, w as f } from "../../chunks/vendor-Ba_pyhza.js";
import { DesignTokenUtils as S } from "../DesignTokens/DesignTokenUtils.js";
import { ThemeProviderView as h } from "./components/ThemeProviderView.js";
import { ETriplexNextTheme as u } from "./ETriplexNextTheme.js";
const m = (t) => `${t}${Date.now()}${Math.floor(Math.random() * 1e6)}`, g = ({
  children: t,
  scopeClassName: o,
  scopeRef: n,
  theme: r = u.LIGHT,
  tokens: s
}) => {
  const [e, a] = c(o || m("triplex-next-theme-"));
  return i(() => {
    a(o || m("triplex-next-theme-"));
  }, [o]), i(() => {
    if (x()) {
      const p = `.${e} {${S.getStyle(r, s || {})}}`;
      d(p, `triplex-next-dynamic-tokens-${e}`);
    } else
      console.log("ThemeProvider", "SSR do not support dynamic theme with css variables.");
  }, [e, r, s]), i(() => () => f(`triplex-next-dynamic-tokens-${e}`), [e]), /* @__PURE__ */ l(h, { scopeClassName: e, scopeRef: n, theme: r, tokens: s, children: t });
};
export {
  g as ThemeProvider
};
