// @sberbusiness/triplex-next/components/SuggestField/mobile/SuggestFieldMobileDropdown.js
import { jsx as t, jsxs as m, Fragment as x } from "react/jsx-runtime";
import y, { useState as H, useRef as p, useEffect as V } from "react";
import { Dropdown as j } from "../../Dropdown/Dropdown.js";
import "../../Dropdown/DropdownListContext.js";
import "../../Dropdown/desktop/DropdownDesktop.js";
import "../../Dropdown/desktop/DropdownList.js";
import "../../../chunks/DropdownListItem-D2hzyBGc.js";
import "../../Dropdown/mobile/DropdownMobile.js";
import { DropdownMobileBody as k } from "../../Dropdown/mobile/DropdownMobileBody.js";
import { DropdownMobileClose as _ } from "../../Dropdown/mobile/DropdownMobileClose.js";
import "../../Dropdown/mobile/DropdownMobileFooter.js";
import { DropdownMobileHeader as N } from "../../Dropdown/mobile/DropdownMobileHeader.js";
import "../../Dropdown/mobile/DropdownMobileInner.js";
import { DropdownMobileInput as O } from "../../Dropdown/mobile/DropdownMobileInput.js";
import { DropdownMobileList as P } from "../../Dropdown/mobile/DropdownMobileList.js";
import { DropdownMobileListItem as T } from "../../Dropdown/mobile/DropdownMobileListItem.js";
import { DropdownMobileLoader as q } from "../../Dropdown/mobile/DropdownMobileLoader.js";
import { s as z, S as A } from "../../../chunks/SuggestFieldMobileDropdownHint-CMKMF0PQ.js";
const E = ({
  value: o,
  options: c,
  placeholder: f,
  dropdownHint: l,
  opened: g,
  loading: h,
  dropdownListLoading: a,
  clearInputOnFocus: b,
  setOpened: n,
  onSelect: s,
  onFilter: w,
  onScrollEnd: d
}, D) => {
  const [M, e] = H((o == null ? void 0 : o.label) || ""), F = p(null), u = p(null);
  V(() => {
    e((o == null ? void 0 : o.label) || "");
  }, [o]);
  const C = () => {
    e(b === !1 && (o == null ? void 0 : o.label) || "");
  }, I = (r) => {
    const { value: i } = r.target;
    e(i), w(i);
  }, R = () => {
    n(!1), s(o);
  }, S = (r) => {
    if (!d)
      return;
    const { clientHeight: i, scrollHeight: B, scrollTop: L } = r.target;
    B - L === i && d();
  };
  return /* @__PURE__ */ t(
    j,
    {
      setOpened: (r) => {
        n(r), r || s(o);
      },
      opened: g,
      targetRef: u,
      ref: D,
      mobileViewProps: {
        children: /* @__PURE__ */ m(x, { children: [
          /* @__PURE__ */ m(N, { closeButton: () => /* @__PURE__ */ t(_, { onClick: R }), children: [
            /* @__PURE__ */ t(
              O,
              {
                value: M,
                placeholder: f,
                autoFocus: !0,
                onFocus: C,
                onChange: I
              }
            ),
            h && /* @__PURE__ */ t(q, {})
          ] }),
          /* @__PURE__ */ t(k, { className: z.suggestFieldMobileBody, onScroll: S, children: l ? /* @__PURE__ */ t(A, { children: l }) : /* @__PURE__ */ t(P, { loading: a, ref: F, children: c.map((r) => /* @__PURE__ */ t(
            T,
            {
              id: r.id,
              selected: r.id === (o == null ? void 0 : o.id),
              onSelect: () => {
                s(r), n(!1);
              },
              children: r.content || r.label
            },
            r.id
          )) }) })
        ] })
      }
    }
  );
}, mo = y.forwardRef(E);
export {
  mo as SuggestFieldMobileDropdown
};
