// @sberbusiness/triplex-next/components/SuggestField/mobile/SuggestFieldMobile.js
import { jsxs as B, jsx as c } from "react/jsx-runtime";
import { useState as E, useRef as p, useEffect as L } from "react";
import { c as N } from "../../../chunks/vendor-Ba_pyhza.js";
import { SuggestFieldMobileTarget as R } from "./SuggestFieldMobileTarget.js";
import { SuggestFieldMobileDropdown as q } from "./SuggestFieldMobileDropdown.js";
import { s as z } from "../../../chunks/SuggestFieldMobileDropdownHint-CMKMF0PQ.js";
function U({
  className: m,
  value: r,
  options: u,
  size: a,
  status: b,
  label: x,
  placeholder: t,
  dropdownHint: g,
  loading: n,
  dropdownListLoading: w,
  clearInputOnFocus: M,
  onFilter: O,
  onSelect: i,
  onScrollEnd: D,
  onTargetInputFocus: o,
  onTargetInputBlur: S,
  renderTarget: l,
  renderTargetInput: h,
  renderTargetLabel: j,
  renderTargetPrefix: v,
  renderTargetPostfix: C,
  ...F
}) {
  const [e, d] = E(!1), s = p(null), f = p(!1), P = l === void 0 ? R : l;
  L(() => {
    f.current && !e && s.current && s.current.scrollIntoView({ block: "center" }), f.current = e;
  }, [e]);
  const V = () => i(void 0), k = (y) => {
    d(!0), o == null || o(y);
  };
  return /* @__PURE__ */ B("div", { className: N(z.suggestFieldMobile, m), ...F, ref: s, children: [
    /* @__PURE__ */ c(
      P,
      {
        size: a,
        status: b,
        inputValue: r ? r.label : "",
        label: x,
        placeholder: t,
        loading: n,
        dropdownOpen: e,
        onClear: V,
        onFocus: k,
        onBlur: S,
        renderInput: h,
        renderLabel: j,
        renderPrefix: v,
        renderPostfix: C
      }
    ),
    /* @__PURE__ */ c(
      q,
      {
        value: r,
        options: u,
        placeholder: t,
        dropdownHint: g,
        opened: e,
        loading: n,
        dropdownListLoading: w,
        clearInputOnFocus: M,
        setOpened: d,
        onFilter: O,
        onSelect: i,
        onScrollEnd: D
      }
    )
  ] });
}
export {
  U as SuggestFieldMobile
};
