// @sberbusiness/triplex-next/components/TextField/TextFieldBase.js
import { jsxs as o, jsx as r } from "react/jsx-runtime";
import { FormField as t } from "../FormField/FormField.js";
import { FormFieldDescription as p } from "../FormField/components/FormFieldDescription.js";
import { FormFieldLabel as f } from "../FormField/components/FormFieldLabel.js";
import { FormFieldPostfix as h } from "../FormField/components/FormFieldPostfix.js";
import { FormFieldPrefix as s } from "../FormField/components/FormFieldPrefix.js";
import { FormFieldCounter as c } from "../FormField/components/FormFieldCounter.js";
import { FormGroup as u } from "../FormGroup/FormGroup.js";
const x = ({
  children: F,
  description: i,
  label: m,
  prefix: e,
  postfix: d,
  counter: l,
  ...n
}) => /* @__PURE__ */ o(u, { children: [
  /* @__PURE__ */ o(t, { ...n, children: [
    e ? /* @__PURE__ */ r(s, { children: e }) : null,
    F,
    m ? /* @__PURE__ */ r(f, { children: m }) : null,
    d ? /* @__PURE__ */ r(h, { children: d }) : null
  ] }),
  i || l ? /* @__PURE__ */ o(p, { children: [
    i,
    l ? /* @__PURE__ */ r(c, { children: l }) : null
  ] }) : null
] });
x.displayName = "TextFieldBase";
export {
  x as TextFieldBase
};
