// @sberbusiness/triplex-next/components/Step/Step.js
import { jsxs as g, jsx as i } from "react/jsx-runtime";
import { useRef as u } from "react";
import { EStepStatus as o, EStepPosition as e } from "./enums.js";
import { ETooltipAlign as n, ETooltipSize as _ } from "../Tooltip/enums.js";
import { Tooltip as a } from "../Tooltip/Tooltip.js";
import "../Typography/Title.js";
import { Text as S } from "../Typography/Text.js";
import "../Typography/Caption.js";
import { EFontWeightText as b, ETextSize as D } from "../Typography/enums.js";
import { c as N } from "../../chunks/vendor-Ba_pyhza.js";
const A = "step__8a2b79f6", R = "done__3fc8c503", x = "warning__d6aeb6f5", h = "active__245ca4e0", v = "error__bb1871d9", w = "disabled__cb7d7874", t = {
  step: A,
  default: "default__979f8e3a",
  done: R,
  warning: x,
  active: h,
  error: v,
  disabled: w
}, L = {
  [o.DEFAULT]: t.default,
  [o.ACTIVE]: t.active,
  [o.WARNING]: t.warning,
  [o.ERROR]: t.error,
  [o.DONE]: t.done,
  [o.DISABLED]: t.disabled
}, y = {
  [e.XFirst]: n.START,
  [e.Default]: n.CENTER,
  [e.XLast]: n.END
}, G = (r, s) => s === 0 ? e.XFirst : s + 1 === r ? e.XLast : e.Default, z = ({
  children: r,
  className: s,
  step: p,
  status: d,
  position: f = e.Default,
  ...m
}) => {
  const l = u(null), T = y[f], E = N(t.step, L[d], s), c = () => /* @__PURE__ */ i("div", { ref: l, className: E, ...m, children: /* @__PURE__ */ i(S, { size: D.B1, weight: b.SEMIBOLD, children: p }) });
  return r ? /* @__PURE__ */ g(a, { size: _.SM, toggleType: "hover", alignTip: T, targetRef: l, children: [
    /* @__PURE__ */ i(a.Target, { children: c() }),
    /* @__PURE__ */ i(a.Body, { children: r })
  ] }) : c();
};
z.displayName = "Step";
export {
  z as Step,
  G as calcPosition
};
