// @sberbusiness/triplex-next/components/Spoiler/Spoiler.js
import { jsx as n, jsxs as l } from "react/jsx-runtime";
import z, { useState as T, useRef as E, useCallback as w } from "react";
import { CaretdownStrokeSrvIcon20 as p, CaretdownStrokeSrvIcon16 as B } from "@sberbusiness/icons-next";
import { Button as L } from "../Button/Button.js";
import "../Button/ButtonBase.js";
import "../Button/ButtonIcon.js";
import "../Button/ButtonDropdown.js";
import "../Button/ButtonDropdownExtended.js";
import { EButtonTheme as R } from "../Button/enums.js";
import { EComponentSize as s } from "../../enums/EComponentSize.js";
import { createSizeToClassNameMap as g } from "../../utils/classNameMaps.js";
import { a as j, c as _ } from "../../chunks/vendor-Ba_pyhza.js";
const y = "spoiler__15264a2f", D = "opened__998aeaab", q = "caretIcon__c89287da", A = "head__6a3545b7", G = "content__c476f938", H = "hidden__51c75bb2", K = "lg__bfc6869e", $ = "md__d78a8e08", F = "sm__d9e2b027", e = {
  spoiler: y,
  opened: D,
  caretIcon: q,
  head: A,
  content: G,
  hidden: H,
  lg: K,
  md: $,
  sm: F
}, J = {
  [s.SM]: /* @__PURE__ */ n(B, { paletteIndex: 5, className: e.caretIcon }),
  [s.MD]: /* @__PURE__ */ n(p, { paletteIndex: 5, className: e.caretIcon }),
  [s.LG]: /* @__PURE__ */ n(p, { paletteIndex: 5, className: e.caretIcon })
}, O = g(e), P = z.forwardRef((f, h) => {
  const {
    children: I,
    expanded: c,
    onToggle: o,
    toggle: a,
    className: x,
    labelExpand: S,
    labelCollapse: u,
    rightBlock: N,
    size: r = s.MD,
    ...b
  } = f, d = c !== void 0, [C, v] = T(c), t = d ? c : C, m = E(`Spoiler-${j()}`), M = w(() => {
    const i = !t;
    d ? a == null || a(i) : v(i), o == null || o(i);
  }, [d, a, t, o]), k = _(
    e.spoiler,
    O[r],
    {
      [e.opened]: t
    },
    x
  );
  return /* @__PURE__ */ l("div", { ...b, className: k, "data-tx": "1.4.0", ref: h, children: [
    /* @__PURE__ */ l("div", { className: e.head, children: [
      /* @__PURE__ */ l(
        L,
        {
          "aria-expanded": t,
          "aria-controls": m.current,
          theme: R.LINK,
          size: r,
          onClick: M,
          children: [
            t ? u : S,
            J[r]
          ]
        }
      ),
      N
    ] }),
    /* @__PURE__ */ n("div", { id: m.current, className: _(e.content, { [e.hidden]: !t }), children: I })
  ] });
});
P.displayName = "Spoiler";
export {
  P as Spoiler
};
