// @sberbusiness/triplex-next/components/DesignTokens/components/ListItemControlsButton.js
const o = [
  "Background_Active",
  "Background_Default",
  "Background_Hover",
  "Color_Active",
  "Color_Default",
  "Color_Hover"
], r = {
  Background_Active: [{ ref: "ColorNeutral.30" }, { ref: "ColorDarkNeutral.100" }],
  // var(--triplex-next-ListItemControlsButton-Background_Active)
  Background_Default: [{ ref: "ColorNeutral.90" }, { ref: "ColorDarkNeutral.60" }],
  // var(--triplex-next-ListItemControlsButton-Background_Default)
  Background_Hover: [{ ref: "ColorNeutral.70" }, { ref: "ColorDarkNeutral.80" }],
  // var(--triplex-next-ListItemControlsButton-Background_Hover)
  Color_Active: [{ ref: "ColorBrand.30" }, { ref: "ColorBrand.40" }],
  // var(--triplex-next-ListItemControlsButton-Color_Active)
  Color_Default: [{ ref: "ColorBrand.40" }, { ref: "ColorBrand.60" }],
  // var(--triplex-next-ListItemControlsButton-Color_Default)
  Color_Hover: [{ ref: "ColorBrand.50" }, { ref: "ColorBrand.70" }]
  // var(--triplex-next-ListItemControlsButton-Color_Hover)
};
export {
  r as ListItemControlsButton_Tokens,
  o as designTokensComponentsListItemControlsButtonKeys
};
