// @sberbusiness/triplex-next/components/DesignTokens/components/SMSInput.js
const r = [
  "Refresh_Fill_Empty",
  "Refresh_Fill_Full",
  "Refresh_Disabled",
  "Submit_Background_Default",
  "Submit_Background_Hover",
  "Submit_Background_Active"
], e = {
  Refresh_Fill_Empty: [{ ref: "ColorDarkNeutralAlpha.70" }, { ref: "ColorNeutralAlpha.70" }],
  // var(--triplex-next-SMSInput-Refresh_Fill_Empty)
  Refresh_Fill_Full: [{ ref: "ColorBrand.40" }, { ref: "ColorBrand.60" }],
  // var(--triplex-next-SMSInput-Refresh_Fill_Full)
  Refresh_Disabled: [{ ref: "ColorDarkNeutralAlpha.90" }, { ref: "ColorNeutralAlpha.90" }],
  // var(--triplex-next-SMSInput-Refresh_Disabled)
  Submit_Background_Default: [{ ref: "ColorDarkNeutralAlpha.90" }, { ref: "ColorNeutralAlpha.90" }],
  // var(--triplex-next-SMSInput-Submit_Background_Default)
  Submit_Background_Hover: [{ ref: "ColorBrand.60" }, { ref: "ColorBrand.60" }],
  // var(--triplex-next-SMSInput-Submit_Background_Hover)
  Submit_Background_Active: [{ ref: "ColorBrand.50" }, { ref: "ColorBrand.50" }]
  // var(--triplex-next-SMSInput-Submit_Background_Active)
};
export {
  e as SMSInput_Tokens,
  r as designTokensComponentsSMSInputKeys
};
