// @sberbusiness/triplex-next/components/DesignTokens/components/StatusTracker.js
const r = [
  "Draft_Background",
  "Warning_Background",
  "Waiting_Background",
  "Rejected_Background",
  "Approved_Background",
  "Waiting_Color",
  "Rejected_Color",
  "Approved_Color",
  "Border_Color"
], o = {
  Draft_Background: [{ ref: "ColorNeutralAlpha.90" }, { ref: "ColorNeutralAlpha.100" }],
  // var(--triplex-next-StatusTracker-Draft_Background)
  Warning_Background: [{ ref: "ColorWarning.60" }, { ref: "ColorWarning.20" }],
  // var(--triplex-next-StatusTracker-Warning_Background)
  Waiting_Background: [{ ref: "ColorSystem.80" }, { ref: "ColorNeutralAlpha.60" }],
  // var(--triplex-next-StatusTracker-Waiting_Background)
  Rejected_Background: [{ ref: "ColorError.70" }, { ref: "ColorError.30" }],
  // var(--triplex-next-StatusTracker-Rejected_Background)
  Approved_Background: [{ ref: "ColorSuccess.60" }, { ref: "ColorSuccess.50" }],
  // var(--triplex-next-StatusTracker-Approved_Background)
  Waiting_Color: [{ ref: "ColorSystem.70" }, { ref: "ColorNeutralAlpha.50" }],
  // var(--triplex-next-StatusTracker-Waiting_Color)
  Rejected_Color: [{ ref: "ColorError.80" }, { ref: "ColorError.20" }],
  // var(--triplex-next-StatusTracker-Rejected_Color)
  Approved_Color: [{ ref: "ColorSuccess.60" }, { ref: "ColorSuccess.30" }],
  // var(--triplex-next-StatusTracker-Approved_Color)
  Border_Color: [{ ref: "ColorNeutral.100" }, { ref: "ColorNeutralAlpha.80" }]
  // var(--triplex-next-StatusTracker-Border_Color)
};
export {
  o as StatusTracker_Tokens,
  r as designTokensComponentsStatusTrackerKeys
};
