// @sberbusiness/triplex-next/components/DesignTokens/components/Tooltip.js
const o = [
  "Background",
  "Color",
  "Link_Desktop_Color_Active",
  "Link_Desktop_Color_Default",
  "Link_Desktop_Color_Hover",
  "Link_Desktop_Shadow_Focus",
  "Link_Mobile_Color_Active",
  "Link_Mobile_Color_Default",
  "Link_Mobile_Color_Hover",
  "Link_Mobile_Shadow_Focus"
], r = {
  Background: [{ ref: "ColorDarkNeutral.20" }, { ref: "ColorNeutral.80" }],
  // var(--triplex-next-Tooltip-Background)
  Color: [{ ref: "ColorNeutralAlpha.0" }, { ref: "ColorDarkNeutralAlpha.0" }],
  // var(--triplex-next-Tooltip-Color)
  // Цвета взяты из самого компонента Link (pixso), только "наоборот", т.к. у тултипа в светлой теме черная подложка
  Link_Desktop_Color_Active: [{ ref: "ColorBrand.40" }, { ref: "ColorBrand.30" }],
  // var(--triplex-next-Tooltip-Link_Desktop_Color_Active)
  Link_Desktop_Color_Default: [{ ref: "ColorBrand.60" }, { ref: "ColorBrand.40" }],
  // var(--triplex-next-Tooltip-Link_Desktop_Color_Default)
  Link_Desktop_Color_Hover: [{ ref: "ColorBrand.70" }, { ref: "ColorBrand.50" }],
  // var(--triplex-next-Tooltip-Link_Desktop_Color_Hover)
  Link_Desktop_Shadow_Focus: [{ value: "0 0 0 1px #FFD169" }, { value: "0 0 0 1px #FFD169" }],
  // var(--triplex-next-Tooltip-Link_Desktop_Shadow_Focus)
  // А в адаптиве почему-то подложка в светлой теме белая, поэтому взято из десктопа "наоборот"
  Link_Mobile_Color_Active: [{ ref: "ColorBrand.30" }, { ref: "ColorBrand.40" }],
  // var(--triplex-next-Tooltip-Link_Mobile_Color_Active)
  Link_Mobile_Color_Default: [{ ref: "ColorBrand.40" }, { ref: "ColorBrand.60" }],
  // var(--triplex-next-Tooltip-Link_Mobile_Color_Default)
  Link_Mobile_Color_Hover: [{ ref: "ColorBrand.50" }, { ref: "ColorBrand.70" }],
  // var(--triplex-next-Tooltip-Link_Mobile_Color_Hover)
  Link_Mobile_Shadow_Focus: [{ value: "0 0 0 1px #FFDD64" }, { value: "0 0 0 1px #FFDD64" }]
  // var(--triplex-next-Tooltip-Link_Mobile_Shadow_Focus)
};
export {
  r as Tooltip_Tokens,
  o as designTokensComponentsTooltipKeys
};
