// @sberbusiness/triplex-next/components/Table/TableBasicSettings/TableBasicSettings.js
import { jsx as o } from "react/jsx-runtime";
import { useRef as s } from "react";
import { ButtonDropdownExtended as a } from "../../Button/ButtonDropdownExtended.js";
import { c as l } from "../../../chunks/vendor-Ba_pyhza.js";
import { TableBasicSettingsBody as u } from "./components/TableBasicSettingsBody.js";
import { ColumnSettings as B } from "./components/ColumnSettings.js";
import { TableBasicSettingsFooter as w } from "./components/TableBasicSettingsFooter.js";
import { TableBasicSettingsHeader as x } from "./components/TableBasicSettingsHeader.js";
import { ETextSize as D } from "../../Typography/enums.js";
import { Link as T } from "../../Link/Link.js";
import { Text as k } from "../../Typography/Text.js";
const L = "tableSettingsLink__da9a023a", h = "tableSettingsDropdown__060fd76f", m = {
  tableSettingsLink: L,
  tableSettingsDropdown: h
}, t = ({ children: d, className: p, linkTitle: c, ...f }) => {
  const r = s(null), i = s(null), g = ({ opened: e, setOpened: n }) => /* @__PURE__ */ o(k, { size: D.B3, children: /* @__PURE__ */ o(
    T,
    {
      href: "#",
      "aria-haspopup": "listbox",
      "aria-controls": "button-dropdown-extended-list",
      "aria-expanded": e,
      onClick: (b) => {
        b.preventDefault(), n(!e);
      },
      ref: r,
      children: c
    }
  ) }), S = ({ className: e, ...n }) => /* @__PURE__ */ o(
    a.Dropdown,
    {
      className: l(m.tableSettingsDropdown, e),
      ...n,
      targetRef: r,
      ref: i,
      children: d
    }
  );
  return /* @__PURE__ */ o(
    a,
    {
      className: l(m.tableSettingsLink, p),
      renderButton: g,
      renderDropdown: S,
      dropdownRef: i,
      ...f
    }
  );
};
t.displayName = "TableBasicSettings";
t.Body = u;
t.ColumnSettings = B;
t.Footer = w;
t.Header = x;
export {
  t as TableBasicSettings
};
