// @sberbusiness/triplex-next/components/Overlay/OverlayPanel.js
import { jsx as o } from "react/jsx-runtime";
import n from "react";
import { c as d } from "../../chunks/vendor-Ba_pyhza.js";
import { EOverlayDirection as r } from "./OverlayBase.js";
import { s as e } from "../../chunks/Overlay.module-q-oPhooa.js";
const P = n.forwardRef(
  ({
    children: s,
    className: m,
    closing: T,
    direction: a,
    onTransitionEnd: l,
    opened: f,
    opening: h,
    setClosing: p,
    setOpened: u,
    setOpening: c,
    ...y
  }, O) => {
    const v = (t) => {
      const { target: g, currentTarget: i } = t;
      g === i && (p(!1), c(!1)), l == null || l(t);
    };
    return /* @__PURE__ */ o(
      "div",
      {
        className: d(e.overlayPanel, m, {
          [e.bottom]: a === r.BOTTOM,
          [e.left]: a === r.LEFT,
          [e.opened]: f,
          [e.right]: a === r.RIGHT,
          [e.top]: a === r.TOP
        }),
        onTransitionEnd: v,
        ref: O,
        ...y,
        children: /* @__PURE__ */ o("div", { className: e.overlayContent, children: s })
      }
    );
  }
);
P.displayName = "OverlayPanel";
export {
  P as OverlayPanel
};
