// @sberbusiness/triplex-next/components/Overlay/Overlay.js
import { jsx as m } from "react/jsx-runtime";
import { useState as h } from "react";
import { c as u } from "../../chunks/vendor-Ba_pyhza.js";
import { OverlayBase as k } from "./OverlayBase.js";
import { OverlayMask as B } from "./OverlayMask.js";
import { OverlayPanel as M } from "./OverlayPanel.js";
import { s as r } from "../../chunks/Overlay.module-q-oPhooa.js";
const s = ({
  children: o,
  className: i,
  direction: c,
  fixed: f,
  onClose: a,
  onClosing: e,
  onOpening: p,
  onOpen: y,
  opened: t,
  setOpened: v,
  ...d
}) => {
  const [n, l] = h(!1), O = () => {
    l(!0), e == null || e();
  }, x = () => {
    l(!1), a == null || a();
  };
  return /* @__PURE__ */ m(
    "div",
    {
      className: u(r.overlay, i, {
        [r.closing]: n,
        [r.fixed]: !!f,
        [r.opened]: t
      }),
      ...d,
      children: /* @__PURE__ */ m(
        k,
        {
          direction: c,
          onClose: x,
          onClosing: O,
          onOpening: p,
          onOpen: y,
          opened: t,
          setOpened: v,
          children: o
        }
      )
    }
  );
};
s.displayName = "Overlay";
s.Mask = B;
s.Panel = M;
export {
  s as Overlay
};
