// @sberbusiness/triplex-next/components/IslandAccordion/components/IslandAccordionItem.js
import { jsx as r, jsxs as c } from "react/jsx-runtime";
import j, { useState as w, useContext as z, useRef as F, useEffect as M } from "react";
import { CaretdownStrokeSrvIcon24 as Y, CrossStrokeSrvIcon24 as H } from "@sberbusiness/icons-next";
import { a as W, c as x } from "../../../chunks/vendor-Ba_pyhza.js";
import { Island as y } from "../../Island/Island.js";
import { ExpandAnimation as $ } from "../../ExpandAnimation/ExpandAnimation.js";
import { s as t } from "../../../chunks/IslandAccordion.module-7xAwgJVI.js";
import { IslandAccordionTitle as q } from "./IslandAccordionTitle.js";
import { IslandAccordionContent as B } from "./IslandAccordionContent.js";
import { IslandAccordionFooter as X } from "./IslandAccordionFooter.js";
import { createSizeToClassNameMap as D } from "../../../utils/classNameMaps.js";
import "../../Button/Button.js";
import "../../Button/ButtonBase.js";
import { ButtonIcon as G } from "../../Button/ButtonIcon.js";
import "../../Button/ButtonDropdown.js";
import "../../Button/ButtonDropdownExtended.js";
import { EIslandType as m } from "../../Island/enums.js";
import "../../Island/components/IslandBody.js";
import "../../Island/components/IslandHeader.js";
import "../../Island/components/IslandFooter.js";
import { Step as J } from "../../Step/Step.js";
import { EStepPosition as K } from "../../Step/enums.js";
import { IslandAccordionContext as L } from "../IslandAccordionContext.js";
const Q = {
  [m.TYPE_1]: t.type1,
  [m.TYPE_2]: t.type2,
  [m.TYPE_3]: t.type3
}, U = D(t), V = Object.assign(
  j.forwardRef(
    ({
      children: S,
      title: v,
      className: A,
      opened: o,
      disabled: a,
      onRemove: i,
      onToggle: s,
      id: n,
      num: p,
      status: d,
      stepHint: E,
      transitionProps: k,
      ...T
    }, _) => {
      const [e, l] = w(o || !1), { size: f, type: I } = z(L), h = F(W()).current, u = `${h}header`, C = `${h}body`;
      M(() => {
        o !== void 0 && e !== o && l(o);
      }, [o, e]);
      const b = () => {
        if (a)
          return;
        const N = !e;
        s == null || s(N, n), o === void 0 && l(N);
      }, O = () => {
        a || i == null || i(n);
      }, P = x(A, t.item, U[f], Q[I], {
        [t.disabled]: !!a,
        [t.opened]: e
      });
      return /* @__PURE__ */ r("li", { ...T, className: P, id: n, ref: _, children: /* @__PURE__ */ c(y, { className: t.island, size: f, type: I, children: [
        /* @__PURE__ */ r(y.Header, { children: /* @__PURE__ */ c(
          "button",
          {
            id: u,
            "aria-controls": C,
            "aria-expanded": e,
            type: "button",
            className: t.header,
            onClick: b,
            disabled: a,
            "data-tx": "1.4.0",
            children: [
              d && p && /* @__PURE__ */ r("div", { className: t.step, children: /* @__PURE__ */ r(J, { step: p, status: d, position: K.XFirst, children: a ? void 0 : E }) }),
              /* @__PURE__ */ c("div", { className: t.titleWrapper, children: [
                " ",
                v
              ] }),
              /* @__PURE__ */ r(
                "span",
                {
                  className: x(t.caretWrapper, "hoverable", {
                    active: e
                  }),
                  children: /* @__PURE__ */ r(
                    Y,
                    {
                      className: t.caretIcon,
                      "aria-hidden": "true",
                      paletteIndex: 5
                    }
                  )
                }
              )
            ]
          }
        ) }),
        i && /* @__PURE__ */ r("div", { className: t.remove, children: /* @__PURE__ */ r(G, { onClick: O, title: "Удалить", children: /* @__PURE__ */ r(H, { paletteIndex: 5 }) }) }),
        /* @__PURE__ */ r(
          $,
          {
            expanded: e && !a,
            id: C,
            role: "region",
            "aria-labelledby": u,
            transitionProps: k,
            children: S
          }
        )
      ] }) });
    }
  ),
  {
    Title: q,
    Content: B,
    Footer: X
  }
);
V.displayName = "IslandAccordionItem";
export {
  V as IslandAccordionItem
};
