// @sberbusiness/triplex-next/components/TreeView/TreeViewAbstractNode.js
var s = Object.defineProperty;
var n = (d, i, e) => i in d ? s(d, i, { enumerable: !0, configurable: !0, writable: !0, value: e }) : d[i] = e;
var t = (d, i, e) => n(d, typeof i != "symbol" ? i + "" : i, e);
import { AbstractTreeNode as o } from "../AbstractTree/AbstractTreeNode.js";
class r extends o {
  constructor(e) {
    super();
    t(this, "id");
    t(this, "active");
    t(this, "opened");
    t(this, "tabIndex", -1);
    /** Возвращает tabIndex ноды. */
    t(this, "getTabIndex", () => this.tabIndex);
    /** Устанавливает tabIndex ноды. */
    t(this, "setTabIndex", (e) => {
      this.tabIndex = e;
    });
    /** Возвращает id ноды. */
    t(this, "getId", () => this.id);
    /** Возвращает флаг активности ноды. */
    t(this, "getActive", () => !!this.active);
    /** Устанавливает флаг активности ноды. */
    t(this, "setActive", (e) => {
      this.active = e;
    });
    /** Возвращает флаг раскрытости ноды. */
    t(this, "getOpened", () => !!this.opened);
    /** Устанавливает флаг раскрытости ноды. */
    t(this, "setOpened", (e) => {
      this.opened = e;
    });
    this.id = e.id;
  }
}
export {
  r as TreeViewAbstractNode
};
