// @sberbusiness/triplex-next/components/TreeView/TreeViewAbstractNodeUtils.js
import { traverseAbstractTree as l } from "../AbstractTree/AbstractTreeNode.js";
const d = {
  getActiveNode: (e) => {
    let t;
    return l(e, (r) => r.getActive() ? (t = r, !1) : !0), t;
  },
  getNextNode: (e) => {
    const t = e.getChildren();
    if (e.getOpened() && t.length)
      return t[0];
    const r = d.getNextSiblingNode(e);
    if (r)
      return r;
    let n = e;
    for (; ; ) {
      const i = n.getParent();
      if (i.getParent()) {
        const g = d.getNextSiblingNode(i);
        if (g) {
          n = g;
          break;
        }
      } else {
        n = i.getChildren()[0];
        break;
      }
      n = i;
    }
    return n;
  },
  getNextSiblingNode: (e) => {
    const t = e.getParent();
    if (!t)
      return;
    const r = t.getChildren();
    let n = 0;
    if (r.some((i, g) => {
      if (i.getId() === e.getId())
        return n = g, !0;
    }), n !== r.length - 1)
      return r[n + 1];
  },
  getNode: (e, t) => {
    let r;
    return l(t, (n) => n.getId() === e ? (r = n, !1) : !0), r;
  },
  getPrevNode: (e) => {
    const t = d.getPrevSiblingNode(e);
    if (!t) {
      const n = e.getParent();
      if (n.getParent())
        return n;
      {
        let i = n.getChildren()[n.getChildren().length - 1];
        for (; i.getChildren().length > 0; )
          i = i.getChildren()[i.getChildren().length - 1];
        return i;
      }
    }
    let r = t;
    for (; ; ) {
      const n = r.getChildren();
      if (r.getOpened() && n.length)
        r = r.getChildren()[r.getChildren().length - 1];
      else
        break;
    }
    return r;
  },
  getPrevSiblingNode: (e) => {
    const t = e.getParent();
    if (!t)
      return;
    const r = t.getChildren();
    let n = 0;
    if (r.some((i, g) => {
      if (i.getId() === e.getId())
        return n = g, !0;
    }), n !== 0)
      return r[n - 1];
  },
  isLastNode: (e) => {
    const t = e.getParent();
    if (t) {
      if (t.getChildren()[t.getChildren().length - 1] == e)
        return d.isLastNode(t);
    } else
      return !0;
    return !1;
  },
  setActiveNextNode: (e) => {
    const t = d.getActiveNode(e);
    if (t) {
      const r = d.getNextNode(t);
      d.setActiveNode(r, e, !0);
    }
  },
  setActiveNode: (e, t, r) => {
    r ? (e.setActive(!0), l(t, (n) => (n.getActive() && n.getId() !== e.getId() && n.setActive(!1), !0))) : e.setActive(!1);
  },
  setActivePrevNode: (e) => {
    const t = d.getActiveNode(e);
    if (t) {
      const r = d.getPrevNode(t);
      d.setActiveNode(r, e, !0);
    }
  }
};
export {
  d as TreeViewAbstractNodeUtils
};
