// @sberbusiness/triplex-next/components/TabsExtended/components/TabsExtendedTabsWrapper.js
import { jsxs as k, Fragment as y, jsx as f } from "react/jsx-runtime";
import a, { useContext as A, useState as D, useRef as g, useLayoutEffect as R, isValidElement as B } from "react";
import { u as F, c as N, p as z } from "../../../chunks/vendor-Ba_pyhza.js";
import { TabsExtendedContext as _ } from "../TabsExtendedContext.js";
import { TabsExtendedTabContext as j } from "./TabsExtendedTabContext.js";
import { s as l } from "../../../chunks/TabsExtended.module-Be8qpZYz.js";
const G = ({
  children: s,
  className: C,
  ...w
}) => {
  const { setDropdownItemsIds: m, setInlineItemsIds: x, dropdownRef: p } = A(_), [c, I] = D(0), r = g(null), h = g(a.Children.count(s)), d = () => {
    if (r.current) {
      const { children: t } = r.current;
      let { right: e } = r.current.getBoundingClientRect();
      const i = [], u = [];
      let n = !1;
      e = e - 4, Array.from(t).forEach((o) => {
        const { right: b } = o.getBoundingClientRect();
        e - c < b ? (i.push(o.getAttribute("data-tab-item-id")), !n && e < b && (n = !0)) : o.getAttribute("data-tab-item-id") && u.push(o.getAttribute("data-tab-item-id"));
      }), m(n ? i : []), x(u);
    }
  }, E = () => {
    if (p.current) {
      const { width: t } = p.current.getBoundingClientRect();
      t !== c && I(t);
    }
  }, v = () => {
    const t = a.Children.count(s);
    h.current !== t && (h.current = t, d());
  };
  R(() => {
    E(), v();
  }), R(d, [c]);
  const T = (t) => a.Children.map(t, (e) => {
    if (!B(e))
      return e;
    const i = z(e.props, (u, n) => !n.startsWith("data-"));
    return a.createElement(e.type, i);
  }), { ref: W } = F({
    handleWidth: !0,
    onResize: d,
    refreshMode: "throttle",
    refreshRate: 150
  });
  return /* @__PURE__ */ k(y, { children: [
    /* @__PURE__ */ f(
      "div",
      {
        className: l.tabsFake,
        ref: (t) => {
          r.current !== t && (r.current = t), W(t);
        },
        children: /* @__PURE__ */ f(j.Provider, { value: { isFakeTab: !0 }, children: T(s) })
      }
    ),
    /* @__PURE__ */ f(
      "div",
      {
        className: N(l.tabsReal, { [l.hidden]: !r.current }, C),
        ...w,
        children: s
      }
    )
  ] });
};
export {
  G as TabsExtendedTabsWrapper
};
