// @sberbusiness/triplex-next/components/TabsExtended/components/TabsExtendedTab.js
import { jsx as p } from "react/jsx-runtime";
import x, { useContext as d } from "react";
import { s as o } from "../../../chunks/TabsExtended.module-Be8qpZYz.js";
import { c as I } from "../../../chunks/vendor-Ba_pyhza.js";
import { TabsExtendedContext as u } from "../TabsExtendedContext.js";
import { TabsExtendedTabContext as E } from "./TabsExtendedTabContext.js";
const h = x.forwardRef(
  ({ children: r, id: e, className: i, onClick: t, ...m }, l) => {
    const { dropdownItemsIds: n, inlineItemsIds: s, selectedId: b, onSelectTab: c } = d(u), { isFakeTab: a } = d(E), T = (f) => {
      c(e), t == null || t(f);
    };
    return /* @__PURE__ */ p(
      "span",
      {
        id: (() => {
          if (!(a || n.includes(e)))
            return e;
        })(),
        className: I(o.tabsExtendedTab, { [o.hidden]: n.includes(e) }, i),
        onClick: a ? void 0 : T,
        ...m,
        role: "presentation",
        "data-tab-item-id": e,
        ref: a ? void 0 : l,
        children: r({
          isFirstInlineTab: s[0] === e,
          isLastInlineTab: s[s.length - 1] === e,
          selected: b === e
        })
      }
    );
  }
);
h.displayName = "TabsExtendedTab";
export {
  h as TabsExtendedTab
};
