// @sberbusiness/triplex-next/components/ThemeProvider/components/ThemeProviderView.js
import { jsx as p } from "react/jsx-runtime";
import { useRef as u, useEffect as k } from "react";
import { ThemeProviderContext as c } from "../ThemeProviderContext.js";
import { ETriplexNextTheme as m } from "../ETriplexNextTheme.js";
import { DesignTokensCore as D } from "../../DesignTokens/DesignTokensCore.js";
import { DesignTokensCoreThemeDark as d } from "../../DesignTokens/DesignTokensCoreThemeDark.js";
import { DesignTokensComponents as v } from "../../DesignTokens/DesignTokensComponents.js";
import { DesignTokensComponentsThemeDark as x } from "../../DesignTokens/DesignTokensComponentsThemeDark.js";
import { h as g } from "../../../chunks/vendor-Ba_pyhza.js";
const I = ({
  children: s,
  scopeClassName: r,
  scopeRef: t,
  theme: e,
  tokens: f
}) => {
  const o = u("");
  k(() => {
    var n;
    return r && ((n = t.current) == null || n.classList.add(r)), o.current = r, () => {
      var i;
      o.current && ((i = t.current) == null || i.classList.remove(o.current));
    };
  }, [r]);
  const T = g(
    {},
    f,
    e === m.LIGHT ? D : d,
    e === m.LIGHT ? v : x
  );
  return /* @__PURE__ */ p(c.Provider, { value: { scopeClassName: r, theme: e, tokens: T }, children: s });
};
export {
  I as ThemeProviderView
};
