// @sberbusiness/triplex-next/components/Tag/Tag.js
import { jsx as o, jsxs as i } from "react/jsx-runtime";
import D, { useCallback as S } from "react";
import { c as E } from "../../chunks/vendor-Ba_pyhza.js";
import { EditStrokeSrvIcon24 as L, EditStrokeSrvIcon20 as N, EditStrokeSrvIcon16 as v, CrossStrokeSrvIcon24 as y, CrossStrokeSrvIcon20 as z, CrossStrokeSrvIcon16 as G } from "@sberbusiness/icons-next";
import { Text as h } from "../Typography/Text.js";
import { ETextSize as I, EFontType as _ } from "../Typography/enums.js";
import { ButtonIcon as T } from "../Button/ButtonIcon.js";
import { createSizeToClassNameMap as j } from "../../utils/classNameMaps.js";
import { EComponentSize as e } from "../../enums/EComponentSize.js";
const A = "tag__e5da6a0e", R = "sm__19da04d7", w = "md__2b874a84", F = "lg__b6a87358", P = "content__6f0bb8df", l = {
  tag: A,
  sm: R,
  md: w,
  lg: F,
  content: P
}, Y = j(l), b = {
  [e.SM]: I.B4,
  [e.MD]: I.B3,
  [e.LG]: I.B2
}, q = {
  [e.SM]: /* @__PURE__ */ o(v, { paletteIndex: 5 }),
  [e.MD]: /* @__PURE__ */ o(N, { paletteIndex: 5 }),
  [e.LG]: /* @__PURE__ */ o(L, { paletteIndex: 5 })
}, H = {
  [e.SM]: /* @__PURE__ */ o(G, { paletteIndex: 5 }),
  [e.MD]: /* @__PURE__ */ o(z, { paletteIndex: 5 }),
  [e.LG]: /* @__PURE__ */ o(y, { paletteIndex: 5 })
}, $ = D.forwardRef(
  ({ children: f, id: s, className: u, size: r, disabled: n, onRemove: a, onEdit: c, removeButtonProps: x, editButtonProps: M, ...g }, C) => {
    const k = S(() => {
      const t = n ? _.DISABLED : _.PRIMARY;
      return /* @__PURE__ */ o(h, { className: l.content, type: t, size: b[r], children: f });
    }, [f, r, n]), B = S(() => {
      const { onClick: t, ...m } = M || {};
      return /* @__PURE__ */ o(
        T,
        {
          disabled: n,
          onClick: (p) => {
            c == null || c(s), t == null || t(p);
          },
          ...m,
          children: q[r]
        }
      );
    }, [s, r, n, M, c]), d = S(() => {
      const { onClick: t, ...m } = x || {};
      return /* @__PURE__ */ o(
        T,
        {
          disabled: n,
          onClick: (p) => {
            a == null || a(s), t == null || t(p);
          },
          ...m,
          children: H[r]
        }
      );
    }, [s, r, n, x, a]);
    return /* @__PURE__ */ i("span", { className: E(l.tag, Y[r], u), ...g, ref: C, children: [
      k(),
      c && B(),
      d()
    ] });
  }
);
export {
  $ as Tag
};
