// @sberbusiness/triplex-next/components/Card/CardAction.js
var N = Object.defineProperty;
var y = (c, i, e) => i in c ? N(c, i, { enumerable: !0, configurable: !0, writable: !0, value: e }) : c[i] = e;
var o = (c, i, e) => y(c, typeof i != "symbol" ? i + "" : i, e);
import { jsx as R } from "react/jsx-runtime";
import u from "react";
import { CardContent as T } from "./components/CardContent/CardContent.js";
import { CardMedia as _ } from "./components/CardMedia.js";
import { ECardRoundingSize as M, ECardTheme as w } from "./enums.js";
import { EFocusSource as r } from "../../enums/EFocusSource.js";
import { mapCardRoundingSizeToCssClass as x, mapCardThemeToCssClass as K } from "./utils.js";
import { isKey as f } from "../../utils/keyboard.js";
import { c as k } from "../../chunks/vendor-Ba_pyhza.js";
import { c as B } from "../../chunks/Card.module-kTbr6OQy.js";
const O = "action__a2b4210b", F = "selected__d176b229", V = "focusVisible__0fb99f00", a = {
  action: O,
  selected: F,
  focusVisible: V
};
class l extends u.Component {
  constructor() {
    super(...arguments);
    o(this, "state", {
      focusSource: r.NONE,
      isControlled: this.props.selected !== void 0,
      isSelected: !!this.props.selected
    });
    o(this, "ref", u.createRef());
    o(this, "handleClick", (e) => {
      const { onClick: s } = this.props;
      s == null || s(e), this.handleToggle();
    });
    o(this, "handleMouseDown", (e) => {
      const { onMouseDown: s } = this.props, { focusSource: t } = this.state;
      s == null || s(e), t || this.setState({ focusSource: r.MOUSE });
    });
    o(this, "handleKeyDown", (e) => {
      const { onKeyDown: s } = this.props;
      s == null || s(e), f(e.keyCode, "SPACE") ? (e.preventDefault(), this.handleToggle()) : f(e.keyCode, "ENTER") && this.handleToggle();
    });
    o(this, "handleFocus", (e) => {
      const { onFocus: s } = this.props, { focusSource: t } = this.state, { current: n } = this.ref;
      s == null || s(e), !t && n === e.target && this.setState({ focusSource: r.KEYBOARD });
    });
    o(this, "handleBlur", (e) => {
      const { onBlur: s } = this.props, { current: t } = this.ref;
      s == null || s(e), t !== document.activeElement && t === e.target && this.setState({ focusSource: r.NONE });
    });
    o(this, "handleToggle", () => {
      const { toggle: e, selected: s, onToggle: t } = this.props, { isControlled: n, isSelected: d } = this.state;
      n ? e == null || e(!s) : this.setState(
        (h) => ({ isSelected: !h.isSelected }),
        () => t == null ? void 0 : t(!d)
      );
    });
  }
  componentDidUpdate(e) {
    const { selected: s, onToggle: t } = this.props;
    s !== e.selected && (t == null || t(!!s));
  }
  render() {
    const {
      children: e,
      className: s,
      onClick: t,
      onMouseDown: n,
      onKeyDown: d,
      onFocus: h,
      onBlur: z,
      roundingSize: p = M.MD,
      onToggle: A,
      selected: m,
      toggle: U,
      theme: S = w.GENERAL,
      ...C
    } = this.props, { isControlled: E, isSelected: b, focusSource: g } = this.state, D = k(
      B.card,
      a.action,
      K[S],
      x[p],
      {
        [a.focusVisible]: g === r.KEYBOARD,
        [a.selected]: E ? !!m : b
      },
      s
    );
    return /* @__PURE__ */ R(
      "div",
      {
        className: D,
        tabIndex: 0,
        onClick: this.handleClick,
        onMouseDown: this.handleMouseDown,
        onKeyDown: this.handleKeyDown,
        onFocus: this.handleFocus,
        onBlur: this.handleBlur,
        role: "button",
        ...C,
        ref: this.ref,
        "data-tx": "1.4.0",
        children: e
      }
    );
  }
}
o(l, "displayName", "CardAction"), o(l, "Content", T), o(l, "Media", _);
export {
  l as CardAction
};
