// @sberbusiness/triplex-next/components/LightBox/LightBox.js
import { jsxs as B, Fragment as T, jsx as o } from "react/jsx-runtime";
import { useRef as l, useEffect as p } from "react";
import { c as V, z as A } from "../../chunks/vendor-Ba_pyhza.js";
import { s as e, L as D } from "../../chunks/LightBoxContent-DmoszkOx.js";
import { LightBoxControls as F } from "./LightBoxControls/LightBoxControls.js";
import { LightBoxSideOverlay as I } from "./LightBoxSideOverlay/LightBoxSideOverlay.js";
import { Portal as k } from "../Portal/Portal.js";
import { addClassNameWithScrollbarWidth as R } from "../../utils/scroll/scrollbar.js";
import { LightBoxTopOverlay as j } from "./LightBoxTopOverlay/LightBoxTopOverlay.js";
import { LightBoxViewManager as q } from "./LightBoxViewManager/LightBoxViewManager.js";
import { MobileView as z } from "../MobileView/MobileView.js";
import { FocusTrapUtils as W } from "../../utils/focus/FocusTrapUtils.js";
import { useToken as $ } from "../ThemeProvider/useToken.js";
const H = {
  "scroll-0": "scroll-0__ee9a2b1c",
  "scroll-15": "scroll-15__7138e655",
  "scroll-16": "scroll-16__1082f6d9",
  "scroll-17": "scroll-17__f763fcf4"
}, L = "LightBox-next-mount-node", N = "LightBox-next-view-manager-node", v = [e.bodyOverflowHidden], r = ({
  children: y,
  className: M,
  focusTrapProps: n,
  forwardRef: s,
  mountNode: m,
  lightBoxViewManagerNodeId: d = N,
  isLoading: u,
  isSideOverlayOpened: i,
  isTopOverlayOpened: E,
  ...b
}) => {
  const c = l(null), g = l(null), { scopeClassName: C } = $(), S = () => {
    let t = null;
    return m ? t = m : (t = document.querySelector(`#${L}`), t || (t = document.createElement("div"), t.setAttribute("id", L), document.body.appendChild(t))), t;
  }, _ = () => {
    let t = null;
    return d && (t = document.querySelector(`#${d}`)), t || (t = document.createElement("div"), t.setAttribute("id", N), document.body.appendChild(t)), t;
  }, a = l(S()), h = l(_()), f = () => {
    v.forEach((t) => document.documentElement.classList.add(t));
  };
  p(() => (R(H), f(), setTimeout(f, 100), () => v.forEach((t) => document.documentElement.classList.remove(t))), []), p(() => {
    if (!i && c.current) {
      const t = Math.round(Math.random() * 100);
      c.current.style.zIndex = t.toString();
    }
  }, [i]);
  const w = (t) => {
    g.current = t, s && (s.current = t);
  }, O = V(
    C,
    e.lightBox,
    {
      [e.isLoading]: !!u,
      [e.lightBoxSideOverlayActive]: !!i,
      [e.lightBoxTopOverlayActive]: !!E
    },
    M
  );
  if (!a.current)
    return null;
  const x = () => /* @__PURE__ */ B("div", { className: O, ref: w, role: "dialog", "aria-modal": "true", ...b, children: [
    /* @__PURE__ */ o("div", { className: e.lightBoxBackdrop }),
    y,
    /* @__PURE__ */ o("span", { ref: c, className: e.tempElSafariBug })
  ] });
  return /* @__PURE__ */ B(T, { children: [
    /* @__PURE__ */ o(k, { container: a.current, children: /* @__PURE__ */ o(
      z,
      {
        fallback: /* @__PURE__ */ o(
          A.FocusTrap,
          {
            active: !u,
            ...n,
            focusTrapOptions: {
              clickOutsideDeactivates: !0,
              initialFocus: () => W.getFirstInteractionElementByDataAttr(g.current),
              preventScroll: !0,
              ...n == null ? void 0 : n.focusTrapOptions
            },
            children: x()
          }
        ),
        children: x()
      }
    ) }),
    h.current && /* @__PURE__ */ o(
      q,
      {
        lightBoxViewManagerNode: h.current,
        lightBoxMountNode: a.current
      }
    )
  ] });
};
r.Content = D;
r.SideOverlay = I;
r.TopOverlay = j;
r.Controls = F;
r.displayName = "LightBox";
export {
  r as LightBox,
  L as lightBoxMountNodeIdDefault,
  N as lightBoxViewManagerNodeIdDefault
};
