// @sberbusiness/triplex-next/components/FormField/FormField.js
import { jsx as n } from "react/jsx-runtime";
import { useState as e } from "react";
import { FormFieldContext as G } from "./FormFieldContext.js";
import { TARGET_PADDING_X_DEFAULT as _ } from "./consts.js";
import { c as I } from "../../chunks/vendor-Ba_pyhza.js";
import { EFormFieldStatus as r } from "./enums.js";
import { EComponentSize as N } from "../../enums/EComponentSize.js";
const P = "formField__7d7fcf94", S = "disabled__b720611e", T = "active__242fb002", L = "error__5d66e340", u = "warning__38b847e0", i = {
  formField: P,
  disabled: S,
  active: T,
  error: L,
  warning: u,
  "size-sm": "size-sm__7540e812",
  "size-md": "size-md__c6a543db",
  "size-lg": "size-lg__1760a0c4"
}, V = ({
  children: p,
  className: F,
  status: t = r.DEFAULT,
  onMouseEnter: s,
  onMouseLeave: o,
  style: x,
  size: c = N.LG,
  ...g
}) => {
  const [a, h] = e(!1), [b, l] = e(!1), [v, z] = e(""), [m, E] = e(_), [f, A] = e(_), [D, R] = e(!1), W = (d) => {
    l(!0), s == null || s(d);
  }, w = (d) => {
    l(!1), o == null || o(d);
  };
  return /* @__PURE__ */ n(
    G.Provider,
    {
      value: {
        status: t,
        focused: a,
        hovered: b,
        id: v,
        postfixWidth: m,
        prefixWidth: f,
        setFocused: h,
        setId: z,
        setPostfixWidth: E,
        setPrefixWidth: A,
        setValueExist: R,
        valueExist: D,
        size: c
      },
      children: /* @__PURE__ */ n(
        "div",
        {
          className: I(
            i.formField,
            i[`size-${c}`],
            {
              [i.active]: a,
              [i.disabled]: t === r.DISABLED,
              [i.error]: t === r.ERROR,
              [i.warning]: t === r.WARNING
            },
            F
          ),
          onMouseEnter: W,
          onMouseLeave: w,
          "data-tx": "1.4.0",
          style: { paddingLeft: f, paddingRight: m, ...x },
          ...g,
          children: p
        }
      )
    }
  );
};
export {
  V as FormField
};
