// @sberbusiness/triplex-next/components/DesignTokens/DesignTokensComponents.js
import { ETriplexNextTheme as o } from "../ThemeProvider/ETriplexNextTheme.js";
import { GetTokensValueByTheme as r } from "./GetTokensValueByTheme.js";
import { Button_Tokens as e } from "./components/Button.js";
import { Checkbox_Tokens as t } from "./components/Checkbox.js";
import { FormField_Tokens as m } from "./components/FormField.js";
import { Link_Tokens as T } from "./components/Link.js";
import { Tabs_Tokens as n } from "./components/Tabs.js";
import { TabsLine_Tokens as i } from "./components/TabsLine.js";
import { Loader_Tokens as s } from "./components/Loader.js";
import { LoaderScreen_Tokens as p } from "./components/LoaderScreen.js";
import { Radio_Tokens as L } from "./components/Radio.js";
import { Typography_Tokens as k } from "./components/Typography.js";
import { Tooltip_Tokens as a } from "./components/Tooltip.js";
import { AlertContext_Tokens as I } from "./components/AlertContext.js";
import { AlertProcess_Tokens as f } from "./components/AlertProcess.js";
import { Divider_Tokens as H } from "./components/Divider.js";
import { Dropdown_Tokens as G } from "./components/Dropdown.js";
import { DropdownList_Tokens as _ } from "./components/DropdownList.js";
import { DropdownMobile_Tokens as d } from "./components/DropdownMobile.js";
import { DropdownMobileList_Tokens as l } from "./components/DropdownMobileList.js";
import { Overlay_Tokens as S } from "./components/Overlay.js";
import { ListItem_Tokens as g } from "./components/ListItem.js";
import { ListItemControlsButton_Tokens as c } from "./components/ListItemControlsButton.js";
import { ListMaster_Tokens as C } from "./components/ListMaster.js";
import { Marker_Tokens as b } from "./components/Marker.js";
import { MarkerStatus_Tokens as M } from "./components/MarkerStatus.js";
import { UploadZone_Tokens as u } from "./components/UploadZone.js";
import { Pagination_Tokens as D } from "./components/Pagination.js";
import { Card_Tokens as h } from "./components/Card.js";
import { Skeleton_Tokens as x } from "./components/Skeleton.js";
import { SegmentedControl_Tokens as B } from "./components/SegmentedControl.js";
import { SegmentedControlSegment_Tokens as w } from "./components/SegmentedControlSegment.js";
import { Stepper_Tokens as P } from "./components/Stepper.js";
import { HeaderPage_Tokens as y } from "./components/HeaderPage.js";
import { FooterPage_Tokens as A } from "./components/FooterPage.js";
import { Island_Tokens as F } from "./components/Island.js";
import { IslandAccordion_Tokens as v } from "./components/IslandAccordion.js";
import { TableBasic_Tokens as N } from "./components/TableBasic.js";
import { Chip_Tokens as O } from "./components/Chip.js";
import { LightBox_Tokens as R } from "./components/LightBox.js";
import { Tag_Tokens as U } from "./components/Tag.js";
import { Notification_Tokens as Z } from "./components/Notification.js";
import { SMSInput_Tokens as E } from "./components/SMSInput.js";
import { StatusTracker_Tokens as V } from "./components/StatusTracker.js";
import { Step_Tokens as j } from "./components/Step.js";
const No = {
  AlertContext: r(
    o.LIGHT,
    I
  ),
  AlertProcess: r(
    o.LIGHT,
    f
  ),
  Button: r(o.LIGHT, e),
  Card: r(o.LIGHT, h),
  Checkbox: r(o.LIGHT, t),
  Chip: r(o.LIGHT, O),
  Divider: r(o.LIGHT, H),
  Dropdown: r(o.LIGHT, G),
  DropdownList: r(
    o.LIGHT,
    _
  ),
  DropdownMobile: r(
    o.LIGHT,
    d
  ),
  DropdownMobileList: r(
    o.LIGHT,
    l
  ),
  FooterPage: r(
    o.LIGHT,
    A
  ),
  FormField: r(o.LIGHT, m),
  HeaderPage: r(
    o.LIGHT,
    y
  ),
  Island: r(o.LIGHT, F),
  IslandAccordion: r(
    o.LIGHT,
    v
  ),
  Link: r(o.LIGHT, T),
  ListItem: r(o.LIGHT, g),
  ListItemControlsButton: r(
    o.LIGHT,
    c
  ),
  ListMaster: r(
    o.LIGHT,
    C
  ),
  Loader: r(o.LIGHT, s),
  LoaderScreen: r(
    o.LIGHT,
    p
  ),
  Marker: r(o.LIGHT, b),
  MarkerStatus: r(
    o.LIGHT,
    M
  ),
  Overlay: r(o.LIGHT, S),
  Pagination: r(
    o.LIGHT,
    D
  ),
  Radio: r(o.LIGHT, L),
  SegmentedControl: r(
    o.LIGHT,
    B
  ),
  SegmentedControlSegment: r(
    o.LIGHT,
    w
  ),
  Skeleton: r(o.LIGHT, x),
  SMSInput: r(o.LIGHT, E),
  Step: r(o.LIGHT, j),
  Stepper: r(o.LIGHT, P),
  TableBasic: r(
    o.LIGHT,
    N
  ),
  Tabs: r(o.LIGHT, n),
  Tag: r(o.LIGHT, U),
  TabsLine: r(o.LIGHT, i),
  Tooltip: r(o.LIGHT, a),
  Typography: r(
    o.LIGHT,
    k
  ),
  UploadZone: r(
    o.LIGHT,
    u
  ),
  LightBox: r(o.LIGHT, R),
  Notification: r(
    o.LIGHT,
    Z
  ),
  StatusTracker: r(
    o.LIGHT,
    V
  )
};
export {
  No as DesignTokensComponents
};
