// @sberbusiness/triplex-next/components/DesignTokens/DesignTokenUtils.js
import { ETriplexNextTheme as i } from "../ThemeProvider/ETriplexNextTheme.js";
import { DesignTokensCore as m } from "./DesignTokensCore.js";
import { DesignTokensCoreThemeDark as a } from "./DesignTokensCoreThemeDark.js";
import { DesignTokensComponents as c } from "./DesignTokensComponents.js";
import { DesignTokensComponentsThemeDark as f } from "./DesignTokensComponentsThemeDark.js";
import { h as l } from "../../chunks/vendor-Ba_pyhza.js";
const n = {
  getCSSVariableByTokenGroup: (t, s) => {
    const e = Object.keys(t)[0], r = "triplex-next";
    let o = "1.4.0";
    return o = o.replace(/\./g, "-"), Object.keys(t[e]).map(
      (T) => `--${r}-${e}-${T}-${o}: ${n.getTokenValue(
        t[e][T],
        s
      )};`
    ).join(`
`);
  },
  // Возвращает токены, которые использует компонент.
  getComponentTokens: (t) => {
    const s = [], e = [];
    return Object.keys(c).forEach((r) => {
      Object.keys(c[r]).forEach(
        (o) => {
          if (r === t) {
            e.push(`${r}.${o}`);
            return;
          }
        }
      );
    }), { components: e, core: s };
  },
  getStyle: (t = i.LIGHT, s) => {
    let e = "";
    switch (t) {
      case i.LIGHT: {
        const r = l({}, s, m, c);
        e = n.getStyleByTokens(r);
        break;
      }
      case i.DARK: {
        const r = l(
          {},
          s,
          a,
          f
        );
        e = n.getStyleByTokens(r);
        break;
      }
    }
    return e;
  },
  // Возвращает строку с css-переменными.
  getStyleByTokens: (t) => Object.keys(t).map(
    (e) => n.getCSSVariableByTokenGroup(
      {
        [e]: t[e]
      },
      t
    )
  ).join(`
`).trim(),
  // Возвращает значение токена. Если токен имеет значение в виде строки, будет возвращена эта строка, если токен ссылается на другой токен, будет возвращено значение другого токена.
  getTokenValue: (t, s) => {
    let e = "";
    if (t.value)
      e = t.value;
    else if (t.ref) {
      const r = t.ref.split("."), o = s[r[0]][r[1]].value;
      o ? e = o : s[r[0]][r[1]].ref && (e = n.getTokenValue(
        s[r[0]][r[1]],
        s
      ));
    }
    return e;
  }
};
export {
  n as DesignTokenUtils
};
