// @sberbusiness/triplex-next/components/DesignTokens/DesignTokensComponentsThemeDark.js
import { Divider_Tokens as e } from "./components/Divider.js";
import { ETriplexNextTheme as o } from "../ThemeProvider/ETriplexNextTheme.js";
import { GetTokensValueByTheme as r } from "./GetTokensValueByTheme.js";
import { Button_Tokens as t } from "./components/Button.js";
import { Checkbox_Tokens as m } from "./components/Checkbox.js";
import { FormField_Tokens as n } from "./components/FormField.js";
import { Link_Tokens as i } from "./components/Link.js";
import { Tabs_Tokens as s } from "./components/Tabs.js";
import { TabsLine_Tokens as p } from "./components/TabsLine.js";
import { Loader_Tokens as T } from "./components/Loader.js";
import { LoaderScreen_Tokens as k } from "./components/LoaderScreen.js";
import { Radio_Tokens as D } from "./components/Radio.js";
import { Typography_Tokens as a } from "./components/Typography.js";
import { Tooltip_Tokens as A } from "./components/Tooltip.js";
import { AlertContext_Tokens as f } from "./components/AlertContext.js";
import { AlertProcess_Tokens as R } from "./components/AlertProcess.js";
import { Dropdown_Tokens as K } from "./components/Dropdown.js";
import { DropdownList_Tokens as _ } from "./components/DropdownList.js";
import { DropdownMobile_Tokens as d } from "./components/DropdownMobile.js";
import { DropdownMobileList_Tokens as l } from "./components/DropdownMobileList.js";
import { Overlay_Tokens as S } from "./components/Overlay.js";
import { ListItem_Tokens as L } from "./components/ListItem.js";
import { ListItemControlsButton_Tokens as g } from "./components/ListItemControlsButton.js";
import { ListMaster_Tokens as c } from "./components/ListMaster.js";
import { Marker_Tokens as C } from "./components/Marker.js";
import { MarkerStatus_Tokens as b } from "./components/MarkerStatus.js";
import { UploadZone_Tokens as M } from "./components/UploadZone.js";
import { Pagination_Tokens as h } from "./components/Pagination.js";
import { Card_Tokens as u } from "./components/Card.js";
import { Skeleton_Tokens as I } from "./components/Skeleton.js";
import { SegmentedControl_Tokens as x } from "./components/SegmentedControl.js";
import { SegmentedControlSegment_Tokens as B } from "./components/SegmentedControlSegment.js";
import { Stepper_Tokens as w } from "./components/Stepper.js";
import { HeaderPage_Tokens as P } from "./components/HeaderPage.js";
import { FooterPage_Tokens as y } from "./components/FooterPage.js";
import { Island_Tokens as F } from "./components/Island.js";
import { IslandAccordion_Tokens as v } from "./components/IslandAccordion.js";
import { TableBasic_Tokens as N } from "./components/TableBasic.js";
import { Chip_Tokens as H } from "./components/Chip.js";
import { LightBox_Tokens as O } from "./components/LightBox.js";
import { Tag_Tokens as U } from "./components/Tag.js";
import { Notification_Tokens as Z } from "./components/Notification.js";
import { SMSInput_Tokens as E } from "./components/SMSInput.js";
import { StatusTracker_Tokens as G } from "./components/StatusTracker.js";
import { Step_Tokens as V } from "./components/Step.js";
const No = {
  AlertContext: r(
    o.DARK,
    f
  ),
  AlertProcess: r(
    o.DARK,
    R
  ),
  Button: r(o.DARK, t),
  Card: r(o.DARK, u),
  Checkbox: r(o.DARK, m),
  Chip: r(o.DARK, H),
  Divider: r(o.DARK, e),
  Dropdown: r(o.DARK, K),
  DropdownList: r(
    o.DARK,
    _
  ),
  DropdownMobile: r(
    o.DARK,
    d
  ),
  DropdownMobileList: r(
    o.DARK,
    l
  ),
  FooterPage: r(
    o.DARK,
    y
  ),
  FormField: r(o.DARK, n),
  HeaderPage: r(
    o.DARK,
    P
  ),
  Island: r(o.DARK, F),
  IslandAccordion: r(
    o.DARK,
    v
  ),
  Link: r(o.DARK, i),
  ListItem: r(o.DARK, L),
  ListItemControlsButton: r(
    o.DARK,
    g
  ),
  ListMaster: r(
    o.DARK,
    c
  ),
  Loader: r(o.DARK, T),
  LoaderScreen: r(
    o.DARK,
    k
  ),
  Marker: r(o.DARK, C),
  MarkerStatus: r(
    o.DARK,
    b
  ),
  Overlay: r(o.DARK, S),
  Pagination: r(
    o.DARK,
    h
  ),
  Radio: r(o.DARK, D),
  SegmentedControl: r(
    o.DARK,
    x
  ),
  SegmentedControlSegment: r(
    o.DARK,
    B
  ),
  Skeleton: r(o.DARK, I),
  SMSInput: r(o.DARK, E),
  Step: r(o.DARK, V),
  Stepper: r(o.DARK, w),
  TableBasic: r(
    o.DARK,
    N
  ),
  Tabs: r(o.DARK, s),
  Tag: r(o.DARK, U),
  TabsLine: r(o.DARK, p),
  Tooltip: r(o.DARK, A),
  Typography: r(
    o.DARK,
    a
  ),
  UploadZone: r(
    o.DARK,
    M
  ),
  LightBox: r(o.DARK, O),
  Notification: r(
    o.DARK,
    Z
  ),
  StatusTracker: r(
    o.DARK,
    G
  )
};
export {
  No as DesignTokensComponentsThemeDark
};
