// @sberbusiness/triplex-next/components/TabsExtended/TabsExtended.js
import { jsx as s } from "react/jsx-runtime";
import { useState as r, useRef as T } from "react";
import { c as b } from "../../chunks/vendor-Ba_pyhza.js";
import { TabsExtendedContext as E } from "./TabsExtendedContext.js";
import { TabsExtendedContent as I } from "./components/TabsExtendedContent.js";
import { s as u } from "../../chunks/TabsExtended.module-Be8qpZYz.js";
import { ETabsExtendedType as C } from "./enums.js";
import { t as v } from "../../chunks/utils-C9-tOm2Z.js";
import "./components/TabsExtendedDropdownWrapper.js";
import "./components/TabsExtendedTab.js";
import "./components/TabsExtendedTabButton.js";
import "./components/TabsExtendedTabContext.js";
import "./components/TabsExtendedTabsWrapper.js";
const w = ({
  children: n,
  className: m,
  selectedId: t,
  onSelectTab: d,
  type: e = C.TYPE_1,
  ...a
}) => {
  const [i, p] = r([]), [l, f] = r([]), c = T(null), x = (o) => {
    t !== o && d(o);
  };
  return /* @__PURE__ */ s(
    E.Provider,
    {
      value: {
        dropdownItemsIds: l,
        dropdownRef: c,
        inlineItemsIds: i,
        onSelectTab: x,
        selectedId: t,
        setDropdownItemsIds: f,
        setInlineItemsIds: p,
        type: e
      },
      children: /* @__PURE__ */ s(
        "div",
        {
          className: b(u.tabsExtended, v[e], m),
          role: "tablist",
          ...a,
          children: n
        }
      )
    }
  );
};
w.Content = I;
export {
  w as TabsExtended
};
