// @sberbusiness/triplex-next/components/Dropdown/desktop/DropdownList.js
import { jsxs as _, jsx as h } from "react/jsx-runtime";
import l, { useContext as y, useRef as B, useState as b, useEffect as f } from "react";
import { c as z } from "../../../chunks/vendor-Ba_pyhza.js";
import { s as C, D as x } from "../../../chunks/DropdownListItem-D2hzyBGc.js";
import { EVENT_KEY_CODES as w } from "../../../utils/keyboard.js";
import { DropdownListContext as V } from "../DropdownListContext.js";
import { LoaderSmall as W } from "../../Loader/LoaderSmall/LoaderSmall.js";
import { ELoaderSmallTheme as j } from "../../Loader/LoaderSmall/enum.js";
import "../../Loader/LoaderMiddle/LoaderMiddle.js";
import { EComponentSize as K } from "../../../enums/EComponentSize.js";
import { createSizeToClassNameMap as P } from "../../../utils/classNameMaps.js";
const U = P(C), Y = (N) => {
  const {
    children: u,
    className: O,
    dropdownOpened: i,
    listRef: R,
    loading: A,
    size: D = K.MD,
    ...M
  } = N, S = z(C.dropdownDesktopList, U[D], O), { activeDescendant: p, setActiveDescendant: d } = y(V), c = R || l.createRef(), v = B([]), [m, a] = b(void 0), L = l.Children.count(u), I = (e) => {
    const t = c == null ? void 0 : c.current, r = v.current[e];
    if (t && r) {
      const { top: n, bottom: o } = t.getBoundingClientRect(), { top: s, bottom: E } = r.getBoundingClientRect(), T = 4;
      n > s ? t.scrollTop = t.scrollTop - n + s - T : E > o && (t.scrollTop = t.scrollTop + E - o + T);
    }
  }, g = () => {
    const e = c == null ? void 0 : c.current;
    e && (e.scrollTop = 0);
  };
  f(() => {
    if (!i)
      return;
    const e = (t) => {
      const { keyCode: r } = t, n = L, o = m;
      let s;
      r === w.ARROW_DOWN ? (o !== void 0 && o < n - 1 ? s = o + 1 : s = 0, t.preventDefault()) : r === w.ARROW_UP && (o !== void 0 && o > 0 ? s = o - 1 : s = n - 1, t.preventDefault()), s !== void 0 && o !== s && (I(s), a(s));
    };
    return document.addEventListener("keydown", e), () => {
      document.removeEventListener("keydown", e);
    };
  }, [i, m, L]), f(() => {
    if (!i)
      return;
    let e = !1;
    l.Children.forEach(u, (t, r) => {
      l.isValidElement(t) && t.props.selected && (e = !0, I(r), a(r));
    }), e || (a(0), g());
  }, [i]), f(() => {
    var e;
    if (i && m !== void 0) {
      d((e = v.current[m]) == null ? void 0 : e.id);
      return;
    }
    !i && p !== void 0 && d();
  }, [i, m, p, d]), f(() => () => {
    p !== void 0 && d();
  }, [p, d]);
  const k = l.Children.map(u, (e, t) => l.isValidElement(e) ? l.cloneElement(e, {
    active: m === t,
    onMouseOver: (r) => {
      var n, o;
      a(t), (o = (n = e.props).onMouseOver) == null || o.call(n, r);
    },
    onMouseOut: (r) => {
      var n, o;
      a(void 0), (o = (n = e.props).onMouseOut) == null || o.call(n, r);
    },
    ref: (r) => {
      v.current[t] = r;
    }
  }) : e);
  return /* @__PURE__ */ _("div", { className: S, role: "listbox", ...M, ref: c, children: [
    k,
    A && /* @__PURE__ */ h(x, { id: "dropdown-desktop-list-loader-item", children: /* @__PURE__ */ h(W, { className: C.dropdownDesktopListLoader, theme: j.BRAND, size: D }) })
  ] });
};
Y.Item = x;
export {
  Y as DropdownList
};
