// @sberbusiness/triplex-next/components/Dropdown/desktop/DropdownDesktop.js
import { jsx as B } from "react/jsx-runtime";
import G, { useState as M, useRef as A, useCallback as i, useEffect as y } from "react";
import { c as F } from "../../../chunks/vendor-Ba_pyhza.js";
import { EDropdownAlignment as a, EDropdownDirection as D } from "../enums.js";
import { isKey as d } from "../../../utils/keyboard.js";
import { useToken as I } from "../../ThemeProvider/useToken.js";
import { EComponentSize as j } from "../../../enums/EComponentSize.js";
const K = "dropdownDesktop__836aeeca", V = "dropdownDesktopOverflowHidden__001366c0", O = {
  dropdownDesktop: K,
  "dropdownDesktop-sm": "dropdownDesktop-sm__ccdecf4e",
  "dropdownDesktop-md": "dropdownDesktop-md__ae42ef15",
  "dropdownDesktop-lg": "dropdownDesktop-lg__34fabc97",
  dropdownDesktopOverflowHidden: V
}, b = O.dropdownDesktopOverflowHidden, $ = G.forwardRef((P, p) => {
  const {
    alignment: m = a.AUTO,
    children: g,
    fixedWidth: u,
    className: C,
    direction: w = D.AUTO,
    opened: f,
    setOpened: q,
    style: N,
    targetRef: _,
    size: W = j.MD,
    ...R
  } = P, { scopeClassName: S } = I(), [x, H] = M({ opacity: 0 }), s = A(null), h = A({ height: 0, width: 0 }), U = F(O.dropdownDesktop, S, C, O[`dropdownDesktop-${W}`]), k = i(
    (e) => {
      var o;
      (o = s.current) != null && o.contains(e.target) || e.preventDefault();
    },
    [s]
  ), E = i((e) => {
    if (e.target === document.body) {
      const o = e.code || e.keyCode;
      (d(o, "SPACE") || d(o, "PAGE_UP") || d(o, "PAGE_DOWN") || d(o, "END") || d(o, "HOME") || d(o, "ARROW_LEFT") || d(o, "ARROW_UP") || d(o, "ARROW_RIGHT") || d(o, "ARROW_DOWN")) && e.preventDefault();
    }
  }, []), v = i(
    /** Запрет скролла всей страницы. */
    (e) => {
      e ? (document.addEventListener("wheel", k, { passive: !1 }), document.addEventListener("keydown", E)) : (document.removeEventListener("wheel", k), document.removeEventListener("keydown", E));
    },
    [k, E]
  ), L = i(
    (e, o, t) => {
      const n = u ? t.width : Math.max(t.width, o.width);
      m === a.AUTO ? t.right - n > 0 ? e.left = t.right - n : t.left + n < document.documentElement.clientWidth ? e.left = t.left : e.left = t.right - n : m === a.RIGHT ? e.left = t.right - n : m === a.LEFT && (e.left = t.left);
    },
    [m, u]
  ), T = i(
    (e, o, t) => {
      w === D.AUTO ? t.bottom + 4 + o.height < document.documentElement.clientHeight ? e.top = t.bottom + 4 : t.top - 4 - o.height > 0 ? e.bottom = document.documentElement.clientHeight - t.top + 4 : e.top = t.bottom + 4 : w === D.BOTTOM ? e.top = t.bottom + 4 : w === D.TOP && (e.bottom = document.documentElement.clientHeight - t.top + 4);
    },
    [w]
  ), r = i(() => {
    const { current: e } = s, { current: o } = _;
    if (e && o) {
      const t = e.getBoundingClientRect(), n = o.getBoundingClientRect(), c = {};
      u ? c.width = n.width : c.minWidth = n.width, T(c, t, n), L(c, t, n), h.current = { height: t.height, width: t.width }, H(c);
    }
  }, [_, u, T, L]);
  y(() => {
    f ? r() : (h.current = { height: 0, width: 0 }, H({ opacity: 0 }));
  }, [f, r]), y(() => {
    if (s.current) {
      const { width: e, height: o } = s.current.getBoundingClientRect();
      (e != h.current.width || o != h.current.height) && r();
    }
  }, [g, r]);
  const l = i(() => {
    setTimeout(r);
  }, [r]);
  y(() => {
    if (f)
      return document.addEventListener("scroll", l, !0), window.addEventListener("resize", l), v(!0), document.body.classList.add(b), l(), () => {
        document.removeEventListener("scroll", l, !0), window.removeEventListener("resize", l), v(!1), document.body.classList.remove(b);
      };
  }, [f, l, v]);
  const z = (e) => {
    s.current = e, typeof p == "function" ? p(e) : p && (p.current = e);
  };
  return f ? /* @__PURE__ */ B(
    "div",
    {
      className: U,
      style: { ...x, ...N },
      ...R,
      "data-tx": "1.4.0",
      ref: z,
      children: g
    }
  ) : null;
});
$.displayName = "DropdownDesktop";
export {
  $ as DropdownDesktop
};
