// @sberbusiness/triplex-next/components/DesignTokens/types/DesignTokensTypes.js
const o = [
  "100",
  "90",
  "80",
  "70",
  "60",
  "50",
  "40",
  "30",
  "20",
  "10",
  "0"
], e = [
  "100",
  "90",
  "80",
  "70",
  "60",
  "50",
  "40",
  "30",
  "20",
  "10",
  "0"
], s = [
  "100",
  "90",
  "80",
  "70",
  "60",
  "50",
  "40",
  "30",
  "20",
  "10",
  "0"
], r = [
  "100",
  "90",
  "80",
  "70",
  "60",
  "50",
  "40",
  "30",
  "20",
  "10",
  "0"
], n = [
  "100",
  "90",
  "80",
  "70",
  "60",
  "50",
  "40",
  "30",
  "20",
  "10",
  "0"
], C = [
  "100",
  "90",
  "80",
  "70",
  "60",
  "50",
  "40",
  "30",
  "20",
  "10",
  "0"
], l = [
  "100",
  "90",
  "80",
  "70",
  "60",
  "50",
  "40",
  "30",
  "20",
  "10",
  "0"
], t = [
  "100",
  "90",
  "80",
  "70",
  "60",
  "50",
  "40",
  "30",
  "20",
  "10",
  "0"
], u = [
  "100",
  "90",
  "80",
  "70",
  "60",
  "50",
  "40",
  "30",
  "20",
  "10",
  "0"
], p = [
  "100",
  "90",
  "80",
  "70",
  "60",
  "50",
  "40",
  "30",
  "20",
  "10",
  "0"
];
export {
  o as designTokensCoreGroupColorBrandKeys,
  e as designTokensCoreGroupColorDarkNeutralAlphaKeys,
  s as designTokensCoreGroupColorDarkNeutralKeys,
  r as designTokensCoreGroupColorErrorKeys,
  n as designTokensCoreGroupColorInfoKeys,
  C as designTokensCoreGroupColorNeutralAlphaKeys,
  l as designTokensCoreGroupColorNeutralKeys,
  t as designTokensCoreGroupColorSuccessKeys,
  u as designTokensCoreGroupColorSystemKeys,
  p as designTokensCoreGroupColorWarningKeys
};
