// @sberbusiness/triplex-next/components/Pagination/components/PaginationNavigation.js
import { jsxs as _, jsx as a } from "react/jsx-runtime";
import B from "react";
import { EPaginationNavigationIconDirection as r } from "../enums.js";
import { PaginationNavigationButton as s } from "./PaginationNavigationButton.js";
import { PaginationUtils as j, PAGINATION_ELLIPSIS_VALUE as v } from "../utils/paginationUtils.js";
import { PaginationPageEllipsis as w } from "./PaginationPageEllipsis.js";
import { PaginationPageButton as y } from "./PaginationPageButton.js";
import { PaginationNavigationExtended as R } from "./PaginationNavigationExtended.js";
import { PaginationNavigationExtendedItem as l } from "./PaginationNavigationExtendedItem.js";
const S = B.forwardRef(
  ({
    buttonNextProps: o,
    buttonPrevProps: d,
    currentPage: n,
    onCurrentPageChange: t,
    totalPages: m,
    boundaryCount: e = 0,
    siblingCount: f = 0,
    ...g
  }, h) => {
    const E = j.createPagesArray({
      boundaryCount: e,
      currentPage: n,
      siblingCount: f,
      totalPages: m
    }), N = n === 1, k = n === m, I = () => {
      t(n - 1);
    }, A = (i) => () => {
      t(i);
    }, C = () => {
      t(n + 1);
    }, L = (i, c) => {
      switch (i) {
        case v:
          return /* @__PURE__ */ a(l, { children: /* @__PURE__ */ a(w, { children: "..." }) }, c);
        default:
          return /* @__PURE__ */ a(l, { children: /* @__PURE__ */ a(y, { isCurrent: i === n, onClick: A(i), children: i }) }, c);
      }
    };
    return /* @__PURE__ */ _(R, { ...g, ref: h, children: [
      /* @__PURE__ */ a(l, { children: /* @__PURE__ */ a(
        s,
        {
          direction: r.BACK,
          ...d,
          onClick: I,
          disabled: N || (d == null ? void 0 : d.disabled)
        }
      ) }),
      E.map((i, c) => L(i, c)),
      /* @__PURE__ */ a(l, { children: /* @__PURE__ */ a(
        s,
        {
          direction: r.NEXT,
          ...o,
          onClick: C,
          disabled: k || (o == null ? void 0 : o.disabled)
        }
      ) })
    ] });
  }
);
S.displayName = "PaginationNavigation";
export {
  S as PaginationNavigation
};
