// @sberbusiness/triplex-next/components/SMSInput/components/SMSInputRefresh.js
import { jsx as p } from "react/jsx-runtime";
import { c as u } from "../../../chunks/vendor-Ba_pyhza.js";
import { forwardRef as x, useContext as M, useMemo as N } from "react";
import "../../Button/Button.js";
import "../../Button/ButtonBase.js";
import { ButtonIcon as y } from "../../Button/ButtonIcon.js";
import "../../Button/ButtonDropdown.js";
import "../../Button/ButtonDropdownExtended.js";
import { EButtonIconShape as B } from "../../Button/enums.js";
import { SMSInputContext as z } from "../SMSInputContext.js";
import { RefreshIcon as D } from "./RefreshIcon.js";
import { s as m } from "../../../chunks/SMSInput.module-Ci4GRnk_.js";
const E = x(
  ({ className: i, disabled: r, countdownTime: s, countdownTimeLeft: e, onClick: o, onRefresh: l, ...n }, f) => {
    const { disabled: t, error: c, size: h, sizeClassName: I, tooltipId: S } = M(z), d = e > 0, a = t && !c || r || d, b = u(m.btnRefresh, I, i, {
      [m.disabled]: a
    }), C = N(() => e >= s || t || r ? 0 : (s - e) / s, [s, e, t, r]);
    return /* @__PURE__ */ p(
      y,
      {
        "aria-describedby": S,
        className: b,
        disabled: a,
        onClick: (R) => {
          l(), o == null || o(R);
        },
        ref: f,
        shape: B.CIRCLE,
        ...n,
        children: /* @__PURE__ */ p(D, { percent: C || 0, size: h, disabled: !!(t || r) })
      }
    );
  }
);
E.displayName = "SMSInputRefresh";
export {
  E as SMSInputRefresh
};
