// @sberbusiness/triplex-next/components/SMSInput/components/SMSInputInput.js
import { jsxs as d, jsx as r, Fragment as g } from "react/jsx-runtime";
import { c as _ } from "../../../chunks/vendor-Ba_pyhza.js";
import v, { useContext as M } from "react";
import { FormField as T } from "../../FormField/FormField.js";
import { FormFieldDescription as j } from "../../FormField/components/FormFieldDescription.js";
import { FormFieldCounter as y } from "../../FormField/components/FormFieldCounter.js";
import "../../FormField/components/FormFieldLabel.js";
import "../../FormField/components/FormFieldClear.js";
import "../../FormField/components/FormFieldTarget.js";
import { F as z } from "../../../chunks/FormFieldInput-CO7yZP0Q.js";
import "../../FormField/components/FormFieldMaskedInput.js";
import "../../FormField/components/FormFieldPostfix.js";
import "../../FormField/components/FormFieldPrefix.js";
import "../../FormField/components/FormFieldTextarea.js";
import { EFormFieldStatus as s } from "../../FormField/enums.js";
import { FormGroup as A } from "../../FormGroup/FormGroup.js";
import { SMSInputContext as L } from "../SMSInputContext.js";
import { s as O } from "../../../chunks/SMSInput.module-Ci4GRnk_.js";
import { EVENT_KEY_CODES as k } from "../../../utils/keyboard.js";
const w = new RegExp(/^[0-9]*$/), B = v.forwardRef(
  ({ className: u, counter: o, description: l, disabled: f, maxLength: c = 8, onChange: e, onKeyDown: m, placeholder: F, ...E }, S) => {
    const {
      code: p,
      disabled: C,
      disabledSubmit: h,
      error: x,
      onChangeCode: I,
      onSubmitCode: N,
      size: R,
      sizeClassName: a
    } = M(L), b = C || f, D = _(O.input, a, u);
    let i = s.DEFAULT;
    return b && (i = s.DISABLED), x && (i = s.ERROR), /* @__PURE__ */ d(A, { children: [
      /* @__PURE__ */ r(T, { className: a, onKeyDown: (t) => {
        t.keyCode === k.ENTER && !h && N(p), m == null || m(t);
      }, size: R, status: i, children: /* @__PURE__ */ r(
        z,
        {
          autoComplete: "off",
          className: D,
          maxLength: c,
          onChange: (t) => {
            const n = t.target.value;
            w.test(n) && (I(n), e == null || e(t));
          },
          placeholder: F,
          ref: S,
          value: p,
          ...E
        }
      ) }),
      /* @__PURE__ */ r(j, { children: l || o ? /* @__PURE__ */ d(g, { children: [
        l,
        o ? /* @__PURE__ */ r(y, { children: o }) : null
      ] }) : null })
    ] });
  }
);
B.displayName = "SMSInputInput";
export {
  B as SMSInputInput
};
