// @sberbusiness/triplex-next/components/SMSInput/components/SMSInputSubmit.js
import { jsx as i } from "react/jsx-runtime";
import { c as S } from "../../../chunks/vendor-Ba_pyhza.js";
import { useContext as f, useEffect as I } from "react";
import "../../Button/Button.js";
import "../../Button/ButtonBase.js";
import { ButtonIcon as C } from "../../Button/ButtonIcon.js";
import "../../Button/ButtonDropdown.js";
import "../../Button/ButtonDropdownExtended.js";
import { EButtonIconShape as h } from "../../Button/enums.js";
import { SMSInputContext as x } from "../SMSInputContext.js";
import { SubmitIcon as N } from "./SubmitIcon.js";
import { s as e } from "../../../chunks/SMSInput.module-Ci4GRnk_.js";
const D = ({ className: r, disabled: a, onClick: o, ...p }) => {
  const {
    code: t,
    disabled: n,
    onSubmitCode: u,
    setDisabledSubmit: s,
    size: c,
    sizeClassName: l
  } = f(x), m = n || a || t === "", b = S(e.btnSubmit, { [e.active]: !!t }, l, r);
  return I(() => {
    s(m);
  }, [m, s]), /* @__PURE__ */ i(
    C,
    {
      active: !!t,
      className: b,
      disabled: m,
      onClick: (d) => {
        u(t), o == null || o(d);
      },
      shape: h.CIRCLE,
      ...p,
      children: /* @__PURE__ */ i(N, { size: c })
    }
  );
};
D.displayName = "SMSInputSubmit";
export {
  D as SMSInputSubmit
};
