// @sberbusiness/triplex-next/components/Pagination/utils/paginationUtils.js
import { j as R } from "../../../chunks/vendor-Ba_pyhza.js";
const m = -1, h = {
  // Создание массива чисел в заданном диапазоне с заданным шагом.
  generateRange: (n, i, e = 1) => n > i ? [] : R(n, i + 1, e),
  // Создание массивов страниц в начале, конце и вокруг текущей.
  generatePageRanges: (n, i, e, s) => {
    const g = h.generateRange(1, Math.min(e, s)), a = h.generateRange(
      Math.max(s - e + 1, e + 1),
      s
    ), c = s - e - 1 - i * 2, r = Math.max(
      Math.min(n - i, c),
      e + 2
    ), S = e + 2 + i * 2, M = Math.min(
      Math.max(n + i, S),
      s - e - 1
    );
    return { boundaryDivider: 2, endPages: a, siblingsEnd: M, siblingsStart: r, startPages: g };
  },
  // Создание итогового массива страниц для отображения.
  createPagesArray: ({
    currentPage: n,
    siblingCount: i,
    boundaryCount: e,
    totalPages: s
  }) => {
    const { boundaryDivider: p, endPages: t, siblingsEnd: g, siblingsStart: a, startPages: c } = h.generatePageRanges(n, i, e, s), r = c;
    return a > e + p ? r.push(m) : e + 1 < s - e && r.push(e + 1), r.push(...h.generateRange(a, g)), g < s - e - 1 ? r.push(m) : s - e > e && r.push(s - e), r.push(...t), r;
  }
};
export {
  m as PAGINATION_ELLIPSIS_VALUE,
  h as PaginationUtils
};
