// @sberbusiness/triplex-next/components/CarouselExtended/CarouselExtended.js
import { jsxs as p, jsx as b } from "react/jsx-runtime";
import C, { useState as w, useRef as D, useCallback as L, useEffect as n } from "react";
import { l as M, c as S } from "../../chunks/vendor-Ba_pyhza.js";
import { scrollSmoothHorizontally as c } from "../../utils/scroll.js";
const R = "carouselExtended__6bcdf1d9", k = {
  carouselExtended: R
}, y = C.forwardRef(
  ({ children: l, buttonPrev: a, buttonNext: i, stepPrev: u, stepNext: f, ...h }, s) => {
    const [o, m] = w({ hidden: !0, nextDisabled: !1, prevDisabled: !0 }), r = D(null), t = L(() => {
      const e = r.current, d = {
        hidden: e.scrollWidth == e.clientWidth,
        nextDisabled: Math.round(e.scrollLeft) + e.offsetWidth >= e.scrollWidth,
        prevDisabled: Math.round(e.scrollLeft) <= 0
      };
      M.isEqual(d, o) == !1 && m(d);
    }, [o]);
    n(() => (window.addEventListener("resize", t), document.addEventListener("scroll", t), () => {
      window.removeEventListener("resize", t), document.removeEventListener("scroll", t);
    }), [t]), n(() => {
      t();
    }, [l, t]);
    const x = () => {
      c(r.current, Math.floor(-u));
    }, v = () => {
      c(r.current, Math.ceil(f));
    }, E = (e) => {
      r.current = e, typeof s == "function" ? s(e) : s && (s.current = e);
    };
    return /* @__PURE__ */ p("div", { ...h, children: [
      a({
        disabled: o.prevDisabled,
        hidden: o.hidden,
        onClick: x
      }),
      /* @__PURE__ */ b("div", { className: S(k.carouselExtended), onScroll: t, ref: E, children: l }),
      i({
        disabled: o.nextDisabled,
        hidden: o.hidden,
        onClick: v
      })
    ] });
  }
);
y.displayName = "CarouselExtended";
export {
  y as CarouselExtended
};
