// @sberbusiness/triplex-next/components/UploadZone/UploadZone.js
var m = Object.defineProperty;
var u = (s, n, e) => n in s ? m(s, n, { enumerable: !0, configurable: !0, writable: !0, value: e }) : s[n] = e;
var t = (s, n, e) => u(s, typeof n != "symbol" ? n + "" : n, e);
import { jsx as g, jsxs as f } from "react/jsx-runtime";
import Z, { createElement as l } from "react";
import { createRoot as C } from "react-dom/client";
import { c as h } from "../../chunks/vendor-Ba_pyhza.js";
import { U as L, s as d } from "../../chunks/UploadZoneInput-pdhICY56.js";
import { UploadZoneContext as U } from "./UploadZoneContext.js";
class c extends Z.PureComponent {
  constructor() {
    super(...arguments);
    // Описание - https://stackoverflow.com/questions/7110353/html5-dragleave-fired-when-hovering-a-child-element.
    // Если counter > 0 - означает, что перетаскиваемый объект в пределах окна браузера.
    /** Каунтер для подсчёта drag-перемещений по странице. */
    t(this, "counter", 0);
    /** Элемент-обёртка для дроп-зоны. */
    t(this, "dropZoneWrapperDiv", null);
    t(this, "state", {
      hoverOnDrag: !1,
      inputNode: void 0
    });
    t(this, "addListeners", (e) => {
      e && (e.addEventListener("dragenter", this.handleDragEnter), e.addEventListener("dragleave", this.handleDragLeave));
    });
    t(this, "removeListeners", (e) => {
      e && (e.removeEventListener("dragenter", this.handleDragEnter), e.removeEventListener("dragleave", this.handleDragLeave));
    });
    t(this, "createDropZoneDiv", () => {
      const { children: e, className: r, onChange: o, renderContainerContent: a, dropZoneContainer: D, ...p } = this.props, i = document.createElement("div");
      return C(i).render(
        /* @__PURE__ */ l(
          "div",
          {
            className: h(d.uploadZoneContainerDragArea, r),
            onDragOver: this.handlePreventDefault,
            onDrop: this.fileDrop,
            ...p,
            key: "uploadZoneDragArea",
            role: "none"
          },
          a == null ? void 0 : a()
        )
      ), i;
    });
    t(this, "handleAreaClick", (e) => {
      e.stopPropagation(), this.openUploadDialog();
    });
    t(this, "openUploadDialog", () => {
      const { inputNode: e } = this.state;
      e == null || e.click();
    });
    t(this, "handlePreventDefault", (e) => {
      const { onDragOver: r } = this.props;
      e.preventDefault(), r == null || r(e);
    });
    t(this, "handleDragEnter", () => {
      this.counter++, this.counter === 1 && this.setState({ hoverOnDrag: !0 });
    });
    t(this, "handleDragLeave", () => {
      this.counter--, this.counter === 0 && this.setState({ hoverOnDrag: !1 });
    });
    t(this, "fileDrop", (e) => {
      const { onDrop: r, onChange: o } = this.props;
      e.preventDefault(), r == null || r(e), o(e.dataTransfer.files, e), this.setState({ hoverOnDrag: !1 }), this.counter = 0;
    });
  }
  componentDidMount() {
    this.addListeners(this.props.dropZoneContainer);
  }
  componentDidUpdate(e, r) {
    const { dropZoneContainer: o } = this.props, { hoverOnDrag: a } = this.state;
    o !== e.dropZoneContainer && (this.removeListeners(e.dropZoneContainer), this.addListeners(o)), a !== r.hoverOnDrag && o && (a ? (this.dropZoneWrapperDiv = this.createDropZoneDiv(), o.appendChild(this.dropZoneWrapperDiv)) : this.dropZoneWrapperDiv && o.removeChild(this.dropZoneWrapperDiv));
  }
  componentWillUnmount() {
    this.removeListeners(this.props.dropZoneContainer);
  }
  render() {
    const { children: e, className: r, onChange: o, renderContainerContent: a, dropZoneContainer: D, ...p } = this.props, { inputNode: i } = this.state;
    return /* @__PURE__ */ g(
      U.Provider,
      {
        value: {
          inputNode: i,
          onChange: o,
          openUploadDialog: this.openUploadDialog,
          setInputNode: (v) => {
            this.setState({ inputNode: v });
          }
        },
        children: /* @__PURE__ */ f("div", { className: d.uploadZone, "data-tx": "1.4.0", children: [
          /* @__PURE__ */ l(
            "div",
            {
              className: h(d.uploadZoneDragArea, r),
              onClick: this.handleAreaClick,
              ...p,
              key: "uploadZoneDragArea",
              role: "none"
            }
          ),
          e({ openUploadDialog: this.openUploadDialog })
        ] })
      }
    );
  }
}
t(c, "displayName", "UploadZone"), t(c, "Input", L);
export {
  c as UploadZone
};
