// @sberbusiness/triplex-next/components/Alert/AlertContext/AlertContext.js
import { jsxs as T, jsx as t } from "react/jsx-runtime";
import x from "react";
import { EAlertType as e } from "../EAlertType.js";
import { alertTypeToClassNameMap as m } from "../AlertTypeUtils.js";
import { Text as y } from "../../Typography/Text.js";
import { EFontType as r, ETextSize as I } from "../../Typography/enums.js";
import { SystemStrokeStsIcon16 as f, ErrorStrokeStsIcon16 as S, WarningStrokeStsIcon16 as i, InfoStrokeStsIcon16 as R } from "@sberbusiness/icons-next";
import { c as N } from "../../../chunks/vendor-Ba_pyhza.js";
const _ = "alertContext__acd8f089", E = "alertContextText__bcc2c646", d = "alertTypeError__ced7c27f", C = "alertTypeInfo__3002ea3b", u = "alertTypeWarning__732e3a96", A = "alertTypeSystem__2c54c9e2", n = {
  alertContext: _,
  alertContextText: E,
  alertTypeError: d,
  alertTypeInfo: C,
  alertTypeWarning: u,
  alertTypeSystem: A
}, O = (o) => {
  switch (o) {
    case e.INFO:
      return /* @__PURE__ */ t(R, { paletteIndex: 3 });
    case e.WARNING:
      return /* @__PURE__ */ t(i, { paletteIndex: 2 });
    case e.ERROR:
      return /* @__PURE__ */ t(S, { paletteIndex: 1 });
    case e.SYSTEM:
      return /* @__PURE__ */ t(f, { paletteIndex: 4 });
  }
}, W = {
  [e.INFO]: r.INFO,
  [e.WARNING]: r.WARNING,
  [e.ERROR]: r.ERROR,
  [e.SYSTEM]: r.SECONDARY
}, F = x.forwardRef(
  ({ children: o, className: l, type: a, renderIcon: s, ...c }, p) => /* @__PURE__ */ T(
    "span",
    {
      role: "alert",
      className: N(n.alertContext, m[a](n), l),
      ...c,
      "data-tx": "1.4.0",
      ref: p,
      children: [
        s || O(a),
        /* @__PURE__ */ t(y, { size: I.B4, type: W[a], className: n.alertContextText, children: o })
      ]
    }
  )
);
F.displayName = "AlertContext";
export {
  F as AlertContext
};
