// @sberbusiness/triplex-next/components/LightBox/LightBoxControls/LightBoxClose.js
import { jsxs as n, jsx as e, Fragment as f } from "react/jsx-runtime";
import { useRef as h } from "react";
import { c as u } from "../../../chunks/vendor-Ba_pyhza.js";
import { CrossStrokeSrvIcon32 as x, CrossStrokeSrvIcon20 as C } from "@sberbusiness/icons-next";
import { TriggerClickOnKeyDownEvent as E } from "../../Triggers/TriggerClickOnKeyDownEvent.js";
import { EVENT_KEY_CODES as g } from "../../../utils/keyboard.js";
import { EButtonTheme as i } from "../../Button/enums.js";
import { Button as l } from "../../Button/Button.js";
import { EComponentSize as a } from "../../../enums/EComponentSize.js";
import { s as o } from "../../../chunks/LightBoxControls.module-42gn-co3.js";
const N = ({
  className: c,
  onClick: t,
  title: r = "Закрыть",
  ...d
}) => {
  const s = h(null), m = (p) => /* @__PURE__ */ n(f, { children: [
    /* @__PURE__ */ e(
      l,
      {
        className: o.lightBoxControlsDesktop,
        onClick: t,
        title: r,
        "data-exclude-modal-focus": !0,
        ref: p,
        icon: /* @__PURE__ */ e(x, { paletteIndex: 0 }),
        size: a.LG,
        theme: i.SECONDARY_LIGHT
      }
    ),
    /* @__PURE__ */ e(
      l,
      {
        className: o.lightBoxControlsMobile,
        onClick: t,
        title: r,
        "data-exclude-modal-focus": !0,
        icon: /* @__PURE__ */ e(C, { paletteIndex: 0 }),
        size: a.MD,
        theme: i.SECONDARY_LIGHT
      }
    )
  ] });
  return /* @__PURE__ */ n("div", { className: u(c, o.lightBoxClose), ...d, children: [
    /* @__PURE__ */ e("span", { className: o.withKeyboardEvent, children: /* @__PURE__ */ e(E, { eventKeyCode: g.ESCAPE, targetRef: s, children: m(s) }) }),
    /* @__PURE__ */ e("span", { className: o.withoutKeyboardEvent, children: m() })
  ] });
};
N.displayName = "LightBoxClose";
export {
  N as LightBoxClose
};
