// @sberbusiness/triplex-next/components/LightBox/LightBoxControls/LightBoxNext.js
import { jsx as e, jsxs as x, Fragment as N } from "react/jsx-runtime";
import { useRef as m } from "react";
import { c as n } from "../../../chunks/vendor-Ba_pyhza.js";
import { CaretrightStrokeSrvIcon32 as E, CaretrightStrokeSrvIcon20 as D } from "@sberbusiness/icons-next";
import { TriggerClickOnKeyDownEvent as I } from "../../Triggers/TriggerClickOnKeyDownEvent.js";
import { EVENT_KEY_CODES as R } from "../../../utils/keyboard.js";
import { Button as c } from "../../Button/Button.js";
import { EButtonTheme as s } from "../../Button/enums.js";
import { EComponentSize as f } from "../../../enums/EComponentSize.js";
import { s as o } from "../../../chunks/LightBoxControls.module-42gn-co3.js";
const v = ({
  className: g,
  clickByArrowRight: u,
  dataTutorialId: d,
  onClick: r,
  title: l,
  ...B
}) => {
  const h = m(null), i = (t) => /* @__PURE__ */ x(N, { children: [
    /* @__PURE__ */ e(
      c,
      {
        className: n(o.lightBoxControlsDesktop, o.lightBoxNextButton),
        "data-test-id": t != null && t.addDataTestId ? "lightBox-next" : void 0,
        "data-tutorial-id": d,
        onClick: r,
        title: l,
        ref: t == null ? void 0 : t.buttonRef,
        icon: /* @__PURE__ */ e(E, { paletteIndex: 0 }),
        size: f.LG,
        theme: s.SECONDARY_LIGHT
      }
    ),
    /* @__PURE__ */ e(
      c,
      {
        className: n(o.lightBoxControlsMobile, o.lightBoxNextButton),
        "data-test-id": t != null && t.addDataTestId ? "lightBox-next" : void 0,
        "data-tutorial-id": d,
        onClick: r,
        title: l,
        ref: t == null ? void 0 : t.buttonRef,
        icon: /* @__PURE__ */ e(D, { paletteIndex: 0 }),
        size: f.MD,
        theme: s.SECONDARY_LIGHT
      }
    )
  ] });
  return /* @__PURE__ */ e("div", { className: n(g, o.lightBoxNext), ...B, children: u ? /* @__PURE__ */ x("span", { children: [
    /* @__PURE__ */ e("span", { className: o.withKeyboardEvent, children: /* @__PURE__ */ e(I, { targetRef: h, eventKeyCode: R.ARROW_RIGHT, children: i({ addDataTestId: !0, buttonRef: h }) }) }),
    /* @__PURE__ */ e("span", { className: o.withoutKeyboardEvent, children: i() })
  ] }) : i({ addDataTestId: !0 }) });
};
v.displayName = "LightBoxNext";
export {
  v as LightBoxNext
};
