// @sberbusiness/triplex-next/components/LightBox/LightBoxControls/LightBoxPrev.js
import { jsx as e, jsxs as f, Fragment as g } from "react/jsx-runtime";
import { useRef as m } from "react";
import { c as n } from "../../../chunks/vendor-Ba_pyhza.js";
import { CaretleftStrokeSrvIcon32 as E, CaretleftStrokeSrvIcon20 as D } from "@sberbusiness/icons-next";
import { EButtonTheme as h } from "../../Button/enums.js";
import { TriggerClickOnKeyDownEvent as C } from "../../Triggers/TriggerClickOnKeyDownEvent.js";
import { EVENT_KEY_CODES as I } from "../../../utils/keyboard.js";
import { Button as s } from "../../Button/Button.js";
import { EComponentSize as v } from "../../../enums/EComponentSize.js";
import { s as o } from "../../../chunks/LightBoxControls.module-42gn-co3.js";
const N = ({
  className: u,
  clickByArrowLeft: x,
  dataTutorialId: r,
  onClick: d,
  title: l,
  ...B
}) => {
  const c = m(null), i = (t) => /* @__PURE__ */ f(g, { children: [
    /* @__PURE__ */ e(
      s,
      {
        className: n(o.lightBoxControlsDesktop, o.lightBoxPrevButton),
        "data-test-id": t != null && t.addDataTestId ? "lightBox-prev" : void 0,
        "data-tutorial-id": r,
        onClick: d,
        title: l,
        ref: t == null ? void 0 : t.buttonRef,
        icon: /* @__PURE__ */ e(E, { paletteIndex: 0 }),
        size: v.LG,
        theme: h.SECONDARY_LIGHT
      }
    ),
    /* @__PURE__ */ e(
      s,
      {
        className: n(o.lightBoxControlsMobile, o.lightBoxPrevButton),
        "data-test-id": t != null && t.addDataTestId ? "lightBox-prev" : void 0,
        "data-tutorial-id": r,
        onClick: d,
        title: l,
        ref: t == null ? void 0 : t.buttonRef,
        icon: /* @__PURE__ */ e(D, { paletteIndex: 0 }),
        size: v.MD,
        theme: h.SECONDARY_LIGHT
      }
    )
  ] });
  return /* @__PURE__ */ e("div", { className: n(u, o.lightBoxPrev), ...B, children: x ? /* @__PURE__ */ f("span", { children: [
    /* @__PURE__ */ e("span", { className: o.withKeyboardEvent, children: /* @__PURE__ */ e(C, { targetRef: c, eventKeyCode: I.ARROW_LEFT, children: i({ addDataTestId: !0, buttonRef: c }) }) }),
    /* @__PURE__ */ e("span", { className: o.withoutKeyboardEvent, children: i() })
  ] }) : i({ addDataTestId: !0 }) });
};
N.displayName = "LightBoxPrev";
export {
  N as LightBoxPrev
};
