// @sberbusiness/triplex-next/components/Chip/ChipSelect.js
import { jsx as t } from "react/jsx-runtime";
import w from "react";
import { SelectExtendedField as D } from "../SelectExtendedField/SelectExtendedField.js";
import "../SelectExtendedField/components/SelectExtendedFieldDropdown.js";
import "../SelectExtendedField/components/SelectExtendedFieldTarget.js";
import { ChipClearButton as S } from "./ChipClearButton.js";
import { s as b, C as k } from "../../chunks/Chip-G6cGXuGj.js";
import { ChipDropdownArrow as u } from "./ChipDropdownArrow.js";
import { SelectExtendedFieldDropdownDefault as B } from "../SelectExtendedField/components/SelectExtendedFieldDropdownDefault.js";
import { c as g } from "../../chunks/vendor-Ba_pyhza.js";
const y = w.forwardRef(
  ({ className: p, clearSelected: m, disabled: d, size: e, label: i, displayedValue: l, onChange: n, options: c, value: r, ...a }, s) => {
    const f = (o) => {
      o.stopPropagation(), m();
    }, C = ({ opened: o, setOpened: x }) => /* @__PURE__ */ t(
      k,
      {
        "aria-expanded": o,
        disabled: d,
        size: e,
        onClick: () => x(!0),
        postfix: r ? /* @__PURE__ */ t(S, { onClick: f }) : /* @__PURE__ */ t(u, { rotated: o }),
        role: "combobox",
        ref: s,
        selected: !!r,
        children: r ? l ?? r.label : i
      }
    ), h = (o) => /* @__PURE__ */ t(
      B,
      {
        ...o,
        mobileTitle: i,
        size: e,
        onChange: n,
        options: c,
        value: r,
        fixedWidth: !1
      }
    );
    return /* @__PURE__ */ t(
      D,
      {
        className: g(b.chipGroupItem, p),
        renderTarget: C,
        ...a,
        children: h
      }
    );
  }
);
y.displayName = "ChipSelect";
export {
  y as ChipSelect
};
