// @sberbusiness/triplex-next/components/FormField/components/FormFieldLabel.js
import { jsx as d } from "react/jsx-runtime";
import p, { useContext as T, useState as g, useEffect as E } from "react";
import { FormFieldContext as h } from "../FormFieldContext.js";
import { TARGET_PADDING_X_DEFAULT as m } from "../consts.js";
import { m as D, c as N } from "../../../chunks/vendor-Ba_pyhza.js";
import { EFormFieldStatus as y } from "../enums.js";
const A = "formFieldLabel__cc7aa592", R = "floating__2a4fa279", S = "formFieldLabelText__09fde7f9", C = "disabled__b720611e", s = {
  formFieldLabel: A,
  "size-sm": "size-sm__7540e812",
  "size-md": "size-md__c6a543db",
  "size-lg": "size-lg__1760a0c4",
  floating: R,
  formFieldLabelText: S,
  disabled: C
}, G = p.forwardRef(
  ({ children: f, className: r, style: c, floating: e, ...n }, F) => {
    const { status: _, focused: t, id: b, prefixWidth: L, postfixWidth: x, size: i, valueExist: a } = T(h), [l, o] = g(!1);
    E(() => {
      !D(e) && e !== l ? o(e) : e !== l && o(t || a);
    }, [t, a, e, l]);
    const u = N(
      s.formFieldLabel,
      {
        [s.disabled]: _ === y.DISABLED,
        [s.floating]: l,
        [s[`size-${i}`]]: i
      },
      r
    ), z = {
      // Левая позиция элемента. Когда label по-середине инпута, позиция учитывает иконки по краям, когда сверху, позиция на все ширину поля ввода.
      left: L || m,
      // Правая позиция элемента. Когда label по-середине инпута, позиция учитывает иконки по краям, когда сверху, позиция на все ширину поля ввода.
      right: x || m,
      ...c
    };
    return /* @__PURE__ */ d("label", { className: u, ref: F, htmlFor: b, ...n, style: z, children: /* @__PURE__ */ d("span", { className: s.formFieldLabelText, children: f }) });
  }
);
G.displayName = "FormFieldLabel";
export {
  G as FormFieldLabel
};
