// @sberbusiness/triplex-next/components/FormField/components/FormFieldMaskedInput.js
import { jsxs as N, jsx as m } from "react/jsx-runtime";
import { useState as O, useRef as A, useContext as B, useEffect as D, useCallback as U } from "react";
import { c as F, n as q, o as p } from "../../../chunks/vendor-Ba_pyhza.js";
import { presets as G } from "./FormFieldMaskedInputPresets.js";
import { s as H, F as J } from "../../../chunks/FormFieldInput-CO7yZP0Q.js";
import { FormFieldContext as K } from "../FormFieldContext.js";
const L = "formFieldMaskedInputWrapper__a2c0a493", Q = "formFieldMaskedInputPlaceholder__b0621a6f", X = "formFieldMaskedInput__8f7dcad2", l = {
  formFieldMaskedInputWrapper: L,
  formFieldMaskedInputPlaceholder: Q,
  "size-sm": "size-sm__7540e812",
  "size-md": "size-md__c6a543db",
  "size-lg": "size-lg__1760a0c4",
  formFieldMaskedInput: X
}, a = ({
  className: I,
  forwardedRef: f,
  disabled: g,
  mask: o,
  onChange: d,
  placeholder: M,
  placeholderChar: i = "0",
  placeholderMask: u,
  value: s,
  ..._
}) => {
  const [V, x] = O(""), c = A(!1), { valueExist: y, focused: z, size: k } = B(K);
  D(() => {
    x((() => {
      let t = [];
      if (s) {
        const { conformedValue: n } = p.conformToMask(s.toString(), o, { guide: !0, placeholderChar: i });
        for (let r = 0; r < o.length; r++)
          typeof o[r] == "string" ? t[r] = n[r] : n[r] === i && !s.toString()[r] ? t[r] = (u == null ? void 0 : u[r]) || i : t[r] = n[r];
      } else if (u)
        t = u.split("");
      else {
        const { conformedValue: n } = p.conformToMask("", o, { guide: !0, placeholderChar: i });
        t = n.split("");
      }
      return t.join("");
    })());
  }, [s, o, i, u]);
  const T = U(
    (e) => {
      const { value: t } = e.target;
      c.current = !1, s !== t && (d == null || d(e));
    },
    [s, d]
  ), E = () => {
    c.current = !0;
  }, W = (e, t) => o === a.presets.masks.phone ? e.length ? P(t.rawValue) : e : o === a.presets.masks.swiftCode ? e.toUpperCase() : e, P = (e) => {
    let t = [];
    if (c.current) {
      let n = /^[78]((\D*\d)*)/;
      e = e.replace(n, "+7 ($1"), n = /^\d7/, e = e.replace(n, (r) => (t = Array.from("+7 (").map((Y, $) => $), `+7 (${r}`));
    } else (e === "7" || e === "8") && (e = "+7 (");
    return {
      indexesOfPipedChars: t,
      value: p.conformToMask(e, o, { guide: !1, placeholderChar: i }).conformedValue
    };
  }, b = () => o === a.presets.masks.phone ? (s = P(s).value, s) : p.conformToMask(s, o, { guide: !1, placeholderChar: i }).conformedValue, j = (e) => (t) => {
    t && e(t), typeof f == "function" ? f(t) : f && (f.current = t);
  }, S = () => !y && !z || !s && M ? "" : V;
  return /* @__PURE__ */ N("div", { className: F(l.formFieldMaskedInputWrapper, l[`size-${k}`]), children: [
    /* @__PURE__ */ m(
      "input",
      {
        className: F(
          H.formFieldInput,
          l.formFieldMaskedInputPlaceholder,
          I
        ),
        disabled: g,
        placeholder: S(),
        readOnly: !0,
        "aria-hidden": "true",
        tabIndex: -1,
        type: "text"
      }
    ),
    /* @__PURE__ */ m(
      q,
      {
        className: F(l.formFieldMaskedInput, { [l.error]: !!_.error }, I),
        defaultValue: "",
        disabled: g,
        guide: !1,
        render: (e, t) => /* @__PURE__ */ m(J, { ...t, value: s, placeholder: M || "", ref: j(e) }),
        mask: o,
        onChange: T,
        onPaste: E,
        placeholderChar: i,
        value: b(),
        pipe: W,
        type: "text",
        ..._
      }
    )
  ] });
};
a.presets = G;
export {
  a as FormFieldMaskedInput
};
