// @sberbusiness/triplex-next/components/FormField/components/FormFieldTarget.js
import { jsx as I } from "react/jsx-runtime";
import x, { useContext as E, useRef as T, useEffect as m } from "react";
import { FormFieldContext as B } from "../FormFieldContext.js";
import { c as S, a as A } from "../../../chunks/vendor-Ba_pyhza.js";
import { EFormFieldStatus as d } from "../enums.js";
const L = "formFieldTarget__8ca0e86d", N = "placeholder__58ee63a8", R = "disabled__b720611e", o = {
  formFieldTarget: L,
  placeholder: N,
  disabled: R,
  "size-sm": "size-sm__7540e812",
  "size-md": "size-md__c6a543db",
  "size-lg": "size-lg__1760a0c4"
}, y = x.forwardRef((F, _) => {
  const { className: p, id: e, onBlur: l, onFocus: i, placeholder: n, children: s, ...g } = F, { status: r, setFocused: f, setId: t, setValueExist: u, size: h } = E(B), b = S(o.formFieldTarget, p, o[`size-${h}`], {
    [o.disabled]: r === d.DISABLED,
    [o.placeholder]: !!n && !s && r !== d.DISABLED
  }), a = T(e || A("formFieldTarget_"));
  m(() => {
    t(a.current);
  }, [t]), m(() => {
    e && (a.current = e, t(a.current));
  }, [e, t]), m(() => {
    u(!!s);
  }, [u, s]);
  const z = (c) => {
    f(!1), l == null || l(c);
  }, D = (c) => {
    f(!0), i == null || i(c);
  };
  return /* @__PURE__ */ I(
    "div",
    {
      ...g,
      className: b,
      "aria-disabled": r === d.DISABLED,
      id: a.current,
      onFocus: D,
      onBlur: z,
      tabIndex: r === d.DISABLED ? -1 : 0,
      ref: _,
      children: s || n
    }
  );
});
y.displayName = "FormFieldTarget";
export {
  y as FormFieldTarget
};
