// @sberbusiness/triplex-next/components/FormField/components/FormFieldTextarea.js
import { jsx as E } from "react/jsx-runtime";
import I, { useContext as b, useRef as g, useEffect as c } from "react";
import { FormFieldContext as h } from "../FormFieldContext.js";
import { a as z, c as R } from "../../../chunks/vendor-Ba_pyhza.js";
import { createSizeToClassNameMap as S } from "../../../utils/classNameMaps.js";
import { EFormFieldStatus as y } from "../enums.js";
const D = "formFieldTextarea__28af73f9", M = "sm__b412b02f", j = "md__c14e799d", q = "lg__96871987", l = {
  formFieldTextarea: D,
  sm: M,
  md: j,
  lg: q
}, w = S(l), A = I.forwardRef(
  ({ className: f, id: e, onBlur: a, onFocus: r, placeholder: n, value: o, ...x }, p) => {
    const { size: u, status: F, focused: _, setFocused: d, setId: t, setValueExist: i } = b(h), s = g(e === void 0 ? z() : ""), T = R(l.formFieldTextarea, w[u], f);
    c(() => {
      t(s.current);
    }, [t]), c(() => {
      e && (s.current = e, t(s.current));
    }, [e, t]), c(() => {
      i(!!o);
    }, [i, o]);
    const N = (m) => {
      d(!1), a == null || a(m);
    }, C = (m) => {
      d(!0), r == null || r(m);
    };
    return /* @__PURE__ */ E(
      "textarea",
      {
        ...x,
        id: s.current,
        className: T,
        disabled: F === y.DISABLED,
        onFocus: C,
        onBlur: N,
        placeholder: _ ? n : void 0,
        value: o,
        ref: p
      }
    );
  }
);
A.displayName = "FormFieldTextarea";
export {
  A as FormFieldTextarea
};
