// @sberbusiness/triplex-next/components/Typography/Caption.js
import { jsx as f } from "react/jsx-runtime";
import { forwardRef as g } from "react";
import { c as y } from "../../chunks/vendor-Ba_pyhza.js";
import { EFontWeightText as n, ECaptionSize as c, EFontType as T } from "./enums.js";
import { t as s, m as b } from "../../chunks/utils-C8a4iCV0.js";
const R = "caption__5fdb3c24", E = "regular__54c9b5fe", u = "semibold__30123ab3", x = "c1__c1c3ad33", S = "c2__7a03151f", h = "d1__46674d4f", o = {
  caption: R,
  regular: E,
  semibold: u,
  c1: x,
  c2: S,
  d1: h
}, F = {
  [n.REGULAR]: o.regular,
  [n.SEMIBOLD]: o.semibold
}, A = {
  [c.C1]: o.c1,
  [c.C2]: o.c2,
  [c.D1]: o.d1
}, L = g(
  ({
    children: e,
    className: p,
    size: r,
    tag: i = "span",
    type: m = T.PRIMARY,
    weight: l = n.REGULAR,
    underline: t,
    strikethrough: a,
    ...d
  }, C) => {
    const _ = y(
      s.typography,
      o.caption,
      A[r],
      b[m],
      F[l],
      {
        [s.strikethrough]: !!a && !t,
        [s.underline]: !!t && !a,
        [s.underlineStrikethrough]: !!a && !!t
      },
      p
    );
    return /* @__PURE__ */ f(i, { ref: C, className: _, ...d, children: e });
  }
);
L.displayName = "Caption";
export {
  L as Caption,
  A as mapCaptionSizeToCssClass,
  F as mapFontWeightTextToCssClass
};
