// @sberbusiness/triplex-next/components/Typography/Text.js
import { jsx as d } from "react/jsx-runtime";
import { forwardRef as g } from "react";
import { c as C } from "../../chunks/vendor-Ba_pyhza.js";
import { ETextSize as e, EFontWeightText as r, ELineType as b, EFontType as f } from "./enums.js";
import { t, m as R } from "../../chunks/utils-C8a4iCV0.js";
const E = "text__0ea99e12", L = "regular__54c9b5fe", A = "semibold__30123ab3", u = "b1__c10b3dd6", B = "b2__a85809e6", M = "b3__f8e6d427", S = "b4__05ec6de9", s = {
  text: E,
  regular: L,
  semibold: A,
  b1: u,
  b2: B,
  b3: M,
  b4: S
}, h = {
  [e.B1]: s.b1,
  [e.B2]: s.b2,
  [e.B3]: s.b3,
  [e.B4]: s.b4
}, F = {
  [r.REGULAR]: s.regular,
  [r.SEMIBOLD]: s.semibold
}, N = {
  [b.NORMAL]: "",
  [b.COMPACT]: t.compact
}, O = g(
  ({
    children: c,
    className: m,
    size: n,
    tag: p = "span",
    type: l = f.PRIMARY,
    weight: T = r.REGULAR,
    line: i = b.NORMAL,
    underline: o,
    strikethrough: a,
    ...x
  }, _) => {
    const y = C(
      t.typography,
      s.text,
      h[n],
      R[l],
      F[T],
      N[i],
      {
        [t.strikethrough]: !!a && !o,
        [t.underline]: !!o && !a,
        [t.underlineStrikethrough]: !!a && !!o
      },
      m
    );
    return /* @__PURE__ */ d(p, { ref: _, className: y, ...x, children: c });
  }
);
O.displayName = "Text";
export {
  O as Text,
  F as mapFontWeightTextToCssClass,
  N as mapTextLineTypeToCssClass,
  h as mapTextSizeToCssClass
};
