// @sberbusiness/triplex-next/components/Typography/Title.js
import { jsx as b } from "react/jsx-runtime";
import { forwardRef as f } from "react";
import { c as T } from "../../chunks/vendor-Ba_pyhza.js";
import { ETitleSize as m, EFontWeightTitle as o, EFontType as g } from "./enums.js";
import { t as s, m as u } from "../../chunks/utils-C8a4iCV0.js";
const y = "title__86287c92", E = "regular__54c9b5fe", C = "semibold__30123ab3", S = "medium__9d0a8fb0", M = "bold__243ddcbb", R = "h1__3ffe964d", D = "h2__6d6e36df", F = "h3__276b4f3e", t = {
  title: y,
  regular: E,
  semibold: C,
  medium: S,
  bold: M,
  h1: R,
  h2: D,
  h3: F
}, I = {
  [m.H1]: t.h1,
  [m.H2]: t.h2,
  [m.H3]: t.h3
}, L = {
  [o.MEDIUM]: t.medium,
  [o.REGULAR]: t.regular,
  [o.SEMIBOLD]: t.semibold,
  [o.BOLD]: t.bold
}, x = f(
  ({
    children: a,
    className: r,
    size: i,
    tag: c = `${i}`,
    type: d = g.PRIMARY,
    weight: n = o.SEMIBOLD,
    underline: e,
    strikethrough: l,
    ...h
  }, p) => {
    const _ = T(
      s.typography,
      t.title,
      I[i],
      u[d],
      L[n],
      {
        [s.strikethrough]: !!l && !e,
        [s.underline]: !!e && !l,
        [s.underlineStrikethrough]: !!l && !!e
      },
      r
    );
    return /* @__PURE__ */ b(c, { ref: p, className: _, ...h, children: a });
  }
);
x.displayName = "Title";
export {
  x as Title,
  L as mapFontWeightTitleToCssClass,
  I as mapTitleSizeToCssClass
};
