// @sberbusiness/triplex-next/components/List/ListSortable.js
var x = Object.defineProperty;
var p = (t, e, r) => e in t ? x(t, e, { enumerable: !0, configurable: !0, writable: !0, value: r }) : t[e] = r;
var o = (t, e, r) => p(t, typeof e != "symbol" ? e + "" : e, r);
import { jsx as d } from "react/jsx-runtime";
import E from "react";
import { b, d as f, D as M, M as T, r as D, e as L, S as w, v as y, f as I, T as C, g as H } from "../../chunks/vendor-Ba_pyhza.js";
import { List as N } from "./List.js";
class u extends H {
}
o(u, "activators", [
  {
    eventName: "onMouseDown",
    handler: ({ nativeEvent: e }, { onActivation: r }) => {
      var s;
      if (e.button === 2)
        return !1;
      if (e.target instanceof HTMLElement) {
        let a = e.target;
        for (; a !== null; ) {
          if (((s = a.dataset) == null ? void 0 : s.draggable) === "false")
            return !1;
          a = a.parentElement;
        }
      }
      return r == null || r({ event: e }), !0;
    }
  }
]);
class g extends C {
}
o(g, "activators", [
  {
    eventName: "onTouchStart",
    handler: ({ nativeEvent: e }, { onActivation: r }) => {
      var s;
      if (e.touches.length > 1)
        return !1;
      if (e.target instanceof HTMLElement) {
        let a = e.target;
        for (; a !== null; ) {
          if (((s = a.dataset) == null ? void 0 : s.draggable) === "false")
            return !1;
          a = a.parentElement;
        }
      }
      return r == null || r({ event: e }), !0;
    }
  }
]);
function R({ items: t, onItemsChange: e, ...r }, s) {
  const a = b(f(u), f(g)), c = (m) => {
    const { active: i, over: n } = m;
    if (n && i.id !== n.id) {
      const h = t.findIndex((l) => l.id === i.id), S = t.findIndex((l) => l.id === n.id);
      e(I(t, h, S));
    }
  };
  return /* @__PURE__ */ d(
    M,
    {
      sensors: a,
      modifiers: [D, L],
      measuring: { droppable: { frequency: T.WhileDragging } },
      onDragEnd: c,
      children: /* @__PURE__ */ d(w, { items: t, strategy: y, children: /* @__PURE__ */ d(N, { ...r, ref: s }) })
    }
  );
}
const W = E.forwardRef(R);
export {
  u as AdvancedMouseSensor,
  g as AdvancedTouchSensor,
  W as ListSortable
};
