// @sberbusiness/triplex-next/components/Table/TableBasic/TableBasic.js
import { jsxs as d, jsx as r } from "react/jsx-runtime";
import { useContext as B, useEffect as T } from "react";
import { i as N } from "../../../chunks/vendor-Ba_pyhza.js";
import { MasterTableContext as g } from "../MasterTableContext.js";
import { s as a } from "../../../chunks/utils-DwX2_kEA.js";
import { TableBasicHeader as C } from "./components/TableBasicHeader.js";
import { TableBasicBody as E } from "./components/TableBasicBody.js";
import { LoaderMiddle as L } from "../../Loader/LoaderMiddle/LoaderMiddle.js";
import { LoaderScreen as j } from "../../LoaderScreen/LoaderScreen.js";
const D = ({
  columns: t,
  data: s,
  highlightRowOnHover: m,
  onOrderBy: c,
  onClickRow: f,
  renderNoColumns: n,
  renderNoData: p,
  headless: b,
  ...v
}) => {
  const o = B(g), y = s.length === 0;
  T(() => {
    N(t, o.columns) || o.setColumns(t);
  }, [t, o]);
  const u = () => t.every((e) => e.hidden) && n ? n() : /* @__PURE__ */ d("table", { ...v, children: [
    b || /* @__PURE__ */ r(C, { columns: t, onOrderBy: c }),
    /* @__PURE__ */ r(
      E,
      {
        columns: t,
        data: s,
        onClickRow: f,
        highlightRowOnHover: m
      }
    )
  ] }, "table"), h = (e) => {
    const { loading: i } = o;
    return i && e ? l(x()) : !i && e ? l(p()) : i && !e ? /* @__PURE__ */ r("div", { className: a.spinnerWrapper, children: /* @__PURE__ */ r(j, { type: "middle" }) }) : null;
  }, l = (e) => /* @__PURE__ */ r("div", { className: a.footerEmptyData, children: e }), x = () => [
    /* @__PURE__ */ r("div", { className: a.overlayCover }, "overlay"),
    /* @__PURE__ */ r(L, {}, "spinner")
  ];
  return /* @__PURE__ */ d("div", { className: a.tableBasic, children: [
    u(),
    h(y)
  ] });
};
D.displayName = "TableBasic";
export {
  D as TableBasic
};
