// @sberbusiness/triplex-next/components/Table/MasterTable.js
import { jsx as e } from "react/jsx-runtime";
import { useState as i } from "react";
import { MasterTableContext as n } from "./MasterTableContext.js";
import { c as b } from "../../chunks/vendor-Ba_pyhza.js";
import { NoColumns as p } from "./NoColumns.js";
import { FilterPanel as T } from "./FilterPanel.js";
import { TabsLinePanel as c } from "./TabsLinePanel.js";
import { TableBasic as f } from "./TableBasic/TableBasic.js";
import { TableBasicSettings as P } from "./TableBasicSettings/TableBasicSettings.js";
import { TableFooter as u } from "./TableFooter/TableFooter.js";
import { PaginationPanel as d } from "./PaginationPanel.js";
import "./TableBasicSettings/components/ColumnSettings.js";
import "./TableBasic/enums.js";
const x = "masterTable__b18bca89", g = {
  masterTable: x
}, a = ({ children: t, className: o, loading: r = !1, ...m }) => {
  const [s, l] = i([]);
  return /* @__PURE__ */ e(
    n.Provider,
    {
      value: {
        columns: s,
        loading: r,
        setColumns: l
      },
      children: /* @__PURE__ */ e(
        "div",
        {
          className: b(g.masterTable, o),
          ...m,
          "data-tx": "1.4.0",
          children: t
        }
      )
    }
  );
};
a.displayName = "MasterTable";
a.NoColumns = p;
a.FilterPanel = T;
a.TabsLinePanel = c;
a.TableBasic = f;
a.TableBasicSettings = P;
a.TableFooter = u;
a.PaginationPanel = d;
export {
  a as MasterTable
};
