// @sberbusiness/triplex-next/components/SwipeableArea/SwipeableArea.js
import { jsxs as j, jsx as b } from "react/jsx-runtime";
import z, { useState as w, useRef as u, useEffect as M, useImperativeHandle as W } from "react";
import { c as g } from "../../chunks/vendor-Ba_pyhza.js";
const H = "swipeableArea__f2b5a350", V = "content__8d64dd62", $ = "disableScroll__4d2d1b89", k = "disablePointerEvents__ef08160c", q = "swipeAnimationFinish__1fa391cf", G = "leftContent__01c5f9bc", J = "rightContent__9b3ba465", s = {
  swipeableArea: H,
  content: V,
  disableScroll: $,
  disablePointerEvents: k,
  swipeAnimationFinish: q,
  leftContent: G,
  rightContent: J
}, p = 24, A = s.swipeAnimationFinish, K = s.disableScroll, Q = s.disablePointerEvents, N = { clientX: 0, clientY: 0 }, U = z.forwardRef(
  ({ children: L, className: I, leftSwipeableArea: m, rightSwipeableArea: C, ...y }, B) => {
    const [S, a] = w(!1), [l, d] = w(), [o, e] = w(0), i = u(null), r = u(null), n = u(0), f = u(N), _ = u(null), X = () => {
      a(!0);
      const t = n.current - o;
      n.current === 0 ? t > 0 ? e(
        t > p ? -r.current.getBoundingClientRect().width : 0
      ) : e(
        Math.abs(t) > p ? i.current.getBoundingClientRect().width : 0
      ) : n.current > 0 ? t > 0 && e(
        t > p ? 0 : i.current.getBoundingClientRect().width
      ) : n.current < 0 && t < 0 && e(
        Math.abs(t) > p ? 0 : -r.current.getBoundingClientRect().width
      );
    }, E = () => {
      f.current = N, d(void 0), document.removeEventListener("touchend", E);
    };
    M(() => {
      !l && o !== n.current && X();
    }, [l]);
    const x = (t) => {
      f.current = { clientX: t.touches[0].clientX, clientY: t.touches[0].clientY }, n.current = o, document.addEventListener("touchend", E);
    }, O = (t) => {
      if (t.touches.length != 1)
        return;
      const c = t.touches[0].clientX - f.current.clientX, F = t.touches[0].clientY - f.current.clientY;
      if (!l) {
        Math.abs(F) > Math.abs(c) ? d(
          "vertical"
          /* vertical */
        ) : d(
          "horizontal"
          /* horizontal */
        );
        return;
      }
      if (l === "vertical")
        return;
      const h = n.current + c;
      n.current === 0 ? c > 0 && m ? e(
        Math.min(h, i.current.getBoundingClientRect().width)
      ) : c < 0 && C && e(
        Math.max(h, -r.current.getBoundingClientRect().width)
      ) : n.current > 0 ? c < 0 && e(Math.max(0, h)) : n.current < 0 && c > 0 && e(Math.min(0, h));
    }, D = () => a(!1), P = () => r.current ? Math.abs(o) / r.current.getBoundingClientRect().width : 1, Y = () => i.current ? Math.abs(o) / i.current.getBoundingClientRect().width : 1;
    M(() => {
      const t = (c) => {
        c.touches.length == 1 && _.current && !_.current.contains(c.target) && (a(!0), e(0), d(void 0));
      };
      return document.addEventListener("touchstart", t), () => document.removeEventListener("touchstart", t);
    }, []);
    const R = () => {
      i.current && (a(!0), e(i.current.getBoundingClientRect().width));
    }, T = () => {
      r.current && (a(!0), e(-r.current.getBoundingClientRect().width));
    }, v = () => {
      (r.current || i.current) && (a(!0), e(0));
    };
    return W(
      B,
      () => ({
        closeSwipe: () => {
          v();
        },
        swipeLeft: () => {
          T();
        },
        swipeRight: () => {
          R();
        }
      }),
      [v, T, R]
    ), /* @__PURE__ */ j(
      "div",
      {
        className: g(s.swipeableArea, I),
        ...y,
        "data-tx": "1.4.0",
        ref: _,
        children: [
          typeof m < "u" ? /* @__PURE__ */ b(
            "div",
            {
              className: g(s.leftContent, {
                [A]: S
              }),
              ref: i,
              style: { opacity: Y() },
              children: m
            }
          ) : null,
          typeof C < "u" ? /* @__PURE__ */ b(
            "div",
            {
              className: g(s.rightContent, {
                [A]: S
              }),
              ref: r,
              style: { opacity: P() },
              children: C
            }
          ) : null,
          /* @__PURE__ */ b(
            "div",
            {
              className: g(s.content, {
                [Q]: o !== 0,
                [K]: l === "horizontal",
                [A]: S
              }),
              onTouchStart: x,
              onTouchMove: O,
              style: {
                transform: `translateX(${o}px)`
              },
              onTransitionEnd: D,
              children: L
            }
          )
        ]
      }
    );
  }
);
U.displayName = "SwipeableArea";
export {
  U as SwipeableArea
};
