// @sberbusiness/triplex-next/components/Notification/Notification.js
import { jsx as f } from "react/jsx-runtime";
import { c as l } from "../../chunks/vendor-Ba_pyhza.js";
import { NotificationBody as d } from "./components/NotificationBody.js";
import { NotificationClose as p } from "./components/NotificationClose.js";
import { NotificationIcon as N } from "./components/NotificationIcon.js";
import { NotificationTime as x } from "./components/NotificationTime.js";
import { s as i } from "../../chunks/Notification.module-C-HCMpAz.js";
const o = ((t) => {
  const {
    children: a,
    className: r,
    withExtraBottomPadding: s = !1,
    isShowCloseOnHover: e = !1,
    onClick: m,
    ...n
  } = t, c = l(r, i.notification, {
    [i.showCloseOnHover]: e,
    [i.extraBottomPadding]: s
  });
  return /* @__PURE__ */ f(
    "div",
    {
      ...n,
      role: "alertdialog",
      className: c,
      onClick: m,
      "data-tx": "1.4.0",
      children: a
    }
  );
});
o.displayName = "Notification";
o.Icon = N;
o.Body = d;
o.Close = p;
o.Time = x;
export {
  o as Notification
};
