// @sberbusiness/triplex-next/components/LightBox/LightBoxTopOverlay/LightBoxTopOverlay.js
import { jsx as l, jsxs as P, Fragment as I } from "react/jsx-runtime";
import { useState as v, useRef as d, useEffect as C } from "react";
import { c as R, F as S } from "../../../chunks/vendor-Ba_pyhza.js";
import { EOverlayDirection as j } from "../../Overlay/OverlayBase.js";
import { Overlay as y } from "../../Overlay/Overlay.js";
const D = "lightBoxTopOverlayMask__9410973f", E = "lightBoxTopOverlayWrapper__11edb4db", L = "closing__a7cdf5fb", A = "opened__700be6cf", V = "lightBoxTopOverlayPanel__79e54db9", $ = "lightBoxTopOverlay__57206756", t = {
  lightBoxTopOverlayMask: D,
  lightBoxTopOverlayWrapper: E,
  closing: L,
  opened: A,
  lightBoxTopOverlayPanel: V,
  lightBoxTopOverlay: $
}, q = ({
  children: B,
  focusTrapProps: r,
  opened: e,
  onClose: s,
  onOpen: i,
  ...u
}) => {
  const [f, h] = v(!1), [m, O] = v(!1), [x, n] = v(0), c = d(e), a = d(null), T = () => {
    if (a.current) {
      const o = getComputedStyle(a.current).getPropertyValue("--lightBox-screen-top") || "0", p = a.current.getBoundingClientRect();
      p.top < 0 ? n(
        (g) => parseInt(g, 10) + Math.abs(p.top) + parseInt(o, 10)
      ) : n(
        // Оверлей расположен ниже, чем нужно.
        (g) => parseInt(g, 10) - p.top + parseInt(o, 10)
      );
    }
  }, _ = () => {
    T(), O(!0), i == null || i();
  };
  C(() => {
    c.current && !e ? h(!0) : !c.current && e && T(), c.current = e;
  }, [e]);
  const W = () => {
    h(!1), n(0), s == null || s();
  }, b = () => O(!1), k = (o) => /* @__PURE__ */ P(I, { children: [
    /* @__PURE__ */ l(y.Mask, { opened: o.opened, className: t.lightBoxTopOverlayMask }),
    /* @__PURE__ */ l(y.Panel, { className: t.lightBoxTopOverlayPanel, ...o, children: B })
  ] }), M = () => {
  }, N = R(t.lightBoxTopOverlayWrapper, {
    [t.closing]: f,
    [t.opened]: e
  }), F = /* @__PURE__ */ l(
    y,
    {
      onClose: W,
      onClosing: b,
      onOpen: _,
      opened: e,
      setOpened: M,
      ...u,
      className: t.lightBoxTopOverlay,
      direction: j.TOP,
      children: k
    }
  );
  return /* @__PURE__ */ l(
    S,
    {
      active: m,
      ...r,
      focusTrapOptions: {
        clickOutsideDeactivates: !0,
        preventScroll: !0,
        ...r == null ? void 0 : r.focusTrapOptions
      },
      children: /* @__PURE__ */ l(
        "div",
        {
          className: N,
          ref: a,
          style: x ? { top: `${x}px` } : void 0,
          children: F
        }
      )
    }
  );
};
q.displayName = "LightBoxTopOverlay";
export {
  q as LightBoxTopOverlay
};
