// @sberbusiness/triplex-next/components/CheckboxTreeExtended/components/CheckboxTreeExtendedArrow.js
var a = Object.defineProperty;
var c = (r, o, e) => o in r ? a(r, o, { enumerable: !0, configurable: !0, writable: !0, value: e }) : r[o] = e;
var n = (r, o, e) => c(r, typeof o != "symbol" ? o + "" : o, e);
import { jsx as p } from "react/jsx-runtime";
import { CaretdownStrokeSrvIcon16 as l } from "@sberbusiness/icons-next";
import { EVENT_KEY_CODES as d } from "../../../utils/keyboard.js";
import h from "react";
class N extends h.Component {
  constructor() {
    super(...arguments);
    n(this, "arrowNode");
    n(this, "setArrowNode", (e) => {
      this.arrowNode = e;
    });
    n(this, "handleClick", () => {
      const { opened: e, toggle: t } = this.props;
      t(!e);
    });
    /**
     * Обработчик нажатия клавиш.
     * Стрелка вправо - раскрыть, влево - свернуть.
     * Enter, space - изменить состояние на противоположное.
     */
    n(this, "handleKeyUp", (e) => {
      const { opened: t, toggle: s } = this.props;
      e.keyCode === d.ARROW_RIGHT ? s(!0) : e.keyCode === d.ARROW_LEFT ? s(!1) : [d.ENTER, d.SPACE].includes(e.keyCode) && s(!t);
    });
  }
  componentDidUpdate(e) {
    var i;
    const { active: t } = this.props, { active: s } = e;
    t && !s && ((i = this.arrowNode) == null || i.focus());
  }
  render() {
    return /* @__PURE__ */ p(
      "span",
      {
        className: "cssClass[caretIconWrapper]",
        onClick: this.handleClick,
        onKeyUp: this.handleKeyUp,
        ref: this.setArrowNode,
        role: "button",
        tabIndex: -1,
        children: /* @__PURE__ */ p(l, { paletteIndex: 0 })
      }
    );
  }
}
export {
  N as CheckboxTreeExtendedArrow
};
