// @sberbusiness/triplex-next/components/CheckboxTreeExtended/components/CheckboxTreeExtendedNode.js
var a = Object.defineProperty;
var x = (d, r, e) => r in d ? a(d, r, { enumerable: !0, configurable: !0, writable: !0, value: e }) : d[r] = e;
var o = (d, r, e) => x(d, typeof r != "symbol" ? r + "" : r, e);
import { jsx as h, jsxs as i } from "react/jsx-runtime";
import p from "react";
import { CollapsableTree as l } from "../../CollapsableTree/CollapsableTree.js";
import { isStaticCheckboxTreeExtended as b } from "../isStaticCheckboxTreeExtended.js";
import { c as m } from "../../../chunks/vendor-Ba_pyhza.js";
import "./CheckboxTreeExtendedArrow.js";
const N = "checkboxTreeExtendedNode__3ec333d4", T = "checkboxTreeExtendedNodeHeader__c4acf373", n = {
  checkboxTreeExtendedNode: N,
  checkboxTreeExtendedNodeHeader: T
};
class C extends p.Component {
  constructor() {
    super(...arguments);
    o(this, "handleHeaderMouseDown", (e) => {
      e.target === e.currentTarget && e.preventDefault();
    });
    o(this, "renderHeader", ({ activeNode: e, opened: c, toggle: s }) => {
      const { checkbox: t, children: k } = this.props;
      return /* @__PURE__ */ i(
        "div",
        {
          className: m(n.checkboxTreeExtendedNodeHeader, { opened: c }),
          onMouseDown: this.handleHeaderMouseDown,
          children: [
            !b,
            t({ active: e, opened: !0 })
          ]
        }
      );
    });
    o(this, "renderBody", () => this.props.children);
  }
  render() {
    const { children: e, checkbox: c, opened: s, ...t } = this.props;
    return /* @__PURE__ */ h(
      l.Node,
      {
        className: n.checkboxTreeExtendedNode,
        opened: !0,
        renderHeader: this.renderHeader,
        renderBody: this.renderBody,
        ...t
      }
    );
  }
}
export {
  C as CheckboxTreeExtendedNode
};
