// @sberbusiness/triplex-next/components/CheckboxTreeExtended/components/CheckboxTreeExtendedCheckbox.js
var i = Object.defineProperty;
var n = (c, o, e) => o in c ? i(c, o, { enumerable: !0, configurable: !0, writable: !0, value: e }) : c[o] = e;
var t = (c, o, e) => n(c, typeof o != "symbol" ? o + "" : o, e);
import { jsx as x } from "react/jsx-runtime";
import b from "react";
import { Checkbox as d } from "../../Checkbox/Checkbox.js";
import { c as m } from "../../../chunks/vendor-Ba_pyhza.js";
class N extends b.Component {
  constructor() {
    super(...arguments);
    t(this, "checkboxNode");
    t(this, "handleFocus", (e) => {
    });
    t(this, "setCheckboxNode", (e) => {
      this.checkboxNode = e;
    });
  }
  componentDidUpdate(e) {
    var s, r;
    const { active: h } = this.props, { active: a } = e;
    h && !a && ((s = document.activeElement) != null && s.contains(this.checkboxNode) || (r = this.checkboxNode) == null || r.focus());
  }
  render() {
    const { active: e, className: h, opened: a, ...s } = this.props;
    return /* @__PURE__ */ x(
      d,
      {
        className: m("cssClass[checkboxTreeCheckbox]", h),
        ref: this.setCheckboxNode,
        labelAttributes: { className: "cssClass[checkboxTreeCheckboxLabel]", onFocus: this.handleFocus },
        ...s
      }
    );
  }
}
export {
  N as CheckboxTreeExtendedCheckbox
};
