// @sberbusiness/triplex-next/chunks/AlertProcessSpoiler-BQv9QI6L.js
import { jsxs as p, jsx as t } from "react/jsx-runtime";
import { useContext as d, useEffect as i } from "react";
import { CaretdownStrokeSrvIcon16 as _ } from "@sberbusiness/icons-next";
import { ButtonIcon as m } from "../components/Button/ButtonIcon.js";
import { AlertProcessContext as x } from "../components/Alert/AlertProcess/AlertProcessContext.js";
import { c as s } from "./vendor-Ba_pyhza.js";
const y = "alertProcess__96d0382e", f = "withSpoiler__1b8522a9", u = "themeIcon__9bfe5b45", T = "alertTypeError__ced7c27f", b = "alertTypeInfo__3002ea3b", B = "alertTypeWarning__732e3a96", C = "alertTypeSystem__2c54c9e2", S = "alertTypeFeature__23e4269b", h = "alertProcessContentBlock__4e11d402", I = "closeButton__2943f73b", P = "expandableContent__fdd6ee88", g = "expanded__d208d07b", k = "expandButton__5db473d1", e = {
  alertProcess: y,
  withSpoiler: f,
  themeIcon: u,
  alertTypeError: T,
  alertTypeInfo: b,
  alertTypeWarning: B,
  alertTypeSystem: C,
  alertTypeFeature: S,
  alertProcessContentBlock: h,
  closeButton: I,
  expandableContent: P,
  expanded: g,
  expandButton: k
}, v = ({ children: a, open: r, onOpen: o, ...l }) => {
  const { setHasSpoiler: n } = d(x);
  i(() => (n(!0), () => {
    n(!1);
  }), [n]);
  const c = () => {
    o == null || o(!r);
  };
  return /* @__PURE__ */ p("div", { className: e.spoiler, children: [
    /* @__PURE__ */ t("div", { className: s(e.expandableContent, { [e.expanded]: r }), ...l, children: a }),
    /* @__PURE__ */ t("div", { className: s(e.expandButton, { [e.expanded]: r }), children: /* @__PURE__ */ t(m, { onClick: c, children: /* @__PURE__ */ t(_, { paletteIndex: 5 }) }) })
  ] });
};
v.displayName = "AlertProcessSpoiler";
export {
  v as A,
  e as s
};
