// @sberbusiness/triplex-next/chunks/Chip-G6cGXuGj.js
import { jsxs as h, jsx as a } from "react/jsx-runtime";
import b, { useCallback as x } from "react";
import { c as n } from "./vendor-Ba_pyhza.js";
import { EComponentSize as u } from "../enums/EComponentSize.js";
import { createSizeToClassNameMap as P } from "../utils/classNameMaps.js";
import { isKey as C } from "../utils/keyboard.js";
const N = "chipGroupItem__cf23b277", w = "chip__c125ac12", I = "lg__0de9bae2", B = "prefix__7e057460", G = "postfix__d3905a5d", g = "md__22d31d53", z = "sm__bb628cd7", M = "disabled__d194d8ca", S = "selected__320abf0a", j = "withPrefix__da7b2b2b", y = "withPostfix__65282022", E = "content__51db1204", e = {
  chipGroupItem: N,
  chip: w,
  lg: I,
  prefix: B,
  postfix: G,
  md: g,
  sm: z,
  disabled: M,
  selected: S,
  withPrefix: j,
  withPostfix: y,
  content: E
}, R = P(e), T = b.forwardRef(
  ({ children: l, className: d, disabled: t, postfix: o, prefix: c, selected: p, size: r = u.MD, onKeyDown: s, ...m }, f) => {
    const _ = x(
      (i) => {
        C(i.code, "SPACE") && i.preventDefault(), s == null || s(i);
      },
      [s]
    );
    return /* @__PURE__ */ h(
      "span",
      {
        className: n(
          e.chip,
          e.chipGroupItem,
          R[r],
          {
            [e.disabled]: !!t,
            [e.selected]: !!p,
            [e.withPostfix]: typeof o < "u",
            [e.withPrefix]: typeof c < "u"
          },
          d
        ),
        role: "button",
        tabIndex: t ? -1 : 0,
        onKeyDown: _,
        ...m,
        ref: f,
        children: [
          c ? /* @__PURE__ */ a(
            "span",
            {
              className: n(
                e.prefix,
                "hoverable",
                // Для иконок.
                {
                  disabled: !!t
                  // Для иконок.
                }
              ),
              children: c
            }
          ) : null,
          /* @__PURE__ */ a("span", { className: e.content, children: l }),
          o ? /* @__PURE__ */ a(
            "span",
            {
              className: n(
                e.postfix,
                "hoverable",
                // Для иконок.
                {
                  disabled: !!t
                  // Для иконок.
                }
              ),
              children: o
            }
          ) : null
        ]
      }
    );
  }
);
T.displayName = "Chip";
export {
  T as C,
  e as s
};
