// @sberbusiness/triplex-next/chunks/DropdownListItem-D2hzyBGc.js
import { jsx as w } from "react/jsx-runtime";
import f, { useEffect as E } from "react";
import { EVENT_KEY_CODES as i } from "../utils/keyboard.js";
import { c as k } from "./vendor-Ba_pyhza.js";
const u = "dropdownDesktopList__c85160d1", v = "sm__68d529af", y = "dropdownDesktopListItem__f75c375c", I = "dropdownDesktopListLoader__c994d3c6", l = "md__d6d3e86e", N = "lg__a0bb2012", b = "active__234b14fb", g = "selected__aa469e95", n = {
  dropdownDesktopList: u,
  sm: v,
  dropdownDesktopListItem: y,
  dropdownDesktopListLoader: I,
  md: l,
  lg: N,
  active: b,
  selected: g
}, O = [i.SPACE, i.ENTER], R = f.forwardRef(
  ({
    active: s,
    keyCodesForSelection: m = O,
    children: e,
    className: a,
    onClick: d,
    onSelect: t,
    selected: p,
    ..._
  }, c) => {
    const r = (o) => {
      const { keyCode: D } = o;
      m.includes(D) && (o.preventDefault(), t == null || t());
    }, L = (o) => {
      t == null || t(), d == null || d(o);
    };
    return E(() => (s ? document.addEventListener("keydown", r) : document.removeEventListener("keydown", r), () => {
      document.removeEventListener("keydown", r);
    }), [s]), /* @__PURE__ */ w(
      "div",
      {
        className: k(
          n.dropdownDesktopListItem,
          { [n.active]: !!s, [n.selected]: !!p },
          a
        ),
        title: typeof e == "string" ? e : void 0,
        role: "option",
        "aria-selected": !!p,
        ..._,
        onClick: L,
        ref: c,
        children: e
      }
    );
  }
);
R.displayName = "DropdownListItem";
export {
  R as D,
  n as s
};
