// @sberbusiness/triplex-next/chunks/FormFieldInput-CO7yZP0Q.js
import { jsx as g } from "react/jsx-runtime";
import z, { useContext as A, useRef as P, useEffect as m } from "react";
import { FormFieldContext as R } from "../components/FormField/FormFieldContext.js";
import { c as k, a as y } from "./vendor-Ba_pyhza.js";
import { EFormFieldStatus as D } from "../components/FormField/enums.js";
import { createSizeToClassNameMap as M } from "../utils/classNameMaps.js";
const T = "formFieldInput__3674a3e7", W = "sm__b412b02f", j = "md__c14e799d", q = "lg__96871987", x = {
  formFieldInput: T,
  sm: W,
  md: j,
  lg: q
}, w = M(x), L = z.forwardRef((c, d) => {
  const { className: S, id: s, onAnimationStart: r, onBlur: n, onFocus: i, placeholder: u, value: o, ...B } = c, { render: f, ...C } = c, { focused: E, status: b, setFocused: p, setId: l, setValueExist: a, size: F } = A(R), I = k(x.formFieldInput, w[F], S), t = P(s || y("input_"));
  m(() => {
    l(t.current);
  }, []), m(() => {
    s && (t.current = s, l(t.current));
  }, [s, l]), m(() => {
    a(!!o);
  }, [a, o]);
  const _ = (e) => {
    p(!1), n == null || n(e);
  }, h = (e) => {
    p(!0), i == null || i(e);
  }, N = (e) => {
    e.animationName.startsWith("autofill-applied-hook") ? a(!0) : e.animationName.startsWith("autofill-cancelled-hook") && (o || a(!1)), r == null || r(e);
  };
  return f ? f(
    {
      ...C,
      className: I,
      id: t.current,
      onAnimationStart: N,
      onBlur: _,
      onFocus: h,
      /* Когда элемент не в фокусе, вместо placeholder показывается Label. */
      placeholder: E ? u : " ",
      size: F
    },
    d
  ) : /* @__PURE__ */ g(
    "input",
    {
      ...B,
      className: I,
      disabled: b === D.DISABLED,
      id: t.current,
      onAnimationStart: N,
      onFocus: h,
      onBlur: _,
      value: o,
      placeholder: u,
      ref: d
    }
  );
});
L.displayName = "FormFieldInput";
export {
  L as F,
  x as s
};
