// @sberbusiness/triplex-next/chunks/LightBoxContent-DmoszkOx.js
import { jsx as i, jsxs as y } from "react/jsx-runtime";
import { useState as v, useRef as _, useCallback as h, useLayoutEffect as C, useEffect as O } from "react";
import { WindowResizeListener as S } from "../components/WindowResizeListener/WindowResizeListener.js";
import { u as T, c as b } from "./vendor-Ba_pyhza.js";
import { LoaderScreen as R } from "../components/LoaderScreen/LoaderScreen.js";
const z = "lightBox__da82777f", A = "lightBoxBackdrop__588a3cfe", L = "lightBoxSideOverlayActive__13153a38", W = "lightBoxTopOverlayActive__4271910e", N = "lightBoxContent__265f0120", E = "loadingContentOverlay__b4a5e5cb", k = "lightBoxContentResizeWrapper__fe389b0a", w = "tempElSafariBug__f2b1ff44", H = "bodyOverflowHidden__41ad88f6", r = {
  lightBox: z,
  lightBoxBackdrop: A,
  lightBoxSideOverlayActive: L,
  lightBoxTopOverlayActive: W,
  lightBoxContent: N,
  loadingContentOverlay: E,
  lightBoxContentResizeWrapper: k,
  tempElSafariBug: w,
  bodyOverflowHidden: H
}, j = (u) => {
  const { children: l, className: a, isLoading: s, loadingTitle: c, ...p } = u, [d, B] = v(0), e = _(null), o = h(() => {
    let n = 0;
    const f = document.querySelector('[data-lightbox-component="controls"]');
    f && (n += f.offsetHeight), B((g) => Math.abs(g - n) > 1 ? n : g);
  }, []), t = h(() => {
    e.current && clearTimeout(e.current), e.current = setTimeout(o, 100);
  }, [o]);
  C(() => (o(), () => {
    e.current && clearTimeout(e.current);
  }), []), O(() => {
    t();
  }, [l, a, s, c, t]);
  const { ref: x } = T({
    handleWidth: !0,
    onResize: t,
    refreshMode: "debounce",
    refreshRate: 100
  }), m = {
    paddingTop: d
  };
  return /* @__PURE__ */ i(S, { onResize: t, children: /* @__PURE__ */ y(
    "div",
    {
      className: b(a, r.lightBoxContent),
      style: d ? m : void 0,
      ...p,
      children: [
        l,
        s && /* @__PURE__ */ i(R, { className: r.loadingContentOverlay, type: "middle", children: c }),
        /* @__PURE__ */ i("div", { className: r.lightBoxContentResizeWrapper, ref: x })
      ]
    }
  ) });
};
j.displayName = "LightBoxContent";
export {
  j as L,
  r as s
};
