// @sberbusiness/triplex-next/chunks/ListItemControlsButton-BoEiexyr.js
import { jsx as s, jsxs as m } from "react/jsx-runtime";
import i from "react";
import { c as I } from "./vendor-Ba_pyhza.js";
import { ETextSize as u } from "../components/Typography/enums.js";
import { Text as c } from "../components/Typography/Text.js";
const a = "listItemControlsButton__b0dd3f67", B = "listItemControlsButtonLabel__bc81bce7", C = "listItemControlsButtonDropdown__443299e2", d = "listItemControlsButtonIcon__5e7f3909", p = "withText__f5363776", _ = "listItemControlsButtonInner__276d9a84", f = "withIcon__7f9afd16", w = "buttonDropdownMenuItem__6565939d", t = {
  listItemControlsButton: a,
  listItemControlsButtonLabel: B,
  listItemControlsButtonDropdown: C,
  listItemControlsButtonIcon: d,
  withText: p,
  listItemControlsButtonInner: _,
  withIcon: f,
  buttonDropdownMenuItem: w
}, b = i.forwardRef(
  ({ children: o, className: e, icon: n, ...l }, r) => /* @__PURE__ */ s(
    "button",
    {
      type: "button",
      className: I(
        t.listItemControlsButton,
        "hoverable",
        {
          [t.withIcon]: typeof n < "u",
          [t.withText]: typeof o < "u"
        },
        e
      ),
      ...l,
      ref: r,
      children: /* @__PURE__ */ m("span", { className: t.listItemControlsButtonInner, children: [
        n ? /* @__PURE__ */ s("span", { className: t.listItemControlsButtonIcon, children: n }) : null,
        o ? /* @__PURE__ */ s(c, { className: t.listItemControlsButtonLabel, size: u.B4, children: o }) : null
      ] })
    }
  )
);
b.displayName = "ListItemControlsButton";
export {
  b as L,
  t as s
};
