// @sberbusiness/triplex-next/chunks/StatusTracker.module-zTyojoH4.js
const t = "statusTrackerWrapper__775b0efc", r = "statusTracker__3bfc5d9a", a = "verticalAlignMiddle__10eca07a", s = "verticalAlignBottom__e98edef8", e = "statusTrackerChild__c2117f3c", c = "statusTrackerSum__41417587", o = "statusTrackerTitle__8ae664d4", u = "statusTrackerDescription__d9140d43", n = "statusTrackerAlert__de3cfd9e", k = "statusTrackerStatus__3e74bd77", T = "statusTrackerFooterWrapper__29901a61", i = "statusTrackerButton__1cf9ab3d", _ = "statusTrackerColor__26702aa1", d = "waiting__c27f0023", l = "warning__46f987da", p = "rejected__59c1bdc4", g = "approved__815bef50", f = "statusTrackerBackground__41087f29", v = {
  statusTrackerWrapper: t,
  statusTracker: r,
  verticalAlignMiddle: a,
  verticalAlignBottom: s,
  statusTrackerChild: e,
  statusTrackerSum: c,
  statusTrackerTitle: o,
  statusTrackerDescription: u,
  statusTrackerAlert: n,
  statusTrackerStatus: k,
  statusTrackerFooterWrapper: T,
  statusTrackerButton: i,
  statusTrackerColor: _,
  waiting: d,
  warning: l,
  rejected: p,
  approved: g,
  statusTrackerBackground: f
};
export {
  v as s
};
