// @sberbusiness/triplex-next/chunks/utils-C8a4iCV0.js
import { EFontType as r } from "../components/Typography/enums.js";
const t = "underline__4b5c155e", n = "strikethrough__3ac02950", s = "underlineStrikethrough__8ca67583", o = "typography__cfdd71a8", c = "primary__041b0166", a = "complementary__c831b0d4", i = "secondary__70bf5058", _ = "tertiary__7e3a68c5", y = "disabled__ed085e78", d = "brand__851e1e6c", p = "info__0b3502f9", m = "success__b870fba5", b = "error__ef201f07", h = "warning__6e8f50ab", l = "system__03010f94", u = "compact__fe7d4ece", e = {
  underline: t,
  strikethrough: n,
  underlineStrikethrough: s,
  typography: o,
  primary: c,
  complementary: a,
  secondary: i,
  tertiary: _,
  disabled: y,
  brand: d,
  info: p,
  success: m,
  error: b,
  warning: h,
  system: l,
  compact: u
}, g = {
  [r.PRIMARY]: e.primary,
  [r.COMPLEMENTARY]: e.complementary,
  [r.SECONDARY]: e.secondary,
  [r.TERTIARY]: e.tertiary,
  [r.DISABLED]: e.disabled,
  [r.BRAND]: e.brand,
  [r.INFO]: e.info,
  [r.SUCCESS]: e.success,
  [r.ERROR]: e.error,
  [r.WARNING]: e.warning,
  [r.SYSTEM]: e.system
};
export {
  g as m,
  e as t
};
