// @sberbusiness/triplex-next/chunks/utils-DwX2_kEA.js
import { EHorizontalAlign as l, EVerticalAlign as t, ECellType as n } from "../components/Table/TableBasic/enums.js";
const a = "tableBasic__9bbbed6f", r = "alignLeft__d6e04405", i = "alignRight__d5b02678", o = "alignCenter__73e37c40", s = "verticalAlignBaseline__f32c92d7", g = "verticalAlignSub__78659fbe", T = "verticalAlignSuper__092db58b", _ = "verticalAlignTextTop__77aa10bb", p = "verticalAlignTextBottom__3b9a6e6e", v = "verticalAlignMiddle__07ec3cd2", A = "verticalAlignTop__c1e20726", b = "verticalAlignBottom__ef2df5f8", d = "checkboxType__70617439", B = "thBlock__28bc50c3", u = "order__e0a17335", m = "orderButton__2f805fcb", E = "sorted__bc731cf3", f = "textType__f0bb033e", y = "componentsType__9b762eca", h = "selected__db3f3473", x = "clickable__bca03636", C = "hoverable__9f9ba463", S = "footerEmptyData__1572f39d", k = "spinnerWrapper__c4d51891", L = "overlayCover__740ca339", e = {
  tableBasic: a,
  alignLeft: r,
  alignRight: i,
  alignCenter: o,
  verticalAlignBaseline: s,
  verticalAlignSub: g,
  verticalAlignSuper: T,
  verticalAlignTextTop: _,
  verticalAlignTextBottom: p,
  verticalAlignMiddle: v,
  verticalAlignTop: A,
  verticalAlignBottom: b,
  checkboxType: d,
  thBlock: B,
  order: u,
  orderButton: m,
  sorted: E,
  textType: f,
  componentsType: y,
  selected: h,
  clickable: x,
  hoverable: C,
  footerEmptyData: S,
  spinnerWrapper: k,
  overlayCover: L
}, M = (c = l.LEFT) => {
  switch (c) {
    case l.LEFT:
      return e.alignLeft;
    case l.RIGHT:
      return e.alignRight;
    case l.CENTER:
      return e.alignCenter;
  }
}, N = (c = t.BASELINE) => {
  switch (c) {
    case t.BASELINE:
      return e.verticalAlignBaseline;
    case t.SUB:
      return e.verticalAlignSub;
    case t.SUPER:
      return e.verticalAlignSuper;
    case t.TEXT_TOP:
      return e.verticalAlignTextTop;
    case t.TEXT_BOTTOM:
      return e.verticalAlignTextBottom;
    case t.MIDDLE:
      return e.verticalAlignMiddle;
    case t.TOP:
      return e.verticalAlignTop;
    case t.BOTTOM:
      return e.verticalAlignBottom;
  }
}, R = (c = n.TEXT) => {
  switch (c) {
    case n.TEXT:
      return e.textType;
    case n.COMPONENTS:
      return e.componentsType;
    case n.CHECKBOX:
      return e.checkboxType;
  }
};
export {
  N as a,
  R as b,
  M as m,
  e as s
};
