// @sberbusiness/triplex-next/components/TabsLine/components/TabsLineDesktop.js
import { jsx as p } from "react/jsx-runtime";
import { useState as D, useRef as T, useEffect as N } from "react";
import { TabsLineDropdown as S } from "./TabsLineDropdown.js";
import { TabsLineItem as k } from "./TabsLineItem.js";
import { isKey as u } from "../../../utils/keyboard.js";
import { l as L } from "../../../chunks/vendor-Ba_pyhza.js";
import { s as J } from "../../../chunks/TabsLine.module-DKAp48Sx.js";
const $ = ({
  tabs: l,
  dropdownTargetHtmlAttributes: g,
  selectedId: R,
  onChangeTab: m,
  maxVisible: d,
  size: i
}) => {
  const [f, E] = D([]), [r, y] = D([]), A = T(null), a = T(0), h = T([]);
  N(() => {
    const o = [], e = [];
    l.forEach((n, s) => {
      (d && s + 1 >= d && l.length > d ? o : e).push(n);
    }), L.isEqual(e, f) === !1 && E(e), L.isEqual(o, r) === !1 && y(o);
  }, [l, d]);
  const x = ({ selected: o, onClick: e, onFocus: n, onBlur: s, size: w, ...c }, b) => {
    const K = a.current === b ? 0 : -1, O = (t) => {
      m(c.id), e == null || e(t);
    }, W = (t) => {
      const j = f.length - 1 === a.current;
      if (u(t.code, "ARROW_LEFT") || u(t.code, "ARROW_RIGHT") && !j) {
        const v = u(t.code, "ARROW_RIGHT") ? 1 : -1, z = a.current + v, I = h.current[z];
        I && (t.preventDefault(), I.focus());
      }
    }, _ = (t) => {
      a.current = b, n == null || n(t);
    }, q = (t) => {
      a.current = 0, s == null || s(t);
    }, G = (t) => {
      h.current[b] = t;
    };
    return /* @__PURE__ */ p(
      k,
      {
        selected: R === c.id,
        onClick: O,
        onKeyDown: W,
        tabIndex: K,
        onFocus: _,
        onBlur: q,
        ...c,
        size: w ?? i,
        ref: G
      },
      c.id
    );
  }, H = () => {
    const o = f.map((e, n) => x(e, n));
    if (r.length > 0) {
      const e = r.find((s) => s.id === R), n = e ? e.label : r[0].label;
      o.push(
        /* @__PURE__ */ p(
          S,
          {
            tabs: r,
            active: e !== void 0,
            label: n,
            onClickTab: (s) => m(s.id),
            selected: e,
            targetHtmlAttributes: g,
            size: i
          },
          "TabsLineDropdown"
        )
      );
    }
    return o;
  };
  return l.length ? /* @__PURE__ */ p("div", { className: J.tabsLine, ref: A, "data-size": i, children: H() }) : null;
};
export {
  $ as TabsLineDesktop
};
