// @sberbusiness/triplex-next/components/TabsLine/components/TabsLineDropdown.js
var N = Object.defineProperty;
var R = (c, i, e) => i in c ? N(c, i, { enumerable: !0, configurable: !0, writable: !0, value: e }) : c[i] = e;
var o = (c, i, e) => R(c, typeof i != "symbol" ? i + "" : i, e);
import { jsxs as w, jsx as a } from "react/jsx-runtime";
import m, { createElement as I } from "react";
import { a as T, c as h } from "../../../chunks/vendor-Ba_pyhza.js";
import { Dropdown as v } from "../../Dropdown/Dropdown.js";
import { DropdownListContext as S } from "../../Dropdown/DropdownListContext.js";
import "../../Dropdown/desktop/DropdownDesktop.js";
import { DropdownList as D } from "../../Dropdown/desktop/DropdownList.js";
import "../../../chunks/DropdownListItem-D2hzyBGc.js";
import "../../Dropdown/mobile/DropdownMobile.js";
import "../../Dropdown/mobile/DropdownMobileBody.js";
import "../../Dropdown/mobile/DropdownMobileClose.js";
import "../../Dropdown/mobile/DropdownMobileFooter.js";
import "../../Dropdown/mobile/DropdownMobileHeader.js";
import "../../Dropdown/mobile/DropdownMobileInner.js";
import "../../Dropdown/mobile/DropdownMobileInput.js";
import "../../Dropdown/mobile/DropdownMobileList.js";
import "../../Dropdown/mobile/DropdownMobileListItem.js";
import "../../Loader/LoaderSmall/LoaderSmall.js";
import "../../Loader/LoaderMiddle/LoaderMiddle.js";
import { EComponentSize as k } from "../../../enums/EComponentSize.js";
import { CaretdownStrokeSrvIcon16 as b } from "@sberbusiness/icons-next";
import { isKey as p } from "../../../utils/keyboard.js";
import { s as t } from "../../../chunks/TabsLine.module-DKAp48Sx.js";
import { createSizeToClassNameMap as O } from "../../../utils/classNameMaps.js";
const A = O(t);
class te extends m.PureComponent {
  constructor(e) {
    super(e);
    o(this, "state", {
      activeDescendant: void 0,
      opened: !1
    });
    o(this, "targetRef");
    o(this, "dropdownRef");
    o(this, "instanceId", T());
    /** Установка значения activeDescendant. */
    o(this, "setActiveDescendant", (e) => {
      this.setState({ activeDescendant: e });
    });
    /** Рендер кнопки, раскрывающей список. */
    o(this, "renderTarget", () => {
      const { active: e, label: n, targetHtmlAttributes: r, size: d = k.MD } = this.props, { activeDescendant: s, opened: l } = this.state, f = h(t.tab, A[d], t.dropdownTarget, {
        [t.active]: e
      }), u = h(t.dropdownTargetCaret, { [t.opened]: l });
      return /* @__PURE__ */ a(
        "button",
        {
          ...r,
          className: f,
          onClick: this.handleTargetClick,
          onKeyDown: this.handleTargetKeyDown,
          type: "button",
          role: "tab",
          "aria-haspopup": "menu",
          "aria-expanded": l,
          "aria-controls": this.instanceId,
          "aria-activedescendant": s,
          children: /* @__PURE__ */ a("span", { className: t.tabInner, children: /* @__PURE__ */ w("span", { className: t.dropdownTargetInner, children: [
            n,
            /* @__PURE__ */ a(b, { className: u, paletteIndex: 5 })
          ] }) })
        }
      );
    });
    /** Рендер дропдаун списка. */
    o(this, "renderDropdown", () => {
      const { selected: e, tabs: n } = this.props, { activeDescendant: r, opened: d } = this.state;
      return /* @__PURE__ */ a(
        v,
        {
          className: t.dropdown,
          opened: d,
          setOpened: this.setOpenedDropdown,
          targetRef: this.targetRef,
          ref: this.dropdownRef,
          children: /* @__PURE__ */ a(
            S.Provider,
            {
              value: { activeDescendant: r, setActiveDescendant: this.setActiveDescendant },
              children: /* @__PURE__ */ a(D, { dropdownOpened: d, id: this.instanceId, children: n.map((s) => {
                const { id: l, label: f, showNotificationIcon: u, ...C } = s, g = h(t.dropdownItem, {
                  [t.withNotification]: !!s.showNotificationIcon
                });
                return /* @__PURE__ */ I(
                  D.Item,
                  {
                    ...C,
                    className: g,
                    id: s.id,
                    key: s.id,
                    onSelect: () => {
                      this.handleClickTab(s);
                    },
                    selected: s === e
                  },
                  /* @__PURE__ */ a("span", { className: t.dropdownItemInner, children: s.label }),
                  s.showNotificationIcon && /* @__PURE__ */ a("span", { className: t.notificationIcon })
                );
              }) })
            }
          )
        }
      );
    });
    /** Открывает/закрывает Dropdown. */
    o(this, "setOpenedDropdown", (e) => {
      this.setState({ opened: e });
    });
    /** Обработчик кнопки, раскрывающей список. */
    o(this, "handleTargetClick", () => {
      const { opened: e } = this.state;
      this.setState({ opened: !e });
    });
    /** Обработчик клика по табу. */
    o(this, "handleClickTab", (e) => {
      const { onClickTab: n } = this.props, { opened: r } = this.state;
      n(e), this.setState({ opened: !r });
    });
    /** Обработчик нажатия клавиш по кнопке, раскрывающей список. */
    o(this, "handleTargetKeyDown", (e) => {
      const { opened: n } = this.state;
      n || (p(e.code, "ARROW_UP") || p(e.code, "ARROW_DOWN")) && (e.preventDefault(), this.setState({ opened: !0 })), n && (p(e.code, "TAB") || p(e.code, "ESCAPE")) && this.setState({ opened: !1 });
    });
    /** Обработчик клика вне дропдауна. */
    o(this, "handleClickOutside", (e) => {
      var r, d;
      const { opened: n } = this.state;
      n && !((r = this.targetRef.current) != null && r.contains(e.target)) && !((d = this.dropdownRef.current) != null && d.contains(e.target)) && this.setState({ opened: !1 });
    });
    this.targetRef = m.createRef(), this.dropdownRef = m.createRef();
  }
  componentDidMount() {
    document.addEventListener("mouseup", this.handleClickOutside);
  }
  componentWillUnmount() {
    document.removeEventListener("mouseup", this.handleClickOutside);
  }
  render() {
    return /* @__PURE__ */ w("div", { className: t.tabsLineDropdown, ref: this.targetRef, children: [
      this.renderTarget(),
      this.renderDropdown()
    ] });
  }
}
export {
  te as TabsLineDropdown
};
