// @sberbusiness/triplex-next/components/CheckboxTree/CheckboxTree.js
import { jsx as n } from "react/jsx-runtime";
import { CheckboxTreeExtended as t } from "../CheckboxTreeExtended/CheckboxTreeExtended.js";
import { checkChildrenCheckboxes as f, traverseCheckboxes as g, checkParentCheckboxes as T } from "./utils.js";
const h = ({ checkboxes: e, onChange: N }) => {
  const p = (d) => (r) => {
    const { checked: i } = r.target;
    d.checked = d.bulk ? !0 : i, f(d), g(e, T), N([...e]);
  }, a = (d, r, i) => /* @__PURE__ */ n(
    t.Node,
    {
      id: d.id,
      checkbox: (l) => /* @__PURE__ */ n(
        t.Checkbox,
        {
          ...l,
          onChange: p(d),
          bulk: d.bulk,
          checked: d.checked,
          children: d.label
        }
      ),
      prevNodeId: r == null ? void 0 : r.id,
      nextNodeId: i == null ? void 0 : i.id,
      children: d.children && d.children.map(
        (l, m) => {
          var s, u;
          return a(l, (s = d.children) == null ? void 0 : s[m - 1], (u = d.children) == null ? void 0 : u[m + 1]);
        }
      )
    },
    d.id
  );
  return /* @__PURE__ */ n(t, { children: e.map((d, r) => a(d, e[r - 1], e[r + 1])) });
};
h.displayName = "CheckboxTree";
export {
  h as CheckboxTree
};
