// @sberbusiness/triplex-next/components/LoaderScreen/LoaderScreen.js
import { jsx as e } from "react/jsx-runtime";
import { LoaderSmall as s } from "../Loader/LoaderSmall/LoaderSmall.js";
import { ELoaderSmallTheme as i } from "../Loader/LoaderSmall/enum.js";
import { LoaderMiddle as n } from "../Loader/LoaderMiddle/LoaderMiddle.js";
import { c as p } from "../../chunks/vendor-Ba_pyhza.js";
import { EComponentSize as t } from "../../enums/EComponentSize.js";
const S = "loaderScreen__6b0f1dc2", B = "loaderSmallBackdrop__62dece38", k = "loaderMiddleBackdrop__d1bab17f", r = {
  loaderScreen: S,
  loaderSmallBackdrop: B,
  loaderMiddleBackdrop: k
}, f = ({
  className: o,
  size: a = t.MD,
  type: d,
  theme: l = i.BRAND,
  ...c
}) => {
  const m = p(o, r.loaderScreen, {
    [r.loaderSmallBackdrop]: d === "small",
    [r.loaderMiddleBackdrop]: d === "middle"
  });
  return /* @__PURE__ */ e("div", { className: m, ...c, children: d === "small" ? /* @__PURE__ */ e(s, { size: a, theme: l }) : /* @__PURE__ */ e("div", { className: r.loaderMiddleBackground, children: /* @__PURE__ */ e(n, {}) }) });
};
f.displayName = "LoaderScreen";
export {
  f as LoaderScreen
};
