// @sberbusiness/triplex-next/components/SelectExtendedField/SelectExtendedField.js
import { jsx as y, jsxs as O } from "react/jsx-runtime";
import { useState as T, useRef as u, useCallback as r, useEffect as x } from "react";
import { isKey as K, EVENT_KEY_CODES as b } from "../../utils/keyboard.js";
import { KeyDownListener as C } from "../KeyDownListener/KeyDownListener.js";
import { c as _ } from "../../chunks/vendor-Ba_pyhza.js";
import { SelectExtendedFieldTarget as j } from "./components/SelectExtendedFieldTarget.js";
import { SelectExtendedFieldDropdown as k } from "./components/SelectExtendedFieldDropdown.js";
const v = "SelectExtendedField__1269567f", A = {
  SelectExtendedField: v
}, L = (S) => {
  const { className: w, onKeyDown: n, children: D, renderTarget: F, closeOnTab: f, onClose: o, onOpen: d, ...g } = S, [e, c] = T(!1), l = u(null), m = u(null), E = r((t) => {
    c(t);
  }, []), i = r(
    (t) => {
      var s, p;
      e && !((s = l.current) != null && s.contains(t.target)) && !((p = m.current) != null && p.contains(t.target)) && c(!1);
    },
    [e]
  ), a = r(() => {
    e && c(!1);
  }, [e]), h = r(
    (t) => {
      const s = t.code || t.keyCode;
      f && K(s, "TAB") && a(), n == null || n(t);
    },
    [f, a, n]
  );
  return x(() => (document.addEventListener("mousedown", i), () => {
    document.removeEventListener("mousedown", i);
  }), [i]), x(() => {
    e ? d == null || d() : o == null || o();
  }, [e, d, o]), /* @__PURE__ */ y(C, { onMatch: a, eventKeyCode: b.ESCAPE, children: /* @__PURE__ */ O(
    "div",
    {
      className: _(A.SelectExtendedField, w),
      onKeyDown: h,
      ref: l,
      ...g,
      children: [
        F({ opened: e, setOpened: E }),
        D({
          dropdownRef: m,
          opened: e,
          setOpened: E,
          targetRef: l
        })
      ]
    }
  ) });
}, $ = Object.assign(L, {
  Target: j,
  Dropdown: k
});
export {
  $ as SelectExtendedField
};
