// @sberbusiness/triplex-next/components/Tabs/Tabs.js
import { jsx as c, jsxs as N } from "react/jsx-runtime";
import { useState as a, useRef as D } from "react";
import { TabsExtended as s } from "../TabsExtended/TabsExtended.js";
import "../TabsExtended/components/TabsExtendedContent.js";
import { c as W } from "../../chunks/vendor-Ba_pyhza.js";
import "../TabsExtended/TabsExtendedContext.js";
import "../TabsExtended/components/TabsExtendedTab.js";
import "../TabsExtended/components/TabsExtendedTabButton.js";
import "../TabsExtended/components/TabsExtendedTabContext.js";
import "../Button/Button.js";
import "../Button/ButtonBase.js";
import "../Button/ButtonIcon.js";
import { ButtonDropdown as x } from "../Button/ButtonDropdown.js";
import "../Button/ButtonDropdownExtended.js";
import { EButtonDotsTheme as j } from "../Button/enums.js";
import { EComponentSize as K } from "../../enums/EComponentSize.js";
import { isKey as _ } from "../../utils/keyboard.js";
const k = "tabButtonDropdown__ff5b844f", G = "sm__7fd386fc", H = "md__b2dc47b6", L = "lg__e904da7b", p = {
  tabButtonDropdown: k,
  sm: G,
  md: H,
  lg: L
}, co = ({
  buttonDropdownAttributes: v,
  selectedId: l,
  onSelectTab: R,
  size: i = K.MD,
  tabs: t,
  ...A
}) => {
  const [h, m] = a(l || t[0].id), [w, C] = a(""), [F, S] = a(""), d = D(null), f = D(null), B = ({ dropdownItemsIds: e, onSelectTab: o }) => t.filter((n) => e.includes(n.id)).map((n) => ({ ...n, onSelect: () => o(n.id) })), I = (e, o) => /* @__PURE__ */ c(s.Content.Tab, { ...e, children: ({ selected: n, isFirstInlineTab: g, isLastInlineTab: E }) => {
    const O = h === e.id ? 0 : -1, y = () => {
      let r;
      return w === e.id ? r = d : F === e.id && (r = f), r;
    };
    return /* @__PURE__ */ c(
      s.Content.TabButton,
      {
        selected: n,
        tabIndex: O,
        ref: y(),
        size: i,
        onFocus: () => {
          C(t[o - 1] ? t[o - 1].id : ""), S(t[o + 1] ? t[o + 1].id : "");
        },
        onKeyDown: (r) => {
          var T, b;
          const { key: u } = r;
          _(u, "ARROW_LEFT") ? (g || (m(t[o - 1].id), (T = d.current) == null || T.focus()), r.preventDefault()) : _(u, "ARROW_RIGHT") && (E || (m(t[o + 1].id), (b = f.current) == null || b.focus()), r.preventDefault());
        },
        children: e.label
      }
    );
  } }, o);
  return /* @__PURE__ */ c(s, { ...A, selectedId: l, onSelectTab: R, children: /* @__PURE__ */ N(s.Content, { className: p.tabsContent, size: i, children: [
    /* @__PURE__ */ c(s.Content.TabsWrapper, { children: t.map(I) }),
    /* @__PURE__ */ c(s.Content.DropdownWrapper, { children: ({ dropdownItemsIds: e, onSelectTab: o }) => /* @__PURE__ */ c(
      x,
      {
        theme: j.DOTS_SECONDARY,
        size: i,
        options: B({ dropdownItemsIds: e, onSelectTab: o }),
        selected: t.filter((n) => n.id === l)[0],
        buttonAttributes: {
          ...v,
          className: W(p.tabButtonDropdown, p[i])
        }
      }
    ) })
  ] }) });
};
export {
  co as Tabs
};
