// @sberbusiness/triplex-next/components/MarkerStatus/MarkerStatus.js
import { jsxs as s, jsx as t } from "react/jsx-runtime";
import { Marker as f } from "../Marker/Marker.js";
import { c as k } from "../../chunks/vendor-Ba_pyhza.js";
import { s as d } from "../../chunks/utils-B7AjUpy-.js";
import "../Typography/Title.js";
import { Text as i } from "../Typography/Text.js";
import { Caption as _ } from "../Typography/Caption.js";
import { ETextSize as n, EFontType as c, ECaptionSize as M } from "../Typography/enums.js";
import { EComponentSize as a } from "../../enums/EComponentSize.js";
import { createSizeToClassNameMap as N } from "../../utils/classNameMaps.js";
const x = "markerStatus__05ee2eb0", T = "contentContainer__be1c340a", u = "md__361f55fb", z = "markerContainer__26e63bbc", E = "lg__959c712a", r = {
  markerStatus: x,
  contentContainer: T,
  md: u,
  markerContainer: z,
  lg: E
}, b = {
  [a.MD]: n.B4,
  [a.LG]: n.B3
}, h = N(r), F = ({
  children: p,
  className: l,
  description: o,
  status: m,
  size: e = a.MD,
  ...C
}) => {
  const S = k(r.markerStatus, h[e], d[m], l);
  return /* @__PURE__ */ s("div", { className: S, "data-tx": "1.4.0", ...C, children: [
    /* @__PURE__ */ t("div", { className: r.markerContainer, children: /* @__PURE__ */ t(f, { status: m, size: e }) }),
    /* @__PURE__ */ s("div", { className: r.contentContainer, children: [
      /* @__PURE__ */ t(i, { size: b[e], children: p }),
      o && (e === a.MD ? /* @__PURE__ */ t(_, { size: M.C1, type: c.SECONDARY, children: o }) : /* @__PURE__ */ t(i, { size: n.B4, type: c.SECONDARY, children: o }))
    ] })
  ] });
};
export {
  F as MarkerStatus
};
