// @sberbusiness/triplex-next/components/Radio/Radio.js
import { jsxs as f, jsx as t } from "react/jsx-runtime";
import N from "react";
import { c as m } from "../../chunks/vendor-Ba_pyhza.js";
import "../Typography/Title.js";
import { Text as x } from "../Typography/Text.js";
import "../Typography/Caption.js";
import { ETextSize as n } from "../Typography/enums.js";
import { EComponentSize as i } from "../../enums/EComponentSize.js";
import { createSizeToClassNameMap as z } from "../../utils/classNameMaps.js";
const y = "label__c3fe9303", T = "md__21a37875", M = "nonempty__6261ba27", u = "lg__1f2875fa", I = "disabled__be8cd257", R = "radio__dcefb078", S = "radioIcon__19939bac", a = {
  label: y,
  md: T,
  nonempty: M,
  lg: u,
  disabled: I,
  radio: R,
  radioIcon: S
}, g = {
  [i.LG]: n.B2,
  [i.MD]: n.B3
}, C = z(a), h = N.forwardRef((r, d) => {
  const { children: s, className: l, disabled: c, labelAttributes: o, size: e = i.MD, ...p } = r, _ = m(a.radio, l, C[e]), b = m(
    a.label,
    a[e],
    { [a.disabled]: !!c, [a.nonempty]: !!s },
    o == null ? void 0 : o.className
  );
  return /* @__PURE__ */ f("label", { ...o, className: b, "data-tx": "1.4.0", children: [
    /* @__PURE__ */ t("input", { type: "radio", className: _, disabled: c, ...p, ref: d }),
    /* @__PURE__ */ t("span", { className: a.radioIcon }),
    s && /* @__PURE__ */ t(x, { size: g[e], children: s })
  ] });
});
h.displayName = "Radio";
export {
  h as Radio
};
