// @sberbusiness/triplex-next/components/SMSInput/SMSInput.js
import { jsx as s } from "react/jsx-runtime";
import { c as C } from "../../chunks/vendor-Ba_pyhza.js";
import { useState as e } from "react";
import { SMSInputContext as h } from "./SMSInputContext.js";
import { SMSInputInput as N } from "./components/SMSInputInput.js";
import { SMSInputRefresh as x } from "./components/SMSInputRefresh.js";
import { SMSInputSubmit as T } from "./components/SMSInputSubmit.js";
import { SMSInputTooltip as v } from "./components/SMSInputTooltip.js";
import { s as r } from "../../chunks/SMSInput.module-Ci4GRnk_.js";
import { createSizeToClassNameMap as z } from "../../utils/classNameMaps.js";
const y = z(r), t = (i) => {
  const { children: m, className: p, code: a, disabled: n, error: S, onChangeCode: l, onSubmitCode: u, size: o, ...d } = i, [c, I] = e(!0), [f, b] = e(), M = C(r.smsInput, p);
  return /* @__PURE__ */ s(
    h.Provider,
    {
      value: {
        code: a,
        disabled: !!n,
        disabledSubmit: c,
        error: !!S,
        onChangeCode: l,
        onSubmitCode: u,
        setDisabledSubmit: I,
        setTooltipId: b,
        size: o,
        sizeClassName: y[o],
        tooltipId: f
      },
      children: /* @__PURE__ */ s("div", { className: M, ...d, "data-tx": "1.4.0", children: m })
    }
  );
};
t.displayName = "SMSInput";
t.Tooltip = v;
t.Refresh = x;
t.Input = N;
t.Submit = T;
export {
  t as SMSInput
};
