// @sberbusiness/triplex-next/components/AmountField/AmountBaseInputFormatter.js
var a = Object.defineProperty;
var o = (r, t, i) => t in r ? a(r, t, { enumerable: !0, configurable: !0, writable: !0, value: i }) : r[t] = i;
var h = (r, t, i) => o(r, typeof t != "symbol" ? t + "" : t, i);
import { AmountConst as n } from "../../consts/AmountConst.js";
class m {
  constructor(t, i) {
    /** Значение. */
    h(this, "value");
    /** Величина сдвига каретки. */
    h(this, "caretOffset");
    /** Максимальное количество знаков (без учёта пробелов). */
    h(this, "maxLength");
    /** Количество чисел после запятой. */
    h(this, "fractionDigits");
    this.value = "", this.caretOffset = 0, this.maxLength = t, this.fractionDigits = i;
  }
  /** Получение значения. */
  getValue() {
    return this.value;
  }
  /** Получение величины сдвига каретки. */
  getCaretOffset() {
    return this.caretOffset;
  }
  /** Применение входных данные для форматирования. */
  apply(t) {
    const i = t.length;
    this.fractionDigits == 0 ? this.value = this.formatInteger(t, i) : this.value = this.formatDecimal(t, i);
  }
  /** Форматирование значения в виде целого значения. */
  formatInteger(t, i) {
    const s = [];
    if (i > 0) {
      let e = 0;
      for (s.push(t[e]); ++e < i; )
        (i - e) % 3 == 0 && (s.push(" "), this.caretOffset++), s.push(t[e]);
    }
    return s.join("");
  }
  /** Форматирование значения в виде десятичной дроби. */
  formatDecimal(t, i) {
    const s = [];
    if (i) {
      const e = i - this.fractionDigits - 1;
      let f = 0;
      for (s.push(t[f]); ++f < e; )
        (e - f) % 3 == 0 && (s.push(" "), this.caretOffset++), s.push(t[f]);
      for (s.push(n.DecimalComma); ++f < i; )
        s.push(t[f]);
    }
    return s.join("");
  }
}
export {
  m as AmountBaseInputFormatter
};
