// @sberbusiness/triplex-next/components/AmountField/AmountField.js
import { jsxs as d, jsx as l } from "react/jsx-runtime";
import R, { useRef as y, useLayoutEffect as x } from "react";
import { FormGroup as E } from "../FormGroup/FormGroup.js";
import { FormField as I } from "../FormField/FormField.js";
import { FormFieldDescription as T } from "../FormField/components/FormFieldDescription.js";
import { FormFieldLabel as V } from "../FormField/components/FormFieldLabel.js";
import { FormFieldPostfix as D } from "../FormField/components/FormFieldPostfix.js";
import { FormFieldClear as K } from "../FormField/components/FormFieldClear.js";
import { F as b } from "../../chunks/FormFieldInput-CO7yZP0Q.js";
import { AmountBaseInputCore as j } from "./AmountBaseInputCore.js";
import { setCaretPosition as A } from "../../utils/inputUtils.js";
import { createPlaceholder as L, setFallbackCaret as g } from "./utils.js";
const M = "currency__67ab2406", $ = {
  currency: M
}, Z = R.forwardRef(
  ({
    label: f,
    description: i,
    currency: h,
    postfix: s,
    inputProps: t,
    maxIntegerDigits: m = 16,
    fractionDigits: n = 2,
    ...F
  }, o) => {
    const u = F["data-test-id"], C = t.placeholder || L(n), a = y(null), e = y();
    e.current === void 0 && (e.current = new j(m, n)), x(() => {
      e.current && a.current == document.activeElement && A(a.current, Math.max(e.current.caret, 0));
    }, [t.value]);
    const _ = () => e.current ? ((t.value != e.current.value || m != e.current.maxIntegerDigits || n != e.current.fractionDigits) && (e.current.maxIntegerDigits = m, e.current.fractionDigits = n, e.current.apply(t.value, t.value.length)), e.current.cache.formattedValue = e.current.formattedValue, e.current.formattedValue) : "", S = (r) => {
      if (!e.current) return;
      const c = r.target.selectionStart ?? r.target.value.length;
      e.current.apply(r.target.value, c), g(r.target, e.current, n), t.onChange(e.current.value);
    }, k = (r) => {
      var c;
      e.current && (e.current.cache.key = r.key, (c = t.onKeyDown) == null || c.call(t, r));
    }, v = (r) => {
      var c;
      e.current && (e.current.cache.key = "", e.current.cache.selectionStart = r.currentTarget.selectionStart, e.current.cache.selectionEnd = r.currentTarget.selectionEnd, e.current.cache.selectionDirection = r.currentTarget.selectionDirection, (c = t.onSelect) == null || c.call(t, r));
    }, w = () => {
      !e.current || !a.current || (e.current.apply("", 0), g(a.current, e.current, n), t.onChange(e.current.value));
    };
    return /* @__PURE__ */ d(E, { children: [
      /* @__PURE__ */ d(I, { ...F, children: [
        f ? /* @__PURE__ */ l(V, { children: f }) : null,
        /* @__PURE__ */ l(
          b,
          {
            ...t,
            ref: (r) => {
              a.current = r, typeof o == "function" ? o(r) : o && (o.current = r);
            },
            value: _(),
            onChange: S,
            onKeyDown: k,
            onSelect: v,
            autoComplete: "off",
            inputMode: "decimal",
            "data-test-id": u && `${u}__input`,
            placeholder: C
          }
        ),
        /* @__PURE__ */ d(D, { children: [
          /* @__PURE__ */ l(K, { onClick: w }),
          h ? /* @__PURE__ */ l("span", { className: $.currency, "data-test-id": u && `${u}__unit`, children: h }) : null,
          s || null
        ] })
      ] }),
      i ? /* @__PURE__ */ l(T, { children: i }) : null
    ] });
  }
);
export {
  Z as AmountField
};
