// @sberbusiness/triplex-next/components/AmountField/AmountBaseInputParser.js
var p = Object.defineProperty;
var m = (a, t, s) => t in a ? p(a, t, { enumerable: !0, configurable: !0, writable: !0, value: s }) : a[t] = s;
var n = (a, t, s) => m(a, typeof t != "symbol" ? t + "" : t, s);
import { AmountConst as c } from "../../consts/AmountConst.js";
import { StringUtils as o } from "../../utils/stringUtils.js";
import { isKey as D } from "../../utils/keyboard.js";
class O {
  constructor(t, s) {
    /** Значение. */
    n(this, "value");
    /** Величина сдвига каретки. */
    n(this, "caretOffset");
    /** Значение нажатой клавиши. */
    n(this, "key");
    /** Максимальное количество знаков перед запятой. */
    n(this, "maxIntegerDigits");
    /** Количество чисел после запятой. */
    n(this, "fractionDigits");
    this.value = "", this.caretOffset = 0, this.maxIntegerDigits = t, this.fractionDigits = s, this.key = "";
  }
  /** Получение значения. */
  getValue() {
    return this.value;
  }
  /** Получение величины сдвига каретки. */
  getCaretOffset() {
    return this.caretOffset;
  }
  /** Применение входных данных для обработки. */
  apply(t, s, r) {
    const i = t.length;
    if (this.key = r, this.maxIntegerDigits > 0) {
      if (this.fractionDigits == 0)
        return this.parseInteger(t, s, i);
      if (this.fractionDigits > 0)
        return this.parseDecimal(t, s, i);
    }
    this.value = "", this.caretOffset -= i;
  }
  /** Обработка значения в виде целого числа. */
  parseInteger(t, s, r) {
    this.parseIntegerPart(t, s, r);
  }
  /** Обработка значения в виде десятичной дроби. */
  parseDecimal(t, s, r) {
    const i = this.findSeparatorIndex(t, s, r), [e, f] = i != -1 ? [i, i + 1] : [s, s], [h, g] = [t.substring(0, e), t.substring(f)];
    this.parseIntegerPart(h, Math.min(s, e), e), this.parseFractionalPart(g, Math.max(s - f, 0), r - f);
  }
  /** Обработка целой части числа. */
  parseIntegerPart(t, s, r) {
    const i = [];
    if ([t, s, r] = this.trimLeadingZeros(t, s, r), s > 0) {
      const e = this.maxIntegerDigits - (r - s - Math.floor((r - s) / 4));
      if (e > 0)
        for (let f = 0, h = 0; f < s && !(o.isDigit(t[f]) && (i.push(t[f]), ++h == e)); f++)
          ;
      this.caretOffset -= s - i.length;
    }
    if (s < r)
      for (let e = s; e < r; e++)
        o.isDigit(t[e]) ? i.push(t[e]) : this.caretOffset--;
    this.value = i.join("");
  }
  /** Обработка дробной части числа. */
  parseFractionalPart(t, s, r) {
    const i = [];
    if (s > 0) {
      const e = this.fractionDigits;
      for (let f = 0, h = 0; f < s; f++)
        if (o.isDigit(t[f])) {
          if (i.push(t[f]), ++h == e)
            break;
        } else
          this.caretOffset--;
    }
    for (; i.length < this.fractionDigits; ) {
      const e = r - (this.fractionDigits - i.length);
      e >= s ? i.push(t[e]) : (i.push("0"), D(this.key, "DELETE") && this.caretOffset++);
    }
    if (this.value.length == 0)
      if (i.some((e) => e != "0"))
        this.value = "0", this.caretOffset++;
      else
        return;
    this.value += c.DecimalPoint, this.value += i.join("");
  }
  /** Удаление ведущих нулей и всего, что им предшествует. */
  trimLeadingZeros(t, s, r) {
    let i = 0;
    for (let e = 0; e < r && !(o.isDigit(t[e]) && (i = e, t[e] != "0")); e++)
      ;
    return this.caretOffset -= i, [t.substring(i), Math.max(s - i, 0), r - i];
  }
  /** Нахождение индекса десятичного разделителя. */
  findSeparatorIndex(t, s, r) {
    if (s < r - (this.fractionDigits + 1)) {
      for (let i = r - 1; i >= 0; i--)
        if (o.isDecimalSeparator(t[i]))
          return i;
    } else
      for (let i = 0; i < r; i++)
        if (o.isDecimalSeparator(t[i]))
          return i;
    return -1;
  }
}
export {
  O as AmountBaseInputParser
};
