// @sberbusiness/triplex-next/components/List/components/ListSortableItem.js
import { jsx as b } from "react/jsx-runtime";
import p from "react";
import { x as u, C as I, c as _ } from "../../../chunks/vendor-Ba_pyhza.js";
import { ListItem as y } from "./ListItem.js";
import { ListSortableItemTarget as L } from "./ListSortableItemTarget.js";
const R = "listSortableItem__18c325b0", x = "dragging__382d516f", i = {
  listSortableItem: R,
  dragging: x
}, N = Object.assign(
  p.forwardRef(function({ children: e, className: n, style: g, disabled: r, ...s }, t) {
    const { transform: m, transition: l, listeners: f, isDragging: a, setNodeRef: c, setActivatorNodeRef: S } = u({
      disabled: r,
      id: s.id,
      transition: {
        duration: 300,
        easing: "ease-out"
      }
    }), d = (o) => {
      c(o), typeof t == "function" ? t(o) : t && (t.current = o);
    };
    return /* @__PURE__ */ b(
      y,
      {
        className: _(i.listSortableItem, { [i.dragging]: a }, n),
        style: { transform: I.Translate.toString(m), transition: l, ...g },
        ...s,
        ref: d,
        children: typeof e == "function" ? e({ disabled: r, dragging: a, listeners: f, setActivatorNodeRef: S }) : e
      }
    );
  }),
  {
    Target: L
  }
);
N.displayName = "ListSortableItem";
export {
  N as ListSortableItem
};
