// @sberbusiness/triplex-next/components/List/components/ListItemControlsButtonDropdown.js
import { jsx as o, jsxs as w, Fragment as _ } from "react/jsx-runtime";
import P, { useRef as m, useState as E, createElement as b } from "react";
import { a as O, c as g } from "../../../chunks/vendor-Ba_pyhza.js";
import { ETextSize as K } from "../../Typography/enums.js";
import { Text as T } from "../../Typography/Text.js";
import { ButtonDropdownExtended as R } from "../../Button/ButtonDropdownExtended.js";
import { isKey as i } from "../../../utils/keyboard.js";
import { DropdownListContext as j } from "../../Dropdown/DropdownListContext.js";
import "../../Dropdown/mobile/DropdownMobile.js";
import { DropdownMobileBody as z } from "../../Dropdown/mobile/DropdownMobileBody.js";
import { DropdownMobileClose as U } from "../../Dropdown/mobile/DropdownMobileClose.js";
import "../../Dropdown/mobile/DropdownMobileFooter.js";
import { DropdownMobileHeader as q } from "../../Dropdown/mobile/DropdownMobileHeader.js";
import "../../Dropdown/mobile/DropdownMobileInner.js";
import "../../Dropdown/mobile/DropdownMobileInput.js";
import { DropdownMobileList as F } from "../../Dropdown/mobile/DropdownMobileList.js";
import { DropdownMobileListItem as H } from "../../Dropdown/mobile/DropdownMobileListItem.js";
import "../../Loader/LoaderSmall/LoaderSmall.js";
import "../../Loader/LoaderMiddle/LoaderMiddle.js";
import { DropdownList as h } from "../../Dropdown/desktop/DropdownList.js";
import { s as y, L as V } from "../../../chunks/ListItemControlsButton-BoEiexyr.js";
const G = P.forwardRef(
  (C, l) => {
    const { buttonAttributes: x, children: c, className: I, icon: N, options: d, selected: n, disabled: B, ...L } = C, f = m(null), u = m(null), [p, v] = E(), D = m(O()), A = ({ opened: e, setOpened: t }) => /* @__PURE__ */ o(
      V,
      {
        className: "hoverable",
        onKeyDown: k({ opened: e, setOpened: t }),
        onClick: S({ opened: e, setOpened: t }),
        disabled: B,
        "aria-haspopup": "menu",
        "aria-expanded": e,
        "aria-controls": D.current,
        "aria-activedescendant": p,
        icon: N,
        ...x,
        ref: W,
        children: c
      }
    ), S = ({ opened: e, setOpened: t }) => () => t(!e), k = ({ opened: e, setOpened: t }) => (a) => {
      const { key: r } = a;
      (i(r, "SPACE") || i(r, "ARROW_UP") || i(r, "ARROW_DOWN")) && a.preventDefault(), !e && (i(r, "ARROW_UP") || i(r, "ARROW_DOWN")) && t(!0);
    }, M = ({ opened: e, setOpened: t, className: a }) => /* @__PURE__ */ o(j.Provider, { value: { activeDescendant: p, setActiveDescendant: v }, children: /* @__PURE__ */ o(
      R.Dropdown,
      {
        className: a,
        opened: e,
        setOpened: t,
        targetRef: f,
        ref: u,
        mobileViewProps: {
          children: /* @__PURE__ */ w(_, { children: [
            /* @__PURE__ */ w(q, { children: [
              /* @__PURE__ */ o(T, { tag: "div", size: K.B1, children: c }),
              /* @__PURE__ */ o(U, { onClick: () => t(!1) })
            ] }),
            /* @__PURE__ */ o(z, { children: /* @__PURE__ */ o(F, { children: d.map((r) => /* @__PURE__ */ b(
              H,
              {
                ...r,
                key: r.id,
                selected: r.id === (n == null ? void 0 : n.id),
                onSelect: () => {
                  var s;
                  (s = r.onSelect) == null || s.call(r), t(!1);
                }
              },
              r.label
            )) }) })
          ] })
        },
        children: /* @__PURE__ */ o(h, { dropdownOpened: e, id: D.current, children: d.map((r) => /* @__PURE__ */ b(
          h.Item,
          {
            ...r,
            className: y.buttonDropdownMenuItem,
            key: r.id,
            selected: r.id === (n == null ? void 0 : n.id),
            onSelect: () => {
              var s;
              (s = r.onSelect) == null || s.call(r), t(!1);
            }
          },
          r.label
        )) })
      }
    ) }), W = (e) => {
      f.current = e, typeof l == "function" ? l(e) : l && (l.current = e);
    };
    return /* @__PURE__ */ o(
      R,
      {
        className: g(y.listItemControlsButtonDropdown, I),
        renderButton: A,
        renderDropdown: M,
        dropdownRef: u,
        closeOnTab: !0,
        ...L
      }
    );
  }
);
G.displayName = "ListItemControlsButtonDropdown";
export {
  G as ListItemControlsButtonDropdown
};
