// @sberbusiness/triplex-next/components/List/components/ListItemSelectable.js
import { jsxs as n, jsx as t } from "react/jsx-runtime";
import p, { useContext as m, useEffect as k } from "react";
import { c as x } from "../../../chunks/vendor-Ba_pyhza.js";
import { Checkbox as d } from "../../Checkbox/Checkbox.js";
import { ListItemContext as _ } from "./ListItemContext.js";
const f = "listItemSelectable__92c53029", L = "selected__4e0411a2", C = "checkboxWrapper__f923c619", W = "childrenWrapper__01a2be1b", I = "checkboxLabel__ca7c9433", S = "checkboxLabelClickArea__98905eba", e = {
  listItemSelectable: f,
  selected: L,
  checkboxWrapper: C,
  childrenWrapper: W,
  checkboxLabel: I,
  checkboxLabelClickArea: S
}, N = p.forwardRef(
  ({ selected: c, children: l, className: r, onSelect: s, ...o }, b) => {
    const { setSelected: a } = m(_), i = (h) => {
      s(h.target.checked);
    };
    return k(() => {
      a(c);
    }, [c, a]), /* @__PURE__ */ n(
      "div",
      {
        className: x(e.listItemSelectable, { [e.selected]: c }, r),
        ...o,
        ref: b,
        children: [
          /* @__PURE__ */ t("div", { className: e.childrenWrapper, children: l }),
          /* @__PURE__ */ t("div", { className: e.checkboxWrapper, children: /* @__PURE__ */ t(
            d,
            {
              checked: c,
              onChange: i,
              labelAttributes: { className: e.checkboxLabel },
              children: /* @__PURE__ */ t("span", { className: e.checkboxLabelClickArea })
            }
          ) })
        ]
      }
    );
  }
);
N.displayName = "ListItemSelectable";
export {
  N as ListItemSelectable
};
