// @sberbusiness/triplex-next/components/List/components/ListSortableItemTarget.js
import { jsxs as i, jsx as e } from "react/jsx-runtime";
import m from "react";
import { DragdotsStrokeSrvIcon24 as n } from "@sberbusiness/icons-next";
import { c as I } from "../../../chunks/vendor-Ba_pyhza.js";
const c = "listSortableItemTarget__88432b5a", b = "dragging__382d516f", S = "listSortableItemTargetContent__25581799", d = "listSortableItemTargetIcon__7bb0dc8d", t = {
  listSortableItemTarget: c,
  dragging: b,
  listSortableItemTargetContent: S,
  listSortableItemTargetIcon: d
}, T = m.forwardRef(
  ({ children: a, className: r, disabled: o, dragging: l, ...s }, g) => /* @__PURE__ */ i(
    "div",
    {
      className: I(t.listSortableItemTarget, { [t.dragging]: l }, "hoverable", r),
      ...s,
      ref: g,
      children: [
        /* @__PURE__ */ e("div", { className: t.listSortableItemTargetContent, children: a }),
        !o && /* @__PURE__ */ e(n, { paletteIndex: 5, className: t.listSortableItemTargetIcon })
      ]
    }
  )
);
T.displayName = "ListSortableItemTarget";
export {
  T as ListSortableItemTarget
};
