// @sberbusiness/triplex-next/components/Tooltip/enums.js
var u = /* @__PURE__ */ ((r) => (r.H = "h", r.W = "w", r))(u || {}), v = /* @__PURE__ */ ((r) => (r.X = "x", r.Y = "y", r))(v || {}), n = /* @__PURE__ */ ((r) => (r.X = "x2", r.Y = "y2", r))(n || {}), O = /* @__PURE__ */ ((r) => (r.MAIN = "main", r.CROSS = "cross", r))(O || {}), L = /* @__PURE__ */ ((r) => (r.COLUMN = "column", r.ROW = "row", r))(L || {}), R = /* @__PURE__ */ ((r) => (r.SIDE = "side", r.STANDING = "standing", r.FLOW = "flow", r))(R || {}), I = /* @__PURE__ */ ((r) => (r.ABOVE = "above", r.BELOW = "below", r.RIGHT = "right", r.LEFT = "left", r))(I || {}), N = /* @__PURE__ */ ((r) => (r.END = "end", r.CENTER = "center", r.START = "start", r))(N || {}), W = /* @__PURE__ */ ((r) => (r.UP = "up", r.DOWN = "down", r.LEFT = "left", r.RIGHT = "right", r))(W || {}), a = /* @__PURE__ */ ((r) => (r.SM = "sm", r.LG = "lg", r))(a || {});
export {
  N as ETooltipAlign,
  O as ETooltipAxesType,
  W as ETooltipDirection,
  n as ETooltipEndCoordinates,
  L as ETooltipFlowTypes,
  I as ETooltipPreferPlace,
  a as ETooltipSize,
  u as ETooltipSizeParameter,
  v as ETooltipStartCoordinates,
  R as ETooltipTypeName
};
