// @sberbusiness/triplex-next/components/Tooltip/Tooltip.js
import { jsx as n } from "react/jsx-runtime";
import d, { useState as E, useRef as X, useEffect as _, useCallback as v } from "react";
import { TooltipContext as W } from "./TootlipContext.js";
import { MobileView as x } from "../MobileView/MobileView.js";
import { TooltipDesktop as D } from "./components/desktop/TooltipDesktop.js";
import { TooltipMobile as L } from "./components/mobile/TooltipMobile.js";
import { TooltipBody as T } from "./components/common/TooltipBody.js";
import { TooltipLink as y } from "./components/common/TooltipLink.js";
import { TooltipTarget as b } from "./components/common/TooltipTarget.js";
import { TooltipXButton as M } from "./components/common/TooltipXButton.js";
import { TooltipMobileHeader as k } from "./components/mobile/components/TooltipMobileHeader.js";
import { useTooltipTheme as V } from "./utils/useTooltipTheme.js";
import { useMatchMedia as j } from "../MediaWidth/useMatchMedia.js";
import { EScreenWidth as c } from "../../helpers/breakpoints.js";
import "./enums.js";
const t = ({
  children: p,
  toggleType: f,
  preferPlace: B,
  disableAdaptiveMode: a,
  isOpen: r,
  toggle: l,
  ...s
}) => {
  const C = j(
    `(max-width: ${c.SM_MAX})`,
    window.innerWidth <= parseInt(c.SM_MAX)
  ), [O, S] = E(!1), m = X(!1), i = r ?? O;
  V(i, C && !a ? document.body : s.renderContainer ?? document.body), _(() => {
    r === !1 && (m.current = !1);
  }, [r]);
  const H = v(() => {
    const o = {
      body: null,
      link: null,
      closeButton: null,
      mobileHeader: null,
      target: null
    };
    return d.Children.map(p, (e) => {
      d.isValidElement(e) && (e.type === b ? o.target = e : e.type === T ? o.body = e : e.type === y ? o.link = e : e.type === M ? o.closeButton = e : e.type === k && (o.mobileHeader = e));
    }), o;
  }, [p]), R = (o) => {
    r === void 0 && (o || (m.current = !1), S(o)), l == null || l(o);
  }, u = () => /* @__PURE__ */ n(D, { isOpen: i, toggleType: f, preferPlace: B, ...s }), w = () => /* @__PURE__ */ n(L, { isOpen: i, ...s });
  return /* @__PURE__ */ n(
    W.Provider,
    {
      value: {
        elements: H(),
        setTooltipOpen: R,
        targetHoveredRef: m,
        toggleType: f,
        tooltipOpen: i
      },
      children: a ? u() : /* @__PURE__ */ n(x, { fallback: u(), children: w() })
    }
  );
};
t.Target = b;
t.Body = T;
t.Link = y;
t.XButton = M;
t.MobileHeader = k;
export {
  t as Tooltip
};
