// @sberbusiness/triplex-next/components/SuggestField/SuggestFieldTarget.js
import { jsxs as B, jsx as o } from "react/jsx-runtime";
import { FormField as E } from "../FormField/FormField.js";
import "../FormField/components/FormFieldDescription.js";
import "../FormField/components/FormFieldCounter.js";
import { FormFieldLabel as p } from "../FormField/components/FormFieldLabel.js";
import "../FormField/components/FormFieldClear.js";
import "../FormField/components/FormFieldTarget.js";
import { F as d } from "../../chunks/FormFieldInput-CO7yZP0Q.js";
import "../FormField/components/FormFieldMaskedInput.js";
import "../FormField/components/FormFieldPostfix.js";
import { FormFieldPrefix as l } from "../FormField/components/FormFieldPrefix.js";
import "../FormField/components/FormFieldTextarea.js";
import { EFormFieldStatus as O } from "../FormField/enums.js";
import { SuggestFieldTargetPostfix as n } from "./SuggestFieldTargetPostfix.js";
import { DataTestId as $ } from "../../consts/DataTestId.js";
const w = ({
  size: t,
  status: i,
  inputValue: c,
  label: F,
  placeholder: f,
  "aria-controls": g,
  "aria-activedescendant": u,
  dataTestId: r,
  loading: x,
  dropdownOpen: A,
  onFocus: v,
  onBlur: b,
  onClick: S,
  onChange: P,
  onClear: j,
  renderInput: m,
  renderLabel: e,
  renderPrefix: a,
  renderPostfix: s,
  ...D
}) => {
  const I = a === void 0 ? l : a, L = e === void 0 ? p : e, T = m === void 0 ? d : m, h = s === void 0 ? n : s;
  return /* @__PURE__ */ B(E, { size: t, status: i, ...D, children: [
    /* @__PURE__ */ o(I, {}),
    /* @__PURE__ */ o(L, { children: F }),
    /* @__PURE__ */ o(
      T,
      {
        value: c,
        placeholder: f,
        role: "combobox",
        "aria-controls": g,
        "aria-activedescendant": u,
        "data-test-id": r && `${r}${$.Suggest.input}`,
        disabled: i === O.DISABLED,
        onFocus: v,
        onBlur: b,
        onClick: S,
        onChange: P
      }
    ),
    /* @__PURE__ */ o(h, { size: t, loading: x, onClear: j })
  ] });
}, X = Object.assign(w, {
  Input: d,
  Label: p,
  Prefix: l,
  Postfix: n
});
export {
  X as SuggestFieldTarget,
  w as SuggestFieldTargetBase
};
