// @sberbusiness/triplex-next/components/SuggestField/SuggestField.js
import "./desktop/SuggestFieldDesktopDropdown.js";
import { SuggestFieldDesktop as L } from "./desktop/SuggestFieldDesktop.js";
import { SuggestFieldMobile as O } from "./mobile/SuggestFieldMobile.js";
import "./mobile/SuggestFieldMobileDropdown.js";
import "../../chunks/SuggestFieldMobileDropdownHint-CMKMF0PQ.js";
import "./mobile/SuggestFieldMobileTarget.js";
import { jsx as e } from "react/jsx-runtime";
import { MobileView as h } from "../MobileView/MobileView.js";
import { SuggestFieldTarget as j } from "./SuggestFieldTarget.js";
const k = (t) => {
  const {
    status: r,
    size: o,
    value: i,
    label: n,
    loading: l,
    options: g,
    placeholder: a,
    tooltipHint: p,
    tooltipOpen: s,
    dropdownListLoading: d,
    clearInputOnFocus: u,
    onSelect: m,
    onFilter: c,
    onScrollEnd: T,
    onTargetInputFocus: f,
    onTargetInputBlur: F,
    renderTarget: S,
    renderTargetInput: b,
    renderTargetLabel: x,
    renderTargetPostfix: I,
    renderTargetPrefix: w
  } = t;
  return /* @__PURE__ */ e(h, { fallback: /* @__PURE__ */ e(L, { ...t }), children: /* @__PURE__ */ e(
    O,
    {
      status: r,
      size: o,
      value: i,
      options: g,
      label: n,
      placeholder: a,
      dropdownHint: s ? p : "",
      loading: l,
      dropdownListLoading: d,
      clearInputOnFocus: u,
      onSelect: m,
      onFilter: c,
      onScrollEnd: T,
      onTargetInputFocus: f,
      onTargetInputBlur: F,
      renderTarget: S,
      renderTargetInput: b,
      renderTargetLabel: x,
      renderTargetPrefix: w,
      renderTargetPostfix: I
    }
  ) });
}, q = Object.assign(k, { Target: j });
export {
  q as SuggestField
};
