// @sberbusiness/triplex-next/components/Header/index.js
import { Header as o } from "./Header.js";
import { HeaderLayoutSidebar as a } from "./components/HeaderLayoutSidebar/HeaderLayoutSidebar.js";
import { HeaderLayoutSidebarContent as f } from "./components/HeaderLayoutSidebar/HeaderLayoutSidebarContent.js";
import { HeaderLayoutSidebarSidebar as p } from "./components/HeaderLayoutSidebar/HeaderLayoutSidebarSidebar.js";
import { HeaderSubheader as H } from "./components/HeaderSubheader/HeaderSubheader.js";
import { HeaderTabs as n } from "./components/HeaderTabs/HeaderTabs.js";
import { HeaderTabsContent as T } from "./components/HeaderTabs/HeaderTabsContent.js";
import { HeaderTabsControls as s } from "./components/HeaderTabs/HeaderTabsControls.js";
import { HeaderTitle as S } from "./components/HeaderTitle/HeaderTitle.js";
import { HeaderTitleContent as y } from "./components/HeaderTitle/HeaderTitleContent.js";
import { HeaderTitleControls as h } from "./components/HeaderTitle/HeaderTitleControls.js";
export {
  o as Header,
  a as HeaderLayoutSidebar,
  f as HeaderLayoutSidebarContent,
  p as HeaderLayoutSidebarSidebar,
  H as HeaderSubheader,
  n as HeaderTabs,
  T as HeaderTabsContent,
  s as HeaderTabsControls,
  S as HeaderTitle,
  y as HeaderTitleContent,
  h as HeaderTitleControls
};
