// @sberbusiness/triplex-next/components/Suggest/Suggest.js
import { SuggestContext as J } from "./SuggestContext.js";
import { jsx as l } from "react/jsx-runtime";
import N, { useState as m, useId as Q, useRef as d, useCallback as r, useEffect as x } from "react";
import { isKey as T } from "../../utils/keyboard.js";
const U = ({
  children: C,
  value: n,
  options: D,
  size: W,
  placeholder: I,
  noOptionsText: L,
  loading: O,
  dropdownListLoading: A,
  tooltipOpen: P,
  clearInputOnFocus: _,
  onKeyDown: e,
  onSelect: g,
  onFilter: R,
  onScrollEnd: j,
  ...k
}, s) => {
  const [i, f] = m((n == null ? void 0 : n.label) || ""), [z, B] = m(), [o, a] = m(!1), F = Q(), p = d(null), S = d(null), u = d(() => {
  }), c = d(() => {
  }), M = r(
    (t) => {
      i !== t && f(t);
    },
    [i]
  );
  u.current = M, x(() => {
    u.current((n == null ? void 0 : n.label) || "");
  }, [n]);
  const b = r(
    (t = (n == null ? void 0 : n.label) || "") => {
      u.current(t), B(void 0), a(!1);
    },
    [n]
  );
  c.current = b, x(() => {
    const t = (h) => {
      var w, E;
      !((w = p.current) != null && w.contains(h.target)) && !((E = S.current) != null && E.contains(h.target)) && c.current();
    };
    if (o)
      return document.addEventListener("mousedown", t), () => {
        document.removeEventListener("mousedown", t);
      };
  }, [o]);
  const q = r(
    (t) => {
      o === !0 && T(t.code, "ESCAPE") && c.current(), e == null || e(t);
    },
    [o, e]
  ), G = r(
    (t) => {
      c.current((t == null ? void 0 : t.label) || ""), g(t);
    },
    [g]
  ), H = r(
    (t) => {
      f(t), R(t);
    },
    [R]
  );
  return /* @__PURE__ */ l("div", { onKeyDown: q, ...k, ref: (t) => {
    p.current = t, typeof s == "function" ? s(t) : s && (s.current = t);
  }, children: /* @__PURE__ */ l(
    J.Provider,
    {
      value: {
        value: n,
        options: D,
        placeholder: I,
        noOptionsText: L,
        loading: O,
        dropdownListLoading: A,
        tooltipOpen: P,
        clearInputOnFocus: _,
        onSelect: G,
        onFilter: H,
        onScrollEnd: j,
        inputValue: i,
        setInputValue: f,
        activeDescendant: z,
        dropdownOpen: o,
        setDropdownOpen: a,
        closeDropdown: b,
        dropdownListId: F,
        suggestRef: p,
        dropdownRef: S
      },
      children: C
    }
  ) });
}, K = N.forwardRef(U);
export {
  K as Suggest
};
