// @sberbusiness/triplex-next/components/Suggest/SuggestMobileDropdownContent.js
import { jsxs as c, Fragment as S, jsx as e } from "react/jsx-runtime";
import { useRef as B, useCallback as r } from "react";
import { useSuggestContext as F } from "./SuggestContext.js";
import "../Dropdown/Dropdown.js";
import "../Dropdown/DropdownListContext.js";
import "../Dropdown/desktop/DropdownDesktop.js";
import "../Dropdown/desktop/DropdownList.js";
import "../../chunks/DropdownListItem-D2hzyBGc.js";
import "../Dropdown/mobile/DropdownMobile.js";
import { DropdownMobileBody as I } from "../Dropdown/mobile/DropdownMobileBody.js";
import { DropdownMobileClose as N } from "../Dropdown/mobile/DropdownMobileClose.js";
import "../Dropdown/mobile/DropdownMobileFooter.js";
import { DropdownMobileHeader as O } from "../Dropdown/mobile/DropdownMobileHeader.js";
import "../Dropdown/mobile/DropdownMobileInner.js";
import { DropdownMobileInput as L } from "../Dropdown/mobile/DropdownMobileInput.js";
import { DropdownMobileList as R } from "../Dropdown/mobile/DropdownMobileList.js";
import { DropdownMobileListItem as _ } from "../Dropdown/mobile/DropdownMobileListItem.js";
import { DropdownMobileLoader as v } from "../Dropdown/mobile/DropdownMobileLoader.js";
import "../Typography/Title.js";
import { Text as E } from "../Typography/Text.js";
import "../Typography/Caption.js";
import { ETextSize as H, EFontType as j } from "../Typography/enums.js";
const k = "suggestDropdownMobileBody__a090dc85", z = "suggestDropdownMobileNoOptionsText__3acd4df6", m = {
  suggestDropdownMobileBody: k,
  suggestDropdownMobileNoOptionsText: z
}, po = () => {
  const {
    value: n,
    options: s,
    inputValue: i,
    placeholder: g,
    noOptionsText: u,
    loading: a,
    clearInputOnFocus: d,
    dropdownListLoading: w,
    onSelect: b,
    onFilter: t,
    onScrollEnd: l,
    closeDropdown: p
  } = F(), f = B(null), D = r(
    (o) => {
      if (l === void 0)
        return;
      const { clientHeight: T, scrollHeight: y, scrollTop: C } = o.target;
      y - C === T && l();
    },
    [l]
  ), h = r(() => p(), [p]), M = r(
    (o) => {
      t(o.target.value);
    },
    [t]
  ), x = r(() => {
    i.length !== 0 && d === !0 && t("");
  }, [i.length, d, t]);
  return /* @__PURE__ */ c(S, { children: [
    /* @__PURE__ */ c(O, { closeButton: () => /* @__PURE__ */ e(N, { onClick: h }), children: [
      /* @__PURE__ */ e(
        L,
        {
          value: i,
          placeholder: g,
          autoFocus: !0,
          onFocus: x,
          onChange: M
        }
      ),
      a && /* @__PURE__ */ e(v, {})
    ] }),
    /* @__PURE__ */ e(I, { className: m.suggestDropdownMobileBody, onScroll: D, children: s.length === 0 ? /* @__PURE__ */ e(
      E,
      {
        className: m.suggestDropdownMobileNoOptionsText,
        type: j.PRIMARY,
        size: H.B2,
        tag: "div",
        children: u
      }
    ) : /* @__PURE__ */ e(R, { loading: w, ref: f, children: s.map((o) => /* @__PURE__ */ e(
      _,
      {
        id: o.id,
        selected: o.id === (n == null ? void 0 : n.id),
        onSelect: () => {
          b(o);
        },
        children: o.content || o.label
      },
      o.id
    )) }) })
  ] });
};
export {
  po as SuggestMobileDropdownContent
};
