// @sberbusiness/triplex-next/components/Tooltip/components/mobile/TooltipMobile.js
import { jsxs as l, Fragment as b, jsx as s } from "react/jsx-runtime";
import { useContext as f, useRef as u, useEffect as M } from "react";
import { TooltipContext as C } from "../../TootlipContext.js";
import { Portal as x } from "../../../Portal/Portal.js";
import { DropdownMobile as _ } from "../../../Dropdown/mobile/DropdownMobile.js";
import { DropdownMobileBody as h } from "../../../Dropdown/mobile/DropdownMobileBody.js";
import { TooltipMobileCloseButton as y } from "./components/TooltipMobileCloseButton.js";
import { c as B } from "../../../../chunks/vendor-Ba_pyhza.js";
const T = "tooltipMobile__cd044b3c", g = "tooltipMobileContent__fd6829ce", H = "headerless__8cd0ef50", r = {
  tooltipMobile: T,
  tooltipMobileContent: g,
  headerless: H
}, z = ({
  children: N,
  className: d,
  renderContainer: R,
  targetRef: j,
  disableAdaptiveMode: D,
  isOpen: o,
  onShow: t,
  ...c
}) => {
  const { elements: e, setTooltipOpen: p } = f(C), i = u(null), m = B(r.tooltipMobile, { [r.headerless]: e.mobileHeader === null }, d);
  M(() => {
    o && (t == null || t(i.current));
  }, [o]);
  const a = () => {
    var n;
    return /* @__PURE__ */ s(y, { ...(n = e.closeButton) == null ? void 0 : n.props });
  };
  return /* @__PURE__ */ l(b, { children: [
    e.target,
    /* @__PURE__ */ s(x, { container: document.body, children: /* @__PURE__ */ l(
      _,
      {
        className: m,
        tabIndex: -1,
        opened: o,
        setOpened: p,
        ...c,
        ref: i,
        children: [
          e.mobileHeader,
          /* @__PURE__ */ l(h, { className: r.tooltipMobileContent, children: [
            e.body,
            e.link,
            e.mobileHeader === null && a()
          ] })
        ]
      }
    ) })
  ] });
};
export {
  z as TooltipMobile
};
