// @sberbusiness/triplex-next/components/TabsLine/TabsLine.js
import { jsx as r } from "react/jsx-runtime";
import { c as f } from "../../chunks/vendor-Ba_pyhza.js";
import { MobileView as n } from "../MobileView/MobileView.js";
import { TabsLineDesktop as b } from "./components/TabsLineDesktop.js";
import { TabsLineMobile as c } from "./components/TabsLineMobile.js";
import { s as d } from "../../chunks/TabsLine.module-DKAp48Sx.js";
import "./components/TabsLineDropdown.js";
import "./components/TabsLineItem.js";
const v = ({
  className: s,
  dropdownTargetHtmlAttributes: t,
  maxVisible: e,
  onChangeTab: i,
  paddingX: p,
  selectedId: o,
  tabs: m,
  size: a,
  ...l
}) => /* @__PURE__ */ r(
  "div",
  {
    role: "tablist",
    className: f(s, d.tabsLineWrapper),
    ...l,
    "data-paddingx-size": p,
    children: /* @__PURE__ */ r(
      n,
      {
        fallback: /* @__PURE__ */ r(
          b,
          {
            tabs: m,
            onChangeTab: i,
            selectedId: o,
            dropdownTargetHtmlAttributes: t,
            maxVisible: e,
            size: a
          }
        ),
        children: /* @__PURE__ */ r(c, { tabs: m, onChangeTab: i, selectedId: o })
      }
    )
  }
);
export {
  v as TabsLine
};
