// @sberbusiness/triplex-next/components/ListMaster/components/ListMasterHeader.js
import { jsx as f } from "react/jsx-runtime";
import u, { useRef as o, useEffect as m } from "react";
import { c as w } from "../../../chunks/vendor-Ba_pyhza.js";
const p = "listMasterHeader__2d5f3949", y = "sticky__3b68f33d", c = {
  listMasterHeader: p,
  sticky: y
}, H = u.forwardRef(
  ({ children: i, className: n, sticky: a = !0, ...l }, t) => {
    const e = o(null), r = o(0);
    m(() => {
      if (e.current)
        return r.current = e.current.getBoundingClientRect().height, window.scrollTo({ top: window.scrollY + r.current }), () => {
          window.scrollTo({ top: Math.max(window.scrollY - r.current, 0) });
        };
    }, []);
    const d = (s) => {
      e.current = s, typeof t == "function" ? t(s) : t && (t.current = s);
    };
    return /* @__PURE__ */ f(
      "div",
      {
        className: w(
          c.listMasterHeader,
          {
            [c.sticky]: a
          },
          n
        ),
        ...l,
        ref: d,
        children: i
      }
    );
  }
);
H.displayName = "ListMasterHeader";
export {
  H as ListMasterHeader
};
