// @sberbusiness/triplex-next/components/SegmentedControl/SegmentedControlSegment.js
import { jsx as a } from "react/jsx-runtime";
import { useContext as x } from "react";
import { c as L } from "../../chunks/vendor-Ba_pyhza.js";
import { ButtonBase as N } from "../Button/ButtonBase.js";
import { ESegmentedControlType as n } from "./enums.js";
import { SegmentedControlContext as _ } from "./SegmentedControlContext.js";
const y = "segmentedControlSegment__62bdda84", E = "selected__38b19bb8", I = "content__a3f72550", r = {
  segmentedControlSegment: y,
  selected: E,
  content: I
}, T = ({
  children: s,
  className: i,
  value: e,
  title: c,
  disabled: g,
  onClick: o,
  ...S
}) => {
  const {
    type: m,
    value: d,
    disabled: p,
    onSegmentSelect: l
  } = x(_), t = (() => {
    switch (m) {
      case n.SINGLE:
        return e === d;
      case n.MULTIPLE:
        return d.includes(e);
    }
  })(), b = L(
    r.segmentedControlSegment,
    { [r.selected]: t },
    "hoverable",
    { active: t },
    i
  ), f = () => {
    if (c)
      return c;
    if (typeof s == "string")
      return s;
  }, C = (u) => {
    switch (m) {
      case n.SINGLE:
        l({ value: e, selected: !0 });
        break;
      case n.MULTIPLE:
        l({ value: e, selected: !t });
        break;
    }
    o == null || o(u);
  };
  return /* @__PURE__ */ a(
    N,
    {
      className: b,
      title: f(),
      disabled: g || p,
      "aria-pressed": t,
      onClick: C,
      ...S,
      children: /* @__PURE__ */ a("span", { className: r.content, children: s })
    }
  );
};
T.displayName = "SegmentedControlSegment";
export {
  T as SegmentedControlSegment
};
