// @sberbusiness/triplex-next/components/SegmentedControl/SegmentedControl.js
import { ESegmentedControlTheme as o, ESegmentedControlType as m } from "./enums.js";
import { SegmentedControlSegment as y } from "./SegmentedControlSegment.js";
import { jsx as c } from "react/jsx-runtime";
import E from "react";
import { c as b } from "../../chunks/vendor-Ba_pyhza.js";
import { SegmentedControlContext as R } from "./SegmentedControlContext.js";
import { createSizeToClassNameMap as T } from "../../utils/classNameMaps.js";
const h = "segmentedControl__92bc0336", x = "general1__39d67cc0", L = "secondary1__c26891fa", A = "general2__616f869e", M = "secondary2__39f5f0ce", G = "sm__96ba2861", O = "md__94bd2d9b", j = "lg__d92a95ae", e = {
  segmentedControl: h,
  general1: x,
  secondary1: L,
  general2: A,
  secondary2: M,
  sm: G,
  md: O,
  lg: j
}, v = {
  [o.GENERAL_1]: e.general1,
  [o.GENERAL_2]: e.general2,
  [o.SECONDARY_1]: e.secondary1,
  [o.SECONDARY_2]: e.secondary2
}, z = T(e), D = Object.assign(
  E.forwardRef(
    ({ children: d, className: l, value: n, theme: g, type: r, size: _, disabled: i, onSelect: t, ...f }, C) => {
      const p = b(
        e.segmentedControl,
        v[g],
        z[_],
        l
      ), S = ({
        selected: a,
        value: s
      }) => {
        switch (r) {
          case m.SINGLE:
            a && t(s);
            break;
          case m.MULTIPLE:
            t(a ? [...n, s] : [...n].filter((N) => N !== s));
        }
      };
      return /* @__PURE__ */ c(
        R.Provider,
        {
          value: {
            type: r,
            value: n,
            disabled: !!i,
            onSegmentSelect: S
          },
          children: /* @__PURE__ */ c("div", { className: p, ...f, "data-tx": "1.4.0", ref: C, children: d })
        }
      );
    }
  ),
  {
    Segment: y
  }
);
D.displayName = "SegmentedControl";
export {
  D as SegmentedControl
};
