// @sberbusiness/triplex-next/components/AccordionBase/protected/AccordionBase.js
var c = Object.defineProperty;
var l = (a, s, e) => s in a ? c(a, s, { enumerable: !0, configurable: !0, writable: !0, value: e }) : a[s] = e;
var o = (a, s, e) => l(a, typeof s != "symbol" ? s + "" : s, e);
import { jsxs as O, Fragment as m, jsx as g } from "react/jsx-runtime";
import h from "react";
import { ExpandAnimation as f } from "../../ExpandAnimation/ExpandAnimation.js";
import { a as u, c as x } from "../../../chunks/vendor-Ba_pyhza.js";
class I extends h.Component {
  constructor(e) {
    super(e);
    o(this, "displayName", "AccordionBase");
    // Уникальный id, для передачи a11y aria-атрибутов в header и body.
    o(this, "instanceId", `AccordionBase-${u()}`);
    o(this, "renderHeader", () => {
      const { renderHeader: e, isOpen: n } = this.props, { animating: t, controlled: i, isOpen: r } = this.state, p = !!(i ? n : r);
      return e({
        animating: t,
        ariaAttributes: {
          controls: this.instanceId,
          expanded: p.toString()
        },
        isOpen: p,
        toggle: this.toggle
      });
    });
    /**
     * Раскрывает/скрывает контент аккордеона.
     */
    o(this, "toggle", () => {
      const { controlled: e, isOpen: n } = this.state, { toggle: t, isOpen: i } = this.props;
      e ? t && t(!!!(e ? i : n)) : this.setState((p) => ({
        animating: !0,
        isOpen: !p.isOpen
      }));
    });
    o(this, "handleAnimationEnd", () => {
      const { controlled: e, isOpen: n } = this.state, { onToggle: t, isOpen: i } = this.props, r = !!(e ? i : n);
      this.setState({ animating: !1 }, () => {
        t && t(r);
      });
    });
    o(this, "renderBody", () => {
      const { expandAnimationClassName: e, renderBody: n, isOpen: t } = this.props, { animating: i, controlled: r, isOpen: p } = this.state, d = !!(r ? t : p);
      return /* @__PURE__ */ g(
        f,
        {
          className: x(e, {
            "cssClass[globalExpandAnimationAnimating]": i,
            "cssClass[globalExpandAnimationOpened]": d
          }),
          expanded: d,
          onEnd: this.handleAnimationEnd,
          id: this.instanceId,
          children: n({
            animating: i,
            isOpen: d,
            toggle: this.toggle
          })
        }
      );
    });
    this.state = {
      animating: !1,
      controlled: typeof e.isOpen < "u",
      isOpen: !!e.isOpen
    };
  }
  componentDidUpdate(e) {
    const { isOpen: n } = this.props, { isOpen: t } = e;
    n !== t && this.setState({ animating: !0 });
  }
  render() {
    const e = this.renderHeader(), n = this.renderBody();
    return /* @__PURE__ */ O(m, { children: [
      e,
      n
    ] });
  }
}
export {
  I as AccordionBase
};
