// @sberbusiness/triplex-next/components/SelectField/SelectField.js
import { jsx as o } from "react/jsx-runtime";
import S, { useState as T, useRef as d } from "react";
import { SelectExtendedField as s } from "../SelectExtendedField/SelectExtendedField.js";
import "../SelectExtendedField/components/SelectExtendedFieldDropdown.js";
import "../SelectExtendedField/components/SelectExtendedFieldTarget.js";
import { SelectExtendedFieldDropdownDefault as v } from "../SelectExtendedField/components/SelectExtendedFieldDropdownDefault.js";
import { DropdownListContext as F } from "../Dropdown/DropdownListContext.js";
import { a as I } from "../../chunks/vendor-Ba_pyhza.js";
const C = S.forwardRef((c, t) => {
  const {
    children: L,
    className: m,
    value: r,
    size: a,
    options: p,
    onChange: f,
    placeholder: u,
    loading: n,
    status: b,
    "aria-labelledby": g,
    dropdownListItemClassName: N,
    mobileTitle: D,
    targetProps: w,
    ...x
  } = c, [l, h] = T(), y = d(null), i = d(I()), R = (e) => {
    y.current = e, typeof t == "function" ? t(e) : t && (t.current = e);
  };
  return /* @__PURE__ */ o(s, { className: m, renderTarget: (e) => /* @__PURE__ */ o(
    s.Target,
    {
      label: r == null ? void 0 : r.label,
      placeholder: u,
      loading: n,
      status: b,
      role: "combobox",
      "aria-controls": i.current,
      "aria-activedescendant": l,
      "aria-labelledby": g,
      ref: R,
      size: a,
      ...w,
      ...e
    }
  ), closeOnTab: !0, ...x, children: (e) => /* @__PURE__ */ o(F.Provider, { value: { activeDescendant: l, setActiveDescendant: h }, children: /* @__PURE__ */ o(
    v,
    {
      ...e,
      size: a,
      loading: n,
      listId: i.current,
      mobileTitle: D,
      onChange: f,
      options: p,
      value: r
    }
  ) }) });
});
C.displayName = "SelectField";
export {
  C as SelectField
};
